From 60676071d10b75b4af844290cd2894bae6be1064 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 26 May 2015 11:22:17 +1000
Subject: [PATCH 1/5] dpnet: Add DirectPlay network structures

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/wine/dplaysp.h | 56 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)

diff --git a/include/wine/dplaysp.h b/include/wine/dplaysp.h
index 1459232..8e6bbbb 100644
--- a/include/wine/dplaysp.h
+++ b/include/wine/dplaysp.h
@@ -370,4 +370,60 @@ typedef HRESULT (WINAPI *LPDPSP_SPINIT)(LPSPINITDATA);
  */
 extern DWORD gdwDPlaySPRefCount;
 
+/* Direct Play 8 Structures */
+#include <pshpack1.h>
+
+struct EnumHeader
+{
+    BYTE Leadbyte;
+    BYTE Command;
+    WORD EnumPayload;
+};
+
+struct EnumQuery
+{
+    struct EnumHeader header;
+    BYTE   QueryType;
+    GUID   ApplicationGUID;
+};
+
+struct EnumQueryResponse
+{
+    struct EnumHeader header;
+    DWORD  ReplyOffset;
+    DWORD  ResponseSize;
+    DWORD  ApplicationDescSize;
+    DWORD  ApplicationDescFlags;
+    DWORD  MaxPlayers;
+    DWORD  CurrentPlayers;
+    DWORD  SessionNameOffset;
+    DWORD  SessionNameSize;
+    DWORD  PasswordOffset;
+    DWORD  PasswordSize;
+    DWORD  ReservedDataOffset;
+    DWORD  ReservedDataSize;
+    DWORD  ApplicationReservedDataOffset;
+    DWORD  ApplicationReservedDataSize;
+    GUID   ApplicationInstanceGUID;
+    GUID   ApplicationGUID;
+    /*
+    The following are optional and here for reference only.
+    WCHAR SessionName  (variable)
+    WCHAR Password     (variable)
+    WCHAR ReservedData (variable)
+    WCHAR ApplicationReservedData (variable)
+    WCHAR ApplicationData (variable)
+    */
+};
+
+#include "poppack.h"
+
+#define PACKET_ENUM_QUERY     0x02
+#define PACKET_ENUM_RESPONSE  0x03
+
+#define PACKET_QUERY_GUID     0x01
+#define PACKET_QUERY_NO_GUID  0x02
+
+#define APPLICATIONDESCSIZE         0x00000050
+
 #endif
-- 
1.9.1

