From c55b34136ad381a6eeee94b53a4bd89333f9bb22 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 13 Dec 2016 12:46:22 +1100
Subject: [PATCH 16/43] dpnet: Only Send an ACK packet when requested

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index 9e44c5eaeee..400df975c0d 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -989,6 +989,11 @@ static HRESULT dplay_connected(SOCKET sock, struct sockaddr_in *serveraddr, DWOR
             {
                 FIXME("Packet not support.... size(%d), command 0x%02x, control 0x%04x, seq 0x%08x, nseq 0x%04x\n", n, packet->command, packet->control, packet->seq, packet->nseq);
             }
+
+            if(connected && (packet->command | PACKET_COMMAND_POLL))
+            {
+                dframe_packet_sack_ack(sock, serveraddr, packet->seq, packet->seq);
+            }
         }
     }
 
-- 
2.23.0

