From 98a42c6b240859a7a9b446cc0467adc92428b925 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 2 Dec 2016 11:22:06 +1100
Subject: [PATCH 17/17] dpnet: Implement IDirectPlay8Client Send

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 39 ++++++++++++++++++++++++++++++++++++---
 1 file changed, 36 insertions(+), 3 deletions(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index 080f6bd..06b5cf6 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -1062,9 +1062,42 @@ static HRESULT WINAPI IDirectPlay8ClientImpl_Send(IDirectPlay8Client *iface,
         const DPN_BUFFER_DESC * const prgBufferDesc, const DWORD cBufferDesc, const DWORD dwTimeOut,
         void * const pvAsyncContext, DPNHANDLE * const phAsyncHandle, const DWORD dwFlags)
 {
-  IDirectPlay8ClientImpl *This = impl_from_IDirectPlay8Client(iface);
-  FIXME("(%p):(%p,%p,%x): Stub\n", This, pvAsyncContext, phAsyncHandle, dwFlags);
-  return DPN_OK; 
+    IDirectPlay8ClientImpl *This = impl_from_IDirectPlay8Client(iface);
+    struct DFRAME_PACKET_TYPE *packet;
+    int err;
+    int size;
+
+    FIXME("(%p):(%p,%d,%u,%p,%p,%x): Semi-stub\n", This, prgBufferDesc, cBufferDesc, dwTimeOut, pvAsyncContext, phAsyncHandle, dwFlags);
+
+    if(This->sock == INVALID_SOCKET)
+        return E_FAIL;
+
+    size = sizeof(*packet) + prgBufferDesc->dwBufferSize;
+    packet = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, size);
+
+    if(!packet)
+        return E_OUTOFMEMORY;
+
+    if(cBufferDesc != 1)
+        FIXME("Currently only one DPN_BUFFER_DESC supported!\n.");
+
+    packet->command = PACKET_COMMAND_DATA | PACKET_COMMAND_SEQUENTIAL | PACKET_COMMAND_POLL |
+                      PACKET_COMMAND_NEW_MSG | PACKET_COMMAND_END_MSG;
+    packet->control  = 0;
+    packet->seq      = 8;  /* TODO */
+    packet->nseq     = 8;  /* TODO */
+    packet->type     = 0x01; /* TRANS_USERDATA_SEND_MESSAGE */
+
+    memcpy( ((char*)packet) + sizeof(struct DFRAME_PACKET_TYPE), prgBufferDesc->pBufferData, prgBufferDesc->dwBufferSize);
+
+    /* TODO - Store the Server addr in Connect patch */
+    err = sendto(This->sock, (void *)&packet, size, 0, (struct sockaddr *)&This->serveraddr, sizeof(struct sockaddr_in));
+    if(err == -1)
+        ERR("Sendto failed (%d).\n", WSAGetLastError());
+    else
+        FIXME("sack sent %d\n", err);
+
+    return DPN_OK;
 }
 
 static HRESULT WINAPI IDirectPlay8ClientImpl_GetSendQueueInfo(IDirectPlay8Client *iface,
-- 
1.9.1

