From 6b7e52cdd0239bb04b0be7de760bb8e356e5dccc Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 17 Jan 2017 10:05:11 +1100
Subject: [PATCH 25/42] dpnet: Implement IDirectPlay8Peer SendTo

---
 dlls/dpnet/peer.c | 36 +++++++++++++++++++++++++++++++++++-
 1 file changed, 35 insertions(+), 1 deletion(-)

diff --git a/dlls/dpnet/peer.c b/dlls/dpnet/peer.c
index f9e939a..93b00e9 100644
--- a/dlls/dpnet/peer.c
+++ b/dlls/dpnet/peer.c
@@ -358,9 +358,43 @@ static HRESULT WINAPI IDirectPlay8PeerImpl_SendTo(IDirectPlay8Peer *iface, const
         const DPN_BUFFER_DESC *pBufferDesc, const DWORD cBufferDesc, const DWORD dwTimeOut,
         void * const pvAsyncContext, DPNHANDLE * const phAsyncHandle, const DWORD dwFlags)
 {
+    IDirectPlay8PeerImpl* This = impl_from_IDirectPlay8Peer(iface);
+    struct DFRAME_PACKET_TYPE *packet;
+    int err;
+    int size;
+    static int sss = 3;
+
     FIXME("(%p)->(%x,%p,%x,%x,%p,%p,%x): stub\n", iface, dpnId, pBufferDesc, cBufferDesc, dwTimeOut, pvAsyncContext, phAsyncHandle, dwFlags);
 
-    return DPNERR_GENERIC;
+    if(This->sock == INVALID_SOCKET)
+        return E_FAIL;
+
+    size = sizeof(struct DFRAME_PACKET) + pBufferDesc->dwBufferSize;
+    packet = heap_alloc(size);
+
+    if(!packet)
+        return E_OUTOFMEMORY;
+
+    if(cBufferDesc != 1)
+        FIXME("Currently only one DPN_BUFFER_DESC supported!\n.");
+
+    packet->command = PACKET_COMMAND_DATA | PACKET_COMMAND_SEQUENTIAL | PACKET_COMMAND_POLL |
+                      PACKET_COMMAND_NEW_MSG | PACKET_COMMAND_END_MSG;
+    packet->control  = 0;
+    packet->seq      = sss++;  /* TODO */
+    packet->nseq     = sss;    /* TODO */
+
+    memcpy(&packet->type, pBufferDesc->pBufferData, pBufferDesc->dwBufferSize);
+
+    err = sendto(This->sock, (void *)packet, size, 0, (struct sockaddr *)&This->serveraddr, This->addr_size);
+    if(err == -1)
+        ERR("Sendto failed (%d).\n", WSAGetLastError());
+    else
+        FIXME("sent sent %d\n", err);
+
+    heap_free(packet);
+
+    return DPN_OK;
 }
 
 static HRESULT WINAPI IDirectPlay8PeerImpl_GetSendQueueInfo(IDirectPlay8Peer *iface,
-- 
1.9.1

