From 444a34ee6c160fc86408282a6cf22874b6e03538 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 13 Dec 2016 12:43:46 +1100
Subject: [PATCH 26/35] dpnet: Handle User data packets

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index fd1957c..ea09e97 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -946,7 +946,21 @@ static HRESULT dplay_connected(SOCKET sock, struct sockaddr_in *serveraddr, DWOR
                             }
                             else
                             {
-                                FIXME("data not support.... command 0x%02x, control 0x%04x, seq 0x%08x, nseq 0x%04x, type %08x\n", packet->command, packet->control, packet->seq, packet->nseq, dframe->type);
+                                DPNMSG_RECEIVE received;
+
+                                TRACE("Passign on user data\n");
+
+                                received.dwSize             = sizeof(received);
+                                received.dpnidSender        = 0;
+                                received.pvPlayerContext    = NULL;
+                                received.pReceiveData       = &dframe->type;
+                                received.dwReceiveDataSize  = n - sizeof(struct DFRAME_PACKET);
+                                received.hBufferHandle      = 0;
+                                received.dwReceiveFlags     = 0;
+
+                                hr = msghandler(NULL, DPN_MSGID_RECEIVE, &received);
+                                if(hr != S_OK)
+                                    ERR("Message Handle for DPN_MSGID_TERMINATE_SESSION failed (0x%08x).\n", hr);
                             }
 
                         }
-- 
1.9.1

