From a2b6781121a4a684029c394cd393425415967b7c Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 13 Dec 2016 12:46:22 +1100
Subject: [PATCH 27/48] dpnet: Only Send an ACK packet when requested

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index ea09e97676..16bf9bd7b3 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -990,6 +990,11 @@ static HRESULT dplay_connected(SOCKET sock, struct sockaddr_in *serveraddr, DWOR
             {
                 FIXME("Packet not support.... size(%d), command 0x%02x, control 0x%04x, seq 0x%08x, nseq 0x%04x\n", n, packet->command, packet->control, packet->seq, packet->nseq);
             }
+
+            if(connected && (packet->command | PACKET_COMMAND_POLL))
+            {
+                dframe_packet_sack_ack(sock, serveraddr, packet->seq, packet->seq);
+            }
         }
     }
 
-- 
2.11.0

