From 6e9bd1714372303c9888213f6e55c10850b297a4 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 19 Jan 2017 16:25:54 +1100
Subject: [PATCH 28/43] dpnet: Fake success from IDirectPlay8Peer
 EnumPlayersAndGroups

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/peer.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/dlls/dpnet/peer.c b/dlls/dpnet/peer.c
index 1062123ff92..0ae4b659e26 100644
--- a/dlls/dpnet/peer.c
+++ b/dlls/dpnet/peer.c
@@ -508,7 +508,19 @@ static HRESULT WINAPI IDirectPlay8PeerImpl_EnumPlayersAndGroups(IDirectPlay8Peer
 {
     FIXME("(%p)->(%p,%p,%x): stub\n", iface, prgdpnid, pcdpnid, dwFlags);
 
-    return DPNERR_GENERIC;
+    if(pcdpnid)
+    {
+        int i = 0;
+        FIXME("pcdpnid %d\n", *pcdpnid);
+
+        for(; i < *pcdpnid; i++)
+        {
+            prgdpnid[i] = 0; /* FIXME */
+        }
+
+    }
+
+    return S_OK;
 }
 
 static HRESULT WINAPI IDirectPlay8PeerImpl_EnumGroupMembers(IDirectPlay8Peer *iface,
-- 
2.23.0

