From c6aa19fd6192582757c37c549cffdd594886a878 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 31 Dec 2016 11:11:48 +1100
Subject: [PATCH 30/33] dpnet: temp trace fixup send_packet_query wrt guid

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index fd01cf5b76..1499f60fe1 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -467,7 +467,7 @@ static int send_packet_enum_query(SOCKET sock, GUID application, char *data, DWO
 
     struct ENUM_QUERY_GUID *query;
     static DWORD payloadvalue = 0;
-    BOOL has_guid = IsEqualGUID(&application, &IID_NULL);
+    BOOL has_guid = !IsEqualGUID(&application, &IID_NULL);
     int err;
     int size = sizeof(struct ENUM_QUERY) + data_size;
     int offset = sizeof(struct ENUM_QUERY);
@@ -489,6 +489,8 @@ static int send_packet_enum_query(SOCKET sock, GUID application, char *data, DWO
     if(data_size)
         memcpy( packet + offset, data, data_size);
 
+    FIXME("%s %d size %d\n", debugstr_guid(&application), data_size, size);
+
     err = sendto(sock, (void *)packet, size, 0, (struct sockaddr *)from, sizeof(struct sockaddr_in));
     if(err == -1)
         ERR("Sendto failed (%d).\n", WSAGetLastError());
@@ -509,6 +511,8 @@ void process_enumhost(SOCKET sock, struct sockaddr_in host, IDirectPlay8Address
     int received = 0;
     HRESULT hr = DPN_OK;
 
+    TRACE("guid %s\n", debugstr_guid(&application));
+
     setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO, (char *)&timeout, sizeof(timeout));
 
     while(TRUE)
-- 
2.11.0

