From 26d01541914de448dfc7d8e38bb8ef2329f276db Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 13 Jan 2017 14:06:48 +1100
Subject: [PATCH 33/33] dpnet: Support IP4/6 in create_sender_address

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index 1c58cb1..828ca74 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -435,7 +435,7 @@ static HRESULT WINAPI IDirectPlay8ClientImpl_EnumServiceProviders(IDirectPlay8Cl
     return enum_services_providers(pguidServiceProvider, pSPInfoBuffer, pcbEnumData, pcReturned);
 }
 
-static IDirectPlay8Address *create_sender_address(struct sockaddr_in6 *addr)
+static IDirectPlay8Address *create_sender_address(SOCKADDR_STORAGE *addr)
 {
     IDirectPlay8Address *sender = NULL;
     char ip[INET6_ADDRSTRLEN] = {0};
@@ -446,9 +446,13 @@ static IDirectPlay8Address *create_sender_address(struct sockaddr_in6 *addr)
     if(FAILED(hr))
         return sender;
 
-    inet_ntop(addr->sin6_family, &addr->sin6_addr, ip, sizeof(ip));
-    port = ntohs(addr->sin6_port);
-    TRACE("Server IP address %s(%d)\n", ip, port);
+    inet_ntop(addr->ss_family, (char*)addr + sizeof(addr->ss_family), ip, sizeof(ip));
+    if(addr->ss_family == AF_INET6)
+        port = ntohs( ((struct sockaddr_in6 *)addr)->sin6_port);
+    else
+        port = ntohs( ((struct sockaddr_in *)addr)->sin_port);
+
+    TRACE("Server IP address %s port (%d)\n", ip, port);
 
     if(FAILED(IDirectPlay8Address_AddComponent(sender, DPNA_KEY_HOSTNAME, &ip, strlen(ip)+1, DPNA_DATATYPE_STRING_ANSI)))
     {
@@ -508,7 +512,7 @@ void process_enumhost(SOCKET sock, struct sockaddr_in host, IDirectPlay8Address
     char buffer[1024];
     int buflen = sizeof(buffer);
     DWORD starttime = 0;
-    struct sockaddr_in6 from;
+    SOCKADDR_STORAGE from;
     int fromlen = sizeof(from);
     int received = 0;
     HRESULT hr = DPN_OK;
-- 
1.9.1

