From 5942e3f0c48e04d173ba097c171ac3e679362649 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 8 Feb 2017 12:36:51 +1100
Subject: [PATCH 38/43] dpnet: Support Alternate Addresses

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/dppacket.h | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/dlls/dpnet/dppacket.h b/dlls/dpnet/dppacket.h
index 16d2013..c4f8e90 100644
--- a/dlls/dpnet/dppacket.h
+++ b/dlls/dpnet/dppacket.h
@@ -201,6 +201,24 @@ struct DN_NAMETABLE_VERSION
     DWORD ver_not_used;
 };
 
+/* 2.2.9 DN_ALTERNATE_ADDRESS IPv4 */
+struct DN_ALTERNATE_ADDRESS_v4
+{
+    BYTE size;
+    BYTE family;
+    SHORT port;
+    IN_ADDR address;
+};
+
+/* 2.2.10 DN_ALTERNATE_ADDRESS IPv6 */
+struct DN_ALTERNATE_ADDRESS_v6
+{
+    BYTE size;
+    BYTE family;
+    SHORT port;
+    IN6_ADDR address;
+};
+
 struct SACK_PACKET
 {
     BYTE command;
-- 
1.9.1

