From 8348403a07673018bbe6a7534acae3410d633711 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 7 Feb 2017 10:03:30 +1100
Subject: [PATCH 47/48] dpnet: Ignore Namespace Version change packets.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c   |  7 +++++++
 dlls/dpnet/dppacket.h | 12 ++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index 01fad5b613..0ecf4050ca 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -1127,6 +1127,13 @@ static HRESULT dplay_connected(SOCKET sock, void *serveraddr, size_t addrsize, D
                         FIXME("DN_MSG_INTERNAL_INSTRUCT_CONNECT dpnid 0x%08x, version 0x%08x\n", instruct->dpnid, instruct->version);
                         break;
                     }
+                    case DN_MSG_INTERNAL_NAMETABLE_VERSION:
+                    case DN_MSG_INTERNAL_RESYNC_VERSION:
+                    {
+                        struct DN_NAMETABLE_VERSION *namespace = (struct DN_NAMETABLE_VERSION *)&dframe->type;
+                        FIXME("DN_NAMETABLE_VERSION ignored. type 0x%08x, version 0x%08x\n", dframe->type, namespace->version);
+                        break;
+                    }
                     default:
                         /* System Messages */
                         if(packet->command & PACKET_COMMAND_USER_1)
diff --git a/dlls/dpnet/dppacket.h b/dlls/dpnet/dppacket.h
index c01960ae52..16d2013d16 100644
--- a/dlls/dpnet/dppacket.h
+++ b/dlls/dpnet/dppacket.h
@@ -192,6 +192,15 @@ struct DN_NAMETABLE_ENTRY_INFO
     DWORD url_size;
 };
 
+/* 2.2.2.4 DN_NAMETABLE_VERSION */
+/* 2.2.2.5 DN_RESYNC_VERSION    */
+struct DN_NAMETABLE_VERSION
+{
+    DWORD type;
+    DWORD version;
+    DWORD ver_not_used;
+};
+
 struct SACK_PACKET
 {
     BYTE command;
@@ -256,6 +265,9 @@ struct CFRAME_PACKET
 #define DN_MSG_INTERNAL_CONNECT_FAILED      0x000000c5
 #define DN_MSG_INTERNAL_INSTRUCT_CONNECT    0x000000c6
 
+#define DN_MSG_INTERNAL_NAMETABLE_VERSION   0x000000c9
+#define DN_MSG_INTERNAL_RESYNC_VERSION      0x000000ca
+
 #define DP_OBECT_TYPE_CLIENT      0x00000002
 #define DP_OBECT_TYPE_PEER        0x00000004
 
-- 
2.11.0

