From eea7ab5696b7ead7480b3e10e395bbf2070aa5b5 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 9 Sep 2025 12:12:16 +1000
Subject: [PATCH 5/8] include: Add coclass InkDisp

---
 include/msinkaut.idl | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/msinkaut.idl b/include/msinkaut.idl
index 743dec82a40..ad61842c794 100644
--- a/include/msinkaut.idl
+++ b/include/msinkaut.idl
@@ -1006,4 +1006,15 @@ cpp_quote("#endif /* _WINGDI_ */")
             [in] InkCollectorEventInterest EventId,
             [in] VARIANT_BOOL Listen);
     }
+
+    [
+        progid("msinkaut.InkObject.1"),
+        vi_progid("msinkaut.InkObject"),
+        threading(both),
+        uuid(937c1a34-151d-4610-9ca6-a8cc9bdb5d83)
+    ]
+    coclass InkDisp
+    {
+        [default] interface IInkDisp;
+    }
 }
-- 
2.51.0

