From 8d810fc6121e97932e4030c48c5f70c61a59826c Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 9 Sep 2025 15:14:35 +1000
Subject: [PATCH 7/8] inkobj: Correct install path.

---
 loader/wine.inf.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 1e505bb9117..0c444ef8028 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -1026,6 +1026,7 @@ HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Version",2,"14.42.344
 16422,Windows Media Player,wmplayer.exe
 16422,Windows NT\Accessories,wordpad.exe
 16427,Microsoft Shared\Ink,tiptsf.dll
+16427,Microsoft Shared\Ink,inkobj.dll
 16427,System\OLE DB,oledb32.dll
 16427,System\OLE DB,msdaps.dll
 16427,System\OLE DB,msdasql.dll
@@ -1094,6 +1095,7 @@ HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Version",2,"14.42.344
 16422,Windows NT\Accessories,wordpad.exe
 16427,Microsoft Shared\TextConv,
 16427,Microsoft Shared\Ink,tiptsf.dll
+16427,Microsoft Shared\Ink,inkobj.dll
 16427,System\OLE DB,oledb32.dll
 16427,System\OLE DB,msdaps.dll
 16427,System\OLE DB,msdasql.dll
-- 
2.51.0

