From 630d32ccfecda39f8ef304ff6cf46b41f6ba7498 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 26 Oct 2020 16:04:30 +1100
Subject: [PATCH 3/4] widl: Support FC_HYPER

---
 tools/widl/typegen.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/widl/typegen.c b/tools/widl/typegen.c
index 5d0f24be06f..62344be1925 100644
--- a/tools/widl/typegen.c
+++ b/tools/widl/typegen.c
@@ -1735,6 +1735,7 @@ static unsigned int write_conf_or_var_desc(FILE *file, const type_t *cont_type,
             case FC_LONG:
                 param_type = FC_LONG;
                 break;
+            case FC_HYPER:
             case FC_ULONG:
                 param_type = FC_ULONG;
                 break;
-- 
2.29.2

