From 8f9f513b08563b9758003f0097fb63c698d79cd2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 26 Oct 2020 15:17:46 +1100
Subject: [PATCH] include: Correct type of DB_UPARAMS

---
 include/oledb.idl | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/oledb.idl b/include/oledb.idl
index c56460262f5..fe5e45c32b9 100644
--- a/include/oledb.idl
+++ b/include/oledb.idl
@@ -37,10 +37,15 @@ typedef ULONG_PTR DBORDINAL;
 typedef ULONG_PTR DBBKMARK;
 typedef DWORD_PTR DB_DWRESERVE;
 typedef ULONG_PTR DBREFCOUNT;
-typedef ULONG_PTR DB_UPARAMS;
 typedef LONG_PTR  DB_LPARAMS;
 typedef DWORD_PTR DBHASHVALUE;
 
+cpp_quote("#ifdef _WIN64")
+typedef ULONGLONG DB_UPARAMS;
+cpp_quote("#else")
+cpp_quote("typedef ULONG DB_UPARAMS;")
+cpp_quote("#endif")
+
 #include "dbs.idl"
 
 #include "access.idl"
-- 
2.28.0

