From 3ca71103f7abba0d8b106923dddc764b89baffe3 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 29 Oct 2020 14:45:48 +1100
Subject: [PATCH] include: Add DBPROPSTATUSENUM enum values

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/dbs.idl | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/include/dbs.idl b/include/dbs.idl
index 19f92b3c05f..cbdb89d23b2 100644
--- a/include/dbs.idl
+++ b/include/dbs.idl
@@ -79,6 +79,22 @@ enum DBPROPOPTIONENUM {
 
 typedef DWORD DBPROPSTATUS;
 
+enum DBPROPSTATUSENUM {
+    DBPROPSTATUS_OK             = 0,
+    DBPROPSTATUS_NOTSUPPORTED   = 1,
+    DBPROPSTATUS_BADVALUE       = 2,
+    DBPROPSTATUS_BADOPTION      = 3,
+    DBPROPSTATUS_BADCOLUMN      = 4,
+    DBPROPSTATUS_NOTALLSETTABLE = 5,
+    DBPROPSTATUS_NOTSETTABLE    = 6,
+    DBPROPSTATUS_NOTSET         = 7,
+    DBPROPSTATUS_CONFLICTING    = 8
+};
+
+enum DBPROPSTATUSENUM21 {
+    DBPROPSTATUS_NOTAVAILABLE   = 9
+};
+
 typedef struct tagDBPROP {
     DBPROPID dwPropertyID;
     DBPROPOPTIONS dwOptions;
-- 
2.28.0

