From aa12f64a8024392590210b78f3fb68916c5920c9 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 19 Oct 2022 16:20:19 +1100
Subject: [PATCH] d3d11: CheckFeatureSupport support more options

D3D11_FEATURE_FORMAT_SUPPORT2
---
 dlls/d3d11/device.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/dlls/d3d11/device.c b/dlls/d3d11/device.c
index 061fd57fe09..9fe737dfead 100644
--- a/dlls/d3d11/device.c
+++ b/dlls/d3d11/device.c
@@ -4240,6 +4240,21 @@ static HRESULT STDMETHODCALLTYPE d3d11_device_CheckFeatureSupport(ID3D11Device2
             return d3d11_device_CheckFormatSupport(iface, data->InFormat, &data->OutFormatSupport);
         }
 
+        case D3D11_FEATURE_FORMAT_SUPPORT2:
+        {
+            D3D11_FEATURE_DATA_FORMAT_SUPPORT2 *data = feature_support_data;
+            if (feature_support_data_size != sizeof(*data))
+            {
+                WARN("Invalid size %u for D3D11_FEATURE_FORMAT_SUPPORT.\n", feature_support_data_size);
+                return E_INVALIDARG;
+            }
+
+            data->OutFormatSupport2 = 0;
+            return S_OK;
+        }
+
+
+
         default:
             FIXME("Unhandled feature %#x.\n", feature);
             return E_NOTIMPL;
-- 
2.43.0

