From 495f300aa43e5e6488c12e95f9ffec67c1915d53 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 19 Oct 2022 16:20:19 +1100
Subject: [PATCH] d3d11: CheckFeatureSupport support
 D3D11_FEATURE_FORMAT_SUPPORT2 feature

---
 dlls/d3d11/device.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/dlls/d3d11/device.c b/dlls/d3d11/device.c
index 5c819f4e983..e0d994d6be5 100644
--- a/dlls/d3d11/device.c
+++ b/dlls/d3d11/device.c
@@ -4805,6 +4805,18 @@ static HRESULT STDMETHODCALLTYPE d3d11_device_CheckFeatureSupport(ID3D11Device2
             return S_OK;
         }
 
+        case D3D11_FEATURE_FORMAT_SUPPORT2:
+        {
+            D3D11_FEATURE_DATA_FORMAT_SUPPORT2 *data = feature_support_data;
+            if (feature_support_data_size != sizeof(*data))
+            {
+                WARN("Invalid size %u for D3D11_FEATURE_FORMAT_SUPPORT2.\n", feature_support_data_size);
+                return E_INVALIDARG;
+            }
+
+            return d3d11_device_CheckFormatSupport(iface, data->InFormat, &data->OutFormatSupport2);
+        }
+
         default:
             FIXME("Unhandled feature %#x.\n", feature);
             return E_NOTIMPL;
-- 
2.47.2

