From d66b4a78d926233ce7159b38ec2c0ddcb481d449 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 7 Dec 2017 11:44:41 +1100
Subject: [PATCH 1/3] New DLL audioses

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 configure    | 2 ++
 configure.ac | 1 +
 2 files changed, 3 insertions(+)

diff --git a/configure b/configure
index 236d8e50837..edbbf356c56 100755
--- a/configure
+++ b/configure
@@ -1153,6 +1153,7 @@ enable_atl80
 enable_atl90
 enable_atlthunk
 enable_atmlib
+enable_audioses
 enable_authz
 enable_avicap32
 enable_avifil32
@@ -20467,6 +20468,7 @@ wine_fn_config_makefile dlls/atl90 enable_atl90
 wine_fn_config_makefile dlls/atlthunk enable_atlthunk
 wine_fn_config_makefile dlls/atlthunk/tests enable_tests
 wine_fn_config_makefile dlls/atmlib enable_atmlib
+wine_fn_config_makefile dlls/audioses enable_audioses
 wine_fn_config_makefile dlls/authz enable_authz
 wine_fn_config_makefile dlls/avicap32 enable_avicap32
 wine_fn_config_makefile dlls/avifil32 enable_avifil32
diff --git a/configure.ac b/configure.ac
index db3fd901b3c..528b35130a0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3066,6 +3066,7 @@ WINE_CONFIG_MAKEFILE(dlls/atl90)
 WINE_CONFIG_MAKEFILE(dlls/atlthunk)
 WINE_CONFIG_MAKEFILE(dlls/atlthunk/tests)
 WINE_CONFIG_MAKEFILE(dlls/atmlib)
+WINE_CONFIG_MAKEFILE(dlls/audioses)
 WINE_CONFIG_MAKEFILE(dlls/authz)
 WINE_CONFIG_MAKEFILE(dlls/avicap32)
 WINE_CONFIG_MAKEFILE(dlls/avifil32)
-- 
2.29.2

