From be883bd8edc8bd843c5c5d39cf9160c14f841243 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 24 May 2017 16:14:39 +1000
Subject: [PATCH 2/2] windowscodecs: Register WICImagingFactory2 interface

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/windowscodecs/windowscodecs_wincodec.idl | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dlls/windowscodecs/windowscodecs_wincodec.idl b/dlls/windowscodecs/windowscodecs_wincodec.idl
index fd7ff01..33c0d4a 100644
--- a/dlls/windowscodecs/windowscodecs_wincodec.idl
+++ b/dlls/windowscodecs/windowscodecs_wincodec.idl
@@ -35,6 +35,13 @@ coclass PSFactoryBuffer { interface IFactoryBuffer; }
 coclass WICImagingFactory { interface IWICImagingFactory; }
 
 [
+    helpstring("WIC Imaging Factory 2"),
+    threading(both),
+    uuid(317d06e8-5f24-433d-bdf7-79ce68d8abc2)
+]
+coclass WICImagingFactory2 { interface IWICImagingFactory2; }
+
+[
     helpstring("WIC BMP Decoder"),
     threading(both),
     uuid(6b462062-7cbf-400d-9fdb-813dd10f2778)
-- 
1.9.1

