From 4eed3af2e392b013a525b008e00ba2c5844c5e64 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 24 May 2017 16:14:39 +1000
Subject: [PATCH 2/2] windowscodecs: Register WICImagingFactory1/2 interfaces

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/windowscodecs/windowscodecs_wincodec.idl | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/dlls/windowscodecs/windowscodecs_wincodec.idl b/dlls/windowscodecs/windowscodecs_wincodec.idl
index fd7ff01067..ac79512853 100644
--- a/dlls/windowscodecs/windowscodecs_wincodec.idl
+++ b/dlls/windowscodecs/windowscodecs_wincodec.idl
@@ -35,6 +35,20 @@ coclass PSFactoryBuffer { interface IFactoryBuffer; }
 coclass WICImagingFactory { interface IWICImagingFactory; }
 
 [
+    helpstring("WIC Imaging Factory"),
+    threading(both),
+    uuid(cacaf262-9370-4615-a13b-9f5539da4c0a)
+]
+coclass WICImagingFactory1 { interface IWICImagingFactory; }
+
+[
+    helpstring("WIC Imaging Factory"),
+    threading(both),
+    uuid(317d06e8-5f24-433d-bdf7-79ce68d8abc2)
+]
+coclass WICImagingFactory2 { interface IWICImagingFactory2; }
+
+[
     helpstring("WIC BMP Decoder"),
     threading(both),
     uuid(6b462062-7cbf-400d-9fdb-813dd10f2778)
-- 
2.11.0

