From b59fd3978a21cdc3efec1c09d27773405650828f Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 17 Jun 2019 10:39:13 +1000
Subject: [PATCH] ieframe: Improve IOleInPlaceSiteEx OnInPlaceActivateEx.

---
 dlls/ieframe/client.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/dlls/ieframe/client.c b/dlls/ieframe/client.c
index 4846e28a05..322f4f23ac 100644
--- a/dlls/ieframe/client.c
+++ b/dlls/ieframe/client.c
@@ -297,7 +297,13 @@ static HRESULT WINAPI InPlaceSite_OnInPlaceActivateEx(IOleInPlaceSiteEx *iface,
     TRACE("(%p)->(%p, %x)\n", This, pfNoRedraw, dwFlags);
 
     /* FIXME: Avoid redraw, when possible */
-    *pfNoRedraw = FALSE;
+    if(pfNoRedraw)
+    {
+        if(GetUpdateRect(This->hwnd, NULL, TRUE))
+            *pfNoRedraw = TRUE;
+        else
+            *pfNoRedraw = FALSE;
+    }
 
     if (dwFlags) {
         FIXME("dwFlags not supported (%x)\n", dwFlags);
-- 
2.17.1

