From 399e5d32873ca7a79f73ce11bd3539e3b3a9d0cf Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sun, 24 Jan 2016 15:47:00 +1100
Subject: [PATCH 1/2] include: Add GdipGetPenTransform,GdipSetPenTransform
 declarations

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/gdiplusflat.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/gdiplusflat.h b/include/gdiplusflat.h
index 29119b4..2e22ff4 100644
--- a/include/gdiplusflat.h
+++ b/include/gdiplusflat.h
@@ -619,6 +619,7 @@ GpStatus WINGDIPAPI GdipGetPenDashCount(GpPen*,INT*);
 GpStatus WINGDIPAPI GdipGetPenDashOffset(GpPen*,REAL*);
 GpStatus WINGDIPAPI GdipGetPenDashStyle(GpPen*,GpDashStyle*);
 GpStatus WINGDIPAPI GdipGetPenMode(GpPen*,GpPenAlignment*);
+GpStatus WINGDIPAPI GdipGetPenTransform(GpPen *, GpMatrix *);
 GpStatus WINGDIPAPI GdipResetPenTransform(GpPen*);
 GpStatus WINGDIPAPI GdipScalePenTransform(GpPen*,REAL,REAL,GpMatrixOrder);
 GpStatus WINGDIPAPI GdipSetPenBrushFill(GpPen*,GpBrush*);
@@ -637,6 +638,7 @@ GpStatus WINGDIPAPI GdipSetPenLineJoin(GpPen*,GpLineJoin);
 GpStatus WINGDIPAPI GdipSetPenMode(GpPen*,GpPenAlignment);
 GpStatus WINGDIPAPI GdipSetPenMiterLimit(GpPen*,REAL);
 GpStatus WINGDIPAPI GdipSetPenStartCap(GpPen*,GpLineCap);
+GpStatus WINGDIPAPI GdipSetPenTransform(GpPen *, GpMatrix *);
 GpStatus WINGDIPAPI GdipSetPenWidth(GpPen*,REAL);
 GpStatus WINGDIPAPI GdipGetPenDashCap197819(GpPen*,GpDashCap*);
 GpStatus WINGDIPAPI GdipGetPenEndCap(GpPen*,GpLineCap*);
-- 
2.7.0.rc3

