From f413dc4ff6aac5503d838dad1aa895d955bb4648 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 28 May 2024 10:25:41 +1000
Subject: [PATCH] include: Add *_SHIFT macros

---
 include/gdipluspixelformats.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/gdipluspixelformats.h b/include/gdipluspixelformats.h
index 2a2a3ccbcfd..91726628be1 100644
--- a/include/gdipluspixelformats.h
+++ b/include/gdipluspixelformats.h
@@ -22,6 +22,12 @@
 typedef DWORD ARGB;
 typedef INT PixelFormat;
 
+#define ALPHA_SHIFT 24
+#define RED_SHIFT   16
+#define GREEN_SHIFT 8
+#define BLUE_SHIFT  0
+#define ALPHA_MASK  ((ARGB) 0xff << ALPHA_SHIFT)
+
 #define    PixelFormatIndexed   0x00010000
 #define    PixelFormatGDI       0x00020000
 #define    PixelFormatAlpha     0x00040000
-- 
2.43.0

