From 386eba75f6606a92d0564b131caf1a39637a5245 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 25 Oct 2024 07:48:56 +1100
Subject: [PATCH] include: Add Get/Set for gdiplys Color class

---
 include/gdipluscolor.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/gdipluscolor.h b/include/gdipluscolor.h
index 39b7cb6debd..46acc51deed 100644
--- a/include/gdipluscolor.h
+++ b/include/gdipluscolor.h
@@ -35,6 +35,11 @@ class Color
 {
 protected:
     ARGB Argb;
+
+public:
+    ARGB GetValue() const { return Argb; }
+
+    void SetValue(ARGB argb) { Argb = argb; }
 };
 
 #else /* end of c++ typedefs */
-- 
2.45.2

