From 6f9e7052c5e244e7af1e733b53eb1336428569d1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 30 Jul 2024 16:14:03 +1000
Subject: [PATCH] include: gdiplus fix ups - to merged otherse

---
 include/gdiplusgraphics.h | 15 ---------------
 include/gdiplusmatrix.h   |  4 ----
 2 files changed, 19 deletions(-)

diff --git a/include/gdiplusgraphics.h b/include/gdiplusgraphics.h
index dbe6a09efa5..43bbce8ba26 100644
--- a/include/gdiplusgraphics.h
+++ b/include/gdiplusgraphics.h
@@ -18,21 +18,6 @@
 #ifndef _GDIPLUSGRAPHICS_H
 #define _GDIPLUSGRAPHICS_H
 
-class Brush;
-class CachedBitmap;
-class Graphics;
-class GraphicsPath;
-class Font;
-class Image;
-class ImageAttributes;
-class Metafile;
-class Pen;
-class Region;
-class StringFormat;
-
-typedef Graphics GpGraphics;
-typedef Pen GpPen;
-
 class Graphics : public GdiplusBase
 {
 public:
diff --git a/include/gdiplusmatrix.h b/include/gdiplusmatrix.h
index c8062cf6acb..660d61ecf61 100644
--- a/include/gdiplusmatrix.h
+++ b/include/gdiplusmatrix.h
@@ -18,10 +18,6 @@
 #ifndef _GDIPLUS_MATRIX
 #define _GDIPLUS_MATRIX
 
-class Matrix;
-
-typedef Matrix GpMatrix;
-
 class Matrix : public GdiplusBase
 {
 public:
-- 
2.45.2

