From d275aabadf5e64b5df8fd382f4fc39b838e56cf2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 30 Aug 2024 11:50:35 +1000
Subject: [PATCH] include: Move ColorChannelLUT outside of if __cplusplus.

---
 include/gdipluscolormatrix.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/gdipluscolormatrix.h b/include/gdipluscolormatrix.h
index 391095ed8fb..2bbfaa6d97a 100644
--- a/include/gdipluscolormatrix.h
+++ b/include/gdipluscolormatrix.h
@@ -60,9 +60,10 @@ enum HistogramFormat
     HistogramFormatA,
 };
 
+typedef BYTE ColorChannelLUT[256];
+
 #ifndef __cplusplus
 
-typedef BYTE ColorChannelLUT[256];
 typedef enum ColorAdjustType ColorAdjustType;
 typedef enum ColorMatrixFlags ColorMatrixFlags;
 typedef enum HistogramFormat HistogramFormat;
-- 
2.45.2

