From e2d637a853b34f049cb657204cb24169aa5fc786 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 30 Aug 2024 12:12:02 +1000
Subject: [PATCH] include/gdiplustypes.h: Add basic constructors for Rect/RectF

---
 include/gdiplustypes.h | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/include/gdiplustypes.h b/include/gdiplustypes.h
index 8350ec45782..a82c46ca3b8 100644
--- a/include/gdiplustypes.h
+++ b/include/gdiplustypes.h
@@ -190,6 +190,18 @@ public:
     REAL Y;
     REAL Width;
     REAL Height;
+
+    RectF()
+       : X(0), Y(0), Width(0), Height(0)
+    {
+    }
+
+    RectF(REAL x, REAL y, REAL width, REAL height)
+       : X(x), Y(y), Width(width), Height(height)
+
+    {
+    }
+
 };
 
 /* FIXME: missing the methods. */
@@ -200,6 +212,17 @@ public:
     INT Y;
     INT Width;
     INT Height;
+
+    Rect()
+       : X(0), Y(0), Width(0), Height(0)
+    {
+    }
+
+    Rect(INT x, INT y, INT width, INT height)
+       : X(x), Y(y), Width(width), Height(height)
+
+    {
+    }
 };
 
 class CharacterRange
-- 
2.45.2

