From 4fe61725b887c1c8da539b2928706c8436f5d559 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 8 Feb 2017 10:36:08 +1100
Subject: [PATCH] uuid: Add comsvcs.h

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/uuid/uuid.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/uuid/uuid.c b/dlls/uuid/uuid.c
index 4834deeae9..92ec9d01c8 100644
--- a/dlls/uuid/uuid.c
+++ b/dlls/uuid/uuid.c
@@ -88,6 +88,7 @@ DEFINE_GUID(GUID_NULL,0,0,0,0,0,0,0,0,0,0,0);
 #include "msdasc.h"
 #include "netcfgx.h"
 #include "shimgdata.h"
+#include "comsvcs.h"
 
 /* FIXME: cguids declares GUIDs but does not define their values */
 
-- 
2.11.0

