From 6808e02a68c3d05e2d1909d2a4b7f875b640d3cb Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 4 Jun 2018 15:45:40 +1000
Subject: [PATCH 2/2] dpnsvr: Only allow dpnsvr to start once

---
 programs/dpnsvr/main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/programs/dpnsvr/main.c b/programs/dpnsvr/main.c
index 72e64d0..6afa570 100644
--- a/programs/dpnsvr/main.c
+++ b/programs/dpnsvr/main.c
@@ -201,6 +201,15 @@ int wmain(int argc, WCHAR *argv[])
     socklen_t addrlen;
     FD_SET readset;
     TIMEVAL delay;
+    HANDLE started;
+
+    started = CreateEventA(NULL, TRUE, FALSE, "__wine_dpnsvr_started");
+    if(WaitForSingleObject(started, 0) == WAIT_OBJECT_0)
+    {
+        WINE_WARN("Attempting to start dpnsvr twice\n");
+        return 0;
+    }
+    SetEvent(started);
 
     if (WSAStartup(MAKEWORD(2, 2), &data))
         return 1;
@@ -249,6 +258,7 @@ int wmain(int argc, WCHAR *argv[])
     }
 
     KillTimer(NULL, timer);
+    CloseHandle(started);
     heap_free(socks);
 
     WSACleanup();
-- 
1.9.1

