From 97e673d265558bf9dba6bb6e57e2345d04e60122 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 4 Jun 2018 15:45:40 +1000
Subject: [PATCH 2/3] dpnsvr: Only allow dpnsvr to start once

---
 programs/dpnsvr/main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/programs/dpnsvr/main.c b/programs/dpnsvr/main.c
index 17ea2a9cd05..50b212dd5d4 100644
--- a/programs/dpnsvr/main.c
+++ b/programs/dpnsvr/main.c
@@ -200,6 +200,15 @@ int __cdecl wmain(int argc, WCHAR *argv[])
     socklen_t addrlen;
     FD_SET readset;
     TIMEVAL delay;
+    HANDLE started;
+
+    started = CreateEventA(NULL, TRUE, FALSE, "__wine_dpnsvr_started");
+    if(WaitForSingleObject(started, 0) == WAIT_OBJECT_0)
+    {
+        WINE_WARN("Attempting to start dpnsvr twice\n");
+        return 0;
+    }
+    SetEvent(started);
 
     if (WSAStartup(MAKEWORD(2, 2), &data))
         return 1;
@@ -248,6 +257,7 @@ int __cdecl wmain(int argc, WCHAR *argv[])
     }
 
     KillTimer(NULL, timer);
+    CloseHandle(started);
     heap_free(socks);
 
     WSACleanup();
-- 
2.30.2

