From d278a897b19c4e0127e445197a4f1fd34fff8298 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 6 Jan 2023 12:10:22 +1100
Subject: [PATCH] include: Add D3DKMT_ENUMADAPTERS2 structure

---
 include/ddk/d3dkmthk.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/ddk/d3dkmthk.h b/include/ddk/d3dkmthk.h
index b65fcb8c9c6..69239a45fd9 100644
--- a/include/ddk/d3dkmthk.h
+++ b/include/ddk/d3dkmthk.h
@@ -667,6 +667,20 @@ typedef struct _D3DKMT_ESCAPE
     D3DKMT_HANDLE      hContext;
 } D3DKMT_ESCAPE;
 
+typedef struct _D3DKMT_ADAPTERINFO
+{
+  D3DKMT_HANDLE hAdapter;
+  LUID          AdapterLuid;
+  ULONG         NumOfSources;
+  BOOL          bPrecisePresentRegionsPreferred;
+} D3DKMT_ADAPTERINFO;
+
+typedef struct _D3DKMT_ENUMADAPTERS2
+{
+  ULONG               NumAdapters;
+  D3DKMT_ADAPTERINFO *pAdapters;
+} D3DKMT_ENUMADAPTERS2;
+
 #ifdef __cplusplus
 extern "C"
 {
-- 
2.39.0

