From 695bd132a0b09196a2c1f49f2b3bc36e753877b1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 18 Apr 2017 14:00:18 +1000
Subject: [PATCH] oleaut32: Pass the correct variable to VariantClear

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/oleaut32/vartype.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/oleaut32/vartype.c b/dlls/oleaut32/vartype.c
index fc5a025..3513399 100644
--- a/dlls/oleaut32/vartype.c
+++ b/dlls/oleaut32/vartype.c
@@ -136,7 +136,7 @@ static HRESULT VARIANT_FromDisp(IDispatch* pdispIn, LCID lcid, void* pOut,
     if (SUCCEEDED(hRet))
     {
       VARIANT_CopyData(&dstVar, vt, pOut);
-      VariantClear(&srcVar);
+      VariantClear(&dstVar);
     }
   }
   else
-- 
1.9.1

