From 2dd1a5bf0e39a797669699e7090be44998bb2a1d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 24 Sep 2022 07:47:10 +1000
Subject: [PATCH] mfc: CWinAppEx Structure update

---
 dlls/mfc/afxstate.c | 73 ++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 72 insertions(+), 1 deletion(-)

diff --git a/dlls/mfc/afxstate.c b/dlls/mfc/afxstate.c
index b5673c23ea8..dc0917313d8 100644
--- a/dlls/mfc/afxstate.c
+++ b/dlls/mfc/afxstate.c
@@ -21,6 +21,14 @@
 
 #include "winbase.h"
 
+/*
+ *  We really need TCHAR to effectively support MFC.
+ *  Each version of the library has an ASCII and unicode version of the library.  Duplicating the
+ *  interfaces/structure doesn't make sense.
+ */
+#undef __WINESRC__
+#include "tchar.h"
+
 WINE_DEFAULT_DEBUG_CHANNEL(mfc);
 
 struct afx_module_state
@@ -64,6 +72,12 @@ int WINAPI AfxWinMainA(HINSTANCE inst, HINSTANCE prev, char *cmdline, int show)
     return 0;
 }
 
+typedef enum _AFX_HELP_TYPE
+{
+    afxWinHelp = 0,
+    afxHTMLHelp = 1
+} AFX_HELP_TYPE;
+
 /*
  * CWinApp
  * CWinAppEx
@@ -71,7 +85,64 @@ int WINAPI AfxWinMainA(HINSTANCE inst, HINSTANCE prev, char *cmdline, int show)
  */
 struct c_winappex
 {
-    BOOL smartupdate;
+    /* CObject */
+    /* - No Variables */
+
+    /* CCmdTarget */
+    void *cmdtarget[8];
+
+    /* CWinThread */
+    void *m_pMainWnd;        /* CWnd* */
+    void *m_pActiveWnd;      /* CWnd* */
+    BOOL m_bAutoDelete;
+    HANDLE m_hThread;
+    DWORD m_nThreadID;
+
+#ifndef _WIN64
+    char unk[14];
+#else
+    char unk[32];
+#endif
+
+    /* CWinApp */
+    HINSTANCE m_hInstance;
+    TCHAR *m_lpCmdLine;
+    int    m_nCmdShow;
+    TCHAR *m_pszAppName;
+    TCHAR *m_pszAppID;
+    TCHAR *m_pszRegistryKey;
+
+    void *win_app_unk;
+
+    BOOL m_bHelpMode;
+    TCHAR *m_pszExeName;
+    TCHAR *m_pszHelpFilePath;
+    TCHAR *m_pszProfileName;
+    AFX_HELP_TYPE m_eHelpType;
+
+#ifndef _WIN64
+    char unk2[56];
+#else
+    char unk2[104];
+#endif
+
+    DWORD m_dwRestartManagerSupportFlags;
+
+#ifndef _WIN64
+    char unk3[80];
+#else
+    char unk3[96];
+#endif
+
+    /* CWinAppEx */
+    BOOL m_bForceImageReset;
+
+#ifndef _WIN64
+    char unk4[16];
+#else
+    char unk4[14];
+#endif
+    void *vtbl;
 };
 
 /*
-- 
2.35.1

