From e073f05f073c07ffd26ad7b1b6b409ad6c30fb68 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 21 Jun 2017 14:36:03 +1000
Subject: [PATCH 1/3] include: Add BackgroundCopyManager3_0 coclass

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/bits3_0.idl | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/include/bits3_0.idl b/include/bits3_0.idl
index 6cf48b7..cb6c309 100644
--- a/include/bits3_0.idl
+++ b/include/bits3_0.idl
@@ -21,6 +21,7 @@
 
 #ifndef DO_NO_IMPORTS
 import "bits.idl";
+import "bits2_0.idl";
 #endif
 
 [
@@ -32,3 +33,39 @@ interface IBackgroundCopyCallback2 : IBackgroundCopyCallback
     HRESULT FileTransferred([in] IBackgroundCopyJob *job,
                             [in] IBackgroundCopyFile *file);
 }
+
+[
+    uuid(659cdeae-489e-11d9-a9cd-000d56965251),
+    odl
+]
+interface IBackgroundCopyJob4 : IBackgroundCopyJob3
+{
+cpp_quote("#define BG_JOB_ENABLE_PEERCACHING_CLIENT 0x0001")
+cpp_quote("#define BG_JOB_ENABLE_PEERCACHING_SERVER 0x0002")
+cpp_quote("#define BG_JOB_DISABLE_BRANCH_CACHE      0x0004")
+
+    HRESULT SetPeerCachingFlags(DWORD flags);
+    HRESULT GetPeerCachingFlags([out, ref] DWORD *flags);
+    HRESULT GetOwnerIntegrityLevel([out, ref] ULONG *level);
+    HRESULT GetOwnerElevationState([out, ref] BOOL *elevated);
+    HRESULT SetMaximumDownloadTime(ULONG timeout);
+    HRESULT GetMaximumDownloadTime([out,ref] ULONG *timeout);
+};
+
+[
+    uuid(659cdea6-489e-11d9-a9cd-000d56965251),
+    lcid(0x0000),
+    version(1.0)
+]
+library BackgroundCopyManager3_0
+{
+    [
+        uuid(659cdea7-489e-11d9-a9cd-000d56965251)
+    ]
+    coclass BackgroundCopyManager3_0
+    {
+        [default] interface IBackgroundCopyManager;
+    };
+
+    interface IBackgroundCopyJob4;
+}
-- 
1.9.1

