From 066c4a137d974fec3aa6fcd86341092af626844d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 14 Jul 2025 10:15:14 +1000
Subject: [PATCH] msado15: ADOConnectionConstruction15::WrapDSOandSession semi
 stub

---
 dlls/msado15/connection.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/dlls/msado15/connection.c b/dlls/msado15/connection.c
index 568eb8b12c9..ded4a401b27 100644
--- a/dlls/msado15/connection.c
+++ b/dlls/msado15/connection.c
@@ -954,7 +954,23 @@ static HRESULT WINAPI adoconstruct_WrapDSOandSession(ADOConnectionConstruction15
 {
     struct connection *connection = impl_from_ADOConnectionConstruction15( iface );
     FIXME("%p, %p, %p\n", connection, dso, session);
-    return E_NOTIMPL;
+
+    if (connection->session)
+        IUnknown_Release( connection->session );
+    connection->session = session;
+    if (connection->session)
+        IUnknown_AddRef(connection->session);
+
+    if (connection->dso)
+        IDBInitialize_Release( connection->dso );
+    connection->dso = NULL;
+    if (dso)
+        IUnknown_QueryInterface( dso, &IID_IDBInitialize, (void**)&connection->dso );
+
+    if (dso && session)
+        connection->state = adStateOpen;
+
+    return S_OK;
 }
 
 struct ADOConnectionConstruction15Vtbl ado_construct_vtbl =
-- 
2.47.2

