From 797b8ae2cb2b2e7c55f0c97332c0876a2e145e5d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 3 Aug 2022 16:25:11 +1000
Subject: [PATCH 1/2] dmime: Store WAVE data when Loading.

---
 dlls/dmime/segment.c | 41 +++++++++++++++++++++++++++++++++++++----
 1 file changed, 37 insertions(+), 4 deletions(-)

diff --git a/dlls/dmime/segment.c b/dlls/dmime/segment.c
index b1dd21c1242..695e533ddc4 100644
--- a/dlls/dmime/segment.c
+++ b/dlls/dmime/segment.c
@@ -33,6 +33,9 @@ typedef struct IDirectMusicSegment8Impl {
     DMUS_IO_SEGMENT_HEADER header;
     IDirectMusicGraph *pGraph;
     struct list Tracks;
+
+    PCMWAVEFORMAT wave_format;
+    void *wave_data;
 } IDirectMusicSegment8Impl;
 
 static inline IDirectMusicSegment8Impl *impl_from_IDirectMusicSegment8(IDirectMusicSegment8 *iface)
@@ -84,6 +87,9 @@ static ULONG WINAPI IDirectMusicSegment8Impl_Release(IDirectMusicSegment8 *iface
     TRACE("(%p) ref=%ld\n", This, ref);
 
     if (!ref) {
+        if (This->wave_data)
+            free(This->wave_data);
+
         HeapFree(GetProcessHeap(), 0, This);
         DMIME_UnlockModule();
     }
@@ -778,6 +784,35 @@ static inline IDirectMusicSegment8Impl *impl_from_IPersistStream(IPersistStream
     return CONTAINING_RECORD(iface, IDirectMusicSegment8Impl, dmobj.IPersistStream_iface);
 }
 
+static HRESULT parse_wave_form(IDirectMusicSegment8Impl *This, IStream *stream, const struct chunk_entry *riff)
+{
+    HRESULT hr;
+    struct chunk_entry chunk = {.parent = riff};
+
+    TRACE("Parsing segment wave in %p: %s\n", stream, debugstr_chunk(riff));
+
+    if ((hr = stream_get_chunk(stream, &chunk)) == S_OK) {
+        if (chunk.id == mmioFOURCC('f','m','t',' ')) {
+            if (FAILED(hr = stream_chunk_get_data(stream, &chunk, &This->wave_format, chunk.size)))
+                return hr;
+            TRACE("Wave Format tag %d\n", This->wave_format.wf.wFormatTag);
+        }
+
+        if ((hr = stream_get_chunk(stream, &chunk)) == S_OK) {
+            if (chunk.id == mmioFOURCC('d','a','t','a')) {
+                TRACE("Wave Data size %lu\n", chunk.size);
+                This->wave_data = malloc(chunk.size);
+                if (!This->wave_data)
+                    return E_OUTOFMEMORY;
+                if (FAILED(hr = stream_chunk_get_data(stream, &chunk, This->wave_data, chunk.size)))
+                    return hr;
+            }
+        }
+    }
+
+    return S_OK;
+}
+
 static HRESULT WINAPI seg_IPersistStream_Load(IPersistStream *iface, IStream *stream)
 {
     IDirectMusicSegment8Impl *This = impl_from_IPersistStream(iface);
@@ -807,10 +842,8 @@ static HRESULT WINAPI seg_IPersistStream_Load(IPersistStream *iface, IStream *st
 
     if (riff.type == DMUS_FOURCC_SEGMENT_FORM)
         hr = parse_segment_form(This, stream, &riff);
-    else {
-        FIXME("WAVE form loading not implemented\n");
-        hr = S_OK;
-    }
+    else
+        hr = parse_wave_form(This, stream, &riff);
 
     return hr;
 }
-- 
2.38.1

