From 13591c7209d8e371ecb5bba2e9b6ca266e1d0a25 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 1 Aug 2023 11:15:00 +1000
Subject: [PATCH 6/6] d3dx11_43: Add DXGI_FORMAT_R8G8B8A8_UNORM_SRGB mapping

---
 dlls/d3dx11_43/texture.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/d3dx11_43/texture.c b/dlls/d3dx11_43/texture.c
index 76b6615d74d..51c59e38483 100644
--- a/dlls/d3dx11_43/texture.c
+++ b/dlls/d3dx11_43/texture.c
@@ -62,6 +62,7 @@ wic_pixel_formats[] =
     { &GUID_WICPixelFormat32bppBGR,           DXGI_FORMAT_B8G8R8X8_UNORM },
     { &GUID_WICPixelFormat32bppBGRA,          DXGI_FORMAT_B8G8R8A8_UNORM },
     { &GUID_WICPixelFormat32bppRGBA,          DXGI_FORMAT_R8G8B8A8_UNORM },
+    { &GUID_WICPixelFormat32bppRGRA,          DXGI_FORMAT_R8G8B8A8_UNORM_SRGB },
     { &GUID_WICPixelFormat32bppRGBA1010102,   DXGI_FORMAT_R10G10B10A2_UNORM },
     { &GUID_WICPixelFormat32bppRGBA1010102XR, DXGI_FORMAT_R10G10B10_XR_BIAS_A2_UNORM },
     { &GUID_WICPixelFormat64bppRGBA,          DXGI_FORMAT_R16G16B16A16_UNORM },
-- 
2.40.1

