From 949571bd75144f97a8addd6db0337c7c82b06daa Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 29 May 2026 11:56:07 +1000
Subject: [PATCH] winedump: Corrections for generating stub dll

---
 tools/winedump/output.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/winedump/output.c b/tools/winedump/output.c
index b039f5d71ad..a35b7e641ce 100644
--- a/tools/winedump/output.c
+++ b/tools/winedump/output.c
@@ -217,7 +217,7 @@ void  output_c_preamble (void)
   fprintf (cfile,
            "/*\n * %s.dll\n *\n * Generated from %s by winedump.\n *\n"
            " * DO NOT SUBMIT GENERATED DLLS FOR INCLUSION INTO WINE!\n *\n */"
-           "\n\n#include \"config.h\"\n\n#include <stdarg.h>\n\n"
+           "\n\n#include <stdarg.h>\n\n"
            "#include \"windef.h\"\n#include \"winbase.h\"\n",
            OUTPUT_DLL_NAME, globals.input_name);
 
@@ -440,7 +440,7 @@ void  output_makefile (void)
   if (globals.forward_dll)
     fprintf (makefile, "IMPORTS   = %s", globals.forward_dll);
 
-  fprintf (makefile, "\n\nC_SRCS = \\\n\t%s_main.c\n", OUTPUT_DLL_NAME);
+  fprintf (makefile, "\n\nSOURCES = \\\n\t%s_main.c\n", OUTPUT_DLL_NAME);
 
   if (globals.forward_dll)
     fprintf (specfile,"#import %s.dll\n", globals.forward_dll);
-- 
2.53.0

