From 060db6e2949e1873c2c13f84ca6262bd6f2ad9f4 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 15 Apr 2016 14:22:08 +1000
Subject: [PATCH 1/7] mimeole.idl: Add missing Mime Property IDS

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/mimeole.idl | 52 +++++++++++++++++++++++++++++++++++++++++++++++++---
 1 file changed, 49 insertions(+), 3 deletions(-)

diff --git a/include/mimeole.idl b/include/mimeole.idl
index ab1239f..c10d8e0 100644
--- a/include/mimeole.idl
+++ b/include/mimeole.idl
@@ -84,10 +84,56 @@ cpp_quote("    PID_HDR_PATH       = 26,")
 cpp_quote("    PID_HDR_FOLLOWUPTO = 27,")
 cpp_quote("    PID_HDR_EXPIRES    = 28,")
 cpp_quote("    PID_HDR_CC         = 29,")
-
+cpp_quote("    PID_HDR_CONTROL    = 30,")
+cpp_quote("    PID_HDR_DISTRIB    = 31,")
+cpp_quote("    PID_HDR_KEYWORDS   = 32,")
+cpp_quote("    PID_HDR_SUMMARY    = 33,")
+cpp_quote("    PID_HDR_APPROVED   = 34,")
+cpp_quote("    PID_HDR_LINES      = 35,")
+cpp_quote("    PID_HDR_XREF       = 36,")
+cpp_quote("    PID_HDR_ORG        = 37,")
+cpp_quote("    PID_HDR_XNEWSRDR   = 38,")
+cpp_quote("    PID_HDR_XPRI       = 39,")
+cpp_quote("    PID_HDR_XMSPRI     = 40,")
+cpp_quote("    PID_PAR_FILENAME   = 41,")
+cpp_quote("    PID_PAR_BOUNDARY   = 42,")
+cpp_quote("    PID_PAR_CHARSET    = 43,")
+cpp_quote("    PID_PAR_NAME       = 44,")
+cpp_quote("    PID_ATT_FILENAME   = 45,")
+cpp_quote("    PID_ATT_GENFNAME   = 46,")
+cpp_quote("    PID_ATT_PRITYPE    = 47,")
+cpp_quote("    PID_ATT_SUBTYPE    = 48,")
+cpp_quote("    PID_ATT_NORMSUBJ   = 49,")
+cpp_quote("    PID_ATT_ILLEGAL    = 50,")
+cpp_quote("    PID_ATT_RENDERED   = 51,")
+cpp_quote("    PID_ATT_SENTTIME   = 52,")
+cpp_quote("    PID_ATT_RECVTIME   = 53,")
+cpp_quote("    PID_ATT_PRIORITY   = 54,")
+cpp_quote("    PID_HDR_COMMENT    = 55,")
+cpp_quote("    PID_HDR_ENCODING   = 56,")
+cpp_quote("    PID_HDR_ENCRYPTED  = 57,")
+cpp_quote("    PID_HDR_OFFSETS    = 58,")
+cpp_quote("    PID_HDR_XUNSENT    = 59,")
+cpp_quote("    PID_HDR_ARTICLEID  = 60,")
 cpp_quote("    PID_HDR_SENDER     = 61,")
-
-cpp_quote("    PID_HDR_INREPLYTO  = 77,")
+cpp_quote("    PID_ATT_SERVER     = 62,")
+cpp_quote("    PID_ATT_ACCOUNT    = 63,")
+cpp_quote("    PID_ATT_UIDL       = 64,")
+cpp_quote("    PID_ATT_STOREMSGID = 65,")
+cpp_quote("    PID_ATT_USERNAME   = 66,")
+cpp_quote("    PID_ATT_FORWARDTO  = 67,")
+cpp_quote("    PID_ATT_STOREFOLDERID = 68,")
+cpp_quote("    PID_ATT_GHOSTED       = 69,")
+cpp_quote("    PID_ATT_UNCACHEDSIZE  = 70,")
+cpp_quote("    PID_ATT_COMBINED      = 71,")
+cpp_quote("    PID_ATT_AUTOINLINED   = 72,")
+cpp_quote("    PID_HDR_DISP_NOTIFICATION_TO = 73,")
+cpp_quote("    PID_PAR_REPLYTYPE     = 74,")
+cpp_quote("    PID_PAR_FORMAT        = 75,")
+cpp_quote("    PID_ATT_FORMAT        = 76,")
+cpp_quote("    PID_HDR_INREPLYTO     = 77,")
+cpp_quote("    PID_ATT_ACCOUNTNAME   = 78,")
+cpp_quote("    PID_LAST              = 79,")
 cpp_quote("} MIMEPROPID;")
 
 cpp_quote("#define TYPEDID_MASK                  ((ULONG)0xffff)")
-- 
1.9.1

