From 227481fb0039121c2718acc0720501f8cefe3f12 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 25 Jul 2016 15:24:15 +1000
Subject: [PATCH] inetcomm: Semi-Implement IMimeMessage GetAddressFormat.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/inetcomm/mimeole.c | 40 ++++++++++++++++++++++++++++++++++++----
 1 file changed, 36 insertions(+), 4 deletions(-)

diff --git a/dlls/inetcomm/mimeole.c b/dlls/inetcomm/mimeole.c
index 2c481ab0d5..ce4e340531 100644
--- a/dlls/inetcomm/mimeole.c
+++ b/dlls/inetcomm/mimeole.c
@@ -2984,12 +2984,44 @@ static HRESULT WINAPI MimeMessage_GetAddressTypes(
 
 static HRESULT WINAPI MimeMessage_GetAddressFormat(
     IMimeMessage *iface,
-    DWORD dwAdrTypes,
+    DWORD addr_type,
     ADDRESSFORMAT format,
-    LPSTR *ppszFormat)
+    LPSTR *value)
 {
-    FIXME("(%p)->(%d, %d, %p)\n", iface, dwAdrTypes, format, ppszFormat);
-    return E_NOTIMPL;
+    PROPVARIANT prop;
+    HRESULT hr = MIME_E_NOT_FOUND;
+
+    TRACE("(%p)->(%d, %d, %p) semi-stub.\n", iface, addr_type, format, value);
+
+    PropVariantInit(&prop);
+
+    prop.vt = VT_LPSTR;
+    switch(addr_type)
+    {
+        case IAT_FROM:
+            hr = MimeMessage_GetBodyProp(iface, HBODY_ROOT, "From", 0, &prop);
+            break;
+        case IAT_TO:
+            hr = MimeMessage_GetBodyProp(iface, HBODY_ROOT, "To", 0, &prop);
+            break;
+        case IAT_CC:
+            hr = MimeMessage_GetBodyProp(iface, HBODY_ROOT, "Cc", 0, &prop);
+            break;
+        default:
+            FIXME("Unknown address type %d\n", addr_type);
+    }
+
+    if(SUCCEEDED(hr))
+    {
+        *value = CoTaskMemAlloc(strlen(prop.u.pszVal)+1 );
+        if(*value)
+            strcpy(*value, prop.u.pszVal);
+        else
+            hr = E_OUTOFMEMORY;
+    }
+    PropVariantClear(&prop);
+
+    return hr;
 }
 
 static HRESULT WINAPI MimeMessage_EnumAddressTypes(
-- 
2.17.1

