From c09fc71bd690e23affd47401ee2511cfcf0a0967 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 25 Jul 2016 15:24:15 +1000
Subject: [PATCH] inetcomm: Semi-Implement IMimeMessage GetAddressFormat.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/inetcomm/mimeole.c | 46 ++++++++++++++++++++++++++++++++++-------
 1 file changed, 39 insertions(+), 7 deletions(-)

diff --git a/dlls/inetcomm/mimeole.c b/dlls/inetcomm/mimeole.c
index 16553d39f78..3e3a1f721ef 100644
--- a/dlls/inetcomm/mimeole.c
+++ b/dlls/inetcomm/mimeole.c
@@ -2986,13 +2986,45 @@ static HRESULT WINAPI MimeMessage_GetAddressTypes(
 }
 
 static HRESULT WINAPI MimeMessage_GetAddressFormat(
-    IMimeMessage *iface,
-    DWORD dwAdrTypes,
-    ADDRESSFORMAT format,
-    LPSTR *ppszFormat)
-{
-    FIXME("(%p)->(%ld, %d, %p)\n", iface, dwAdrTypes, format, ppszFormat);
-    return E_NOTIMPL;
+     IMimeMessage *iface,
+     DWORD addr_type,
+     ADDRESSFORMAT format,
+     LPSTR *value)
+ {
+    PROPVARIANT prop;
+    HRESULT hr = MIME_E_NOT_FOUND;
+
+    TRACE("(%p)->(%ld, %d, %p) semi-stub.\n", iface, addr_type, format, value);
+
+    PropVariantInit(&prop);
+
+    prop.vt = VT_LPSTR;
+    switch(addr_type)
+    {
+        case IAT_FROM:
+            hr = MimeMessage_GetBodyProp(iface, HBODY_ROOT, "From", 0, &prop);
+            break;
+        case IAT_TO:
+            hr = MimeMessage_GetBodyProp(iface, HBODY_ROOT, "To", 0, &prop);
+            break;
+        case IAT_CC:
+            hr = MimeMessage_GetBodyProp(iface, HBODY_ROOT, "Cc", 0, &prop);
+            break;
+        default:
+            FIXME("Unknown address type %ld\n", addr_type);
+    }
+
+    if(SUCCEEDED(hr))
+    {
+        *value = CoTaskMemAlloc(strlen(prop.pszVal)+1 );
+        if(*value)
+            strcpy(*value, prop.pszVal);
+        else
+            hr = E_OUTOFMEMORY;
+    }
+    PropVariantClear(&prop);
+
+    return hr;
 }
 
 static HRESULT WINAPI MimeMessage_EnumAddressTypes(
-- 
2.47.2

