From 971bc0c7ab702daac6fd5932b74934a8f3d01661 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 19 Apr 2016 10:39:15 +1000
Subject: [PATCH 3/4] mimeole.idl: Add more MIME_E defines

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/mimeole.idl | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/include/mimeole.idl b/include/mimeole.idl
index d1b0bf8..7cebb77 100644
--- a/include/mimeole.idl
+++ b/include/mimeole.idl
@@ -48,6 +48,23 @@ cpp_quote("#define MIME_E_INVALID_HANDLE         0x800cce09")
 cpp_quote("#define MIME_E_CHARSET_TRANSLATE      0x800cce0a")
 cpp_quote("#define MIME_E_NOT_INITIALIZED        0x800cce0b")
 
+cpp_quote("#define MIME_E_INVALID_OPTION_VALUE   0x800cce1f")
+cpp_quote("#define MIME_E_INVALID_OPTION_ID      0x800cce20")
+cpp_quote("#define MIME_E_INVALID_HEADER_NAME    0x800cce21")
+cpp_quote("#define MIME_E_NOT_BOUND              0x800cce22")
+cpp_quote("#define MIME_E_MAX_SIZE_TOO_SMALL     0x800cce23")
+cpp_quote("#define MIME_E_MULTIPART_HAS_CHILDREN 0x800cce25")
+cpp_quote("#define MIME_E_INVALID_PROP_FLAGS     0x800cce26")
+cpp_quote("#define MIME_E_INVALID_ADDRESS_TYPE   0x800cce27")
+cpp_quote("#define MIME_E_INVALID_OBJECT_IID     0x800cce28")
+cpp_quote("#define MIME_E_MLANG_DLL_NOT_FOUND    0x800cce29")
+cpp_quote("#define MIME_E_ROOT_NOT_EMPTY         0x800cce2a")
+cpp_quote("#define MIME_E_MLANG_BAD_DLL          0x800cce2b")
+cpp_quote("#define MIME_E_REG_OPEN_KEY           0x800cce2c")
+cpp_quote("#define MIME_E_INVALID_INET_DATE      0x800cce2d")
+cpp_quote("#define MIME_E_INVALID_BODYTYPE       0x800cce2e")
+cpp_quote("#define MIME_E_INVALID_DELETE_TYPE    0x800cce2f")
+cpp_quote("#define MIME_E_OPTION_HAS_NO_VALUE    0x800cce30")
 cpp_quote("#define MIME_E_INVALID_CHARSET_TYPE   0x800cce31")
 
 cpp_quote("#define MIME_E_INVALID_TEXT_TYPE      0x800cce38")
-- 
1.9.1

