From 9591f58e24d80c9031cd0e2e6bab440a8ad3b0d1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 15 Aug 2016 09:24:40 +1000
Subject: [PATCH] inetcomm/test: Add Attachment tests

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/inetcomm/tests/mimeole.c | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/dlls/inetcomm/tests/mimeole.c b/dlls/inetcomm/tests/mimeole.c
index 011c19f..173874f 100644
--- a/dlls/inetcomm/tests/mimeole.c
+++ b/dlls/inetcomm/tests/mimeole.c
@@ -338,6 +338,45 @@ static void test_CreateMessage(void)
     hr = IMimeMessage_GetAttachments(msg, &count, &body_list);
     ok(hr == S_OK, "ret %08x\n", hr);
     ok(count == 2, "got %d\n", count);
+    if(count == 2)
+    {
+        ENCODINGTYPE encoding;
+        IMimeBody *attachment;
+
+        hr = IMimeMessage_BindToObject(msg, body_list[0], &IID_IMimeBody, (void**)&attachment);
+        ok(hr == S_OK, "ret %08x\n", hr);
+        if(hr == S_OK)
+        {
+            hr = IMimeBody_IsContentType(attachment, "multipart", NULL);
+            ok(hr == S_FALSE, "ret %08x\n", hr);
+
+            hr = IMimeBody_GetCurrentEncoding(attachment, &encoding);
+            ok(hr == S_OK, "ret %08x\n", hr);
+            todo_wine ok(encoding == IET_8BIT, "ret %d\n", encoding);
+
+            hr = IMimeBody_IsType(attachment, IBT_ATTACHMENT);
+            todo_wine ok(hr == S_FALSE, "ret %08x\n", hr);
+
+            IMimeBody_Release(attachment);
+        }
+
+        hr = IMimeMessage_BindToObject(msg, body_list[1], &IID_IMimeBody, (void**)&attachment);
+        ok(hr == S_OK, "ret %08x\n", hr);
+        if(hr == S_OK)
+        {
+            hr = IMimeBody_IsContentType(attachment, "multipart", NULL);
+            ok(hr == S_FALSE, "ret %08x\n", hr);
+
+            hr = IMimeBody_GetCurrentEncoding(attachment, &encoding);
+            ok(hr == S_OK, "ret %08x\n", hr);
+            todo_wine ok(encoding == IET_7BIT, "ret %d\n", encoding);
+
+            hr = IMimeBody_IsType(attachment, IBT_ATTACHMENT);
+            ok(hr == S_OK, "ret %08x\n", hr);
+
+            IMimeBody_Release(attachment);
+        }
+    }
     CoTaskMemFree(body_list);
 
     hr = IMimeBody_GetCharset(body, &hcs);
-- 
1.9.1

