From 0c580a41db36441255ae918ac52240dd3ec5c275 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 30 Jun 2016 12:27:51 +1000
Subject: [PATCH 4/7] inetcomm: Ignore option OID_CLEANUP_TREE_ON_SAVE

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/inetcomm/mimeole.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/inetcomm/mimeole.c b/dlls/inetcomm/mimeole.c
index d542844..f1fed8b 100644
--- a/dlls/inetcomm/mimeole.c
+++ b/dlls/inetcomm/mimeole.c
@@ -2509,6 +2509,9 @@ static HRESULT WINAPI MimeMessage_SetOption(
     case OID_SAVEBODY_KEEPBOUNDARY:
         FIXME("OID_SAVEBODY_KEEPBOUNDARY (value %d): ignoring\n", pValue->u.boolVal);
         break;
+    case OID_CLEANUP_TREE_ON_SAVE:
+        FIXME("OID_CLEANUP_TREE_ON_SAVE (value %d): ignoring\n", pValue->u.boolVal);
+        break;
     default:
         FIXME("Unhandled oid %08x\n", oid);
         hr = MIME_E_INVALID_OPTION_ID;
-- 
1.9.1

