From 3a23bb8498ca177301a588e6cbb200bc896a5999 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 27 Apr 2016 11:42:31 +1000
Subject: [PATCH 5/8] inetcomm: Support OID_SAVEBODY_KEEPBOUNDARY value in
 IMimeMessage_SetOption

---
 dlls/inetcomm/mimeole.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/inetcomm/mimeole.c b/dlls/inetcomm/mimeole.c
index 45df88b..bcfa2db 100644
--- a/dlls/inetcomm/mimeole.c
+++ b/dlls/inetcomm/mimeole.c
@@ -2338,6 +2338,9 @@ static HRESULT WINAPI MimeMessage_SetOption(
     case OID_SHOW_MACBINARY:
         FIXME("OID_SHOW_MACBINARY (value %d): ignoring\n", pValue->u.boolVal);
         break;
+    case OID_SAVEBODY_KEEPBOUNDARY:
+        FIXME("OID_SAVEBODY_KEEPBOUNDARY (value %d): ignoring\n", pValue->u.boolVal);
+        break;
     default:
         FIXME("Unhandled oid %08x\n", oid);
         hr = MIME_E_INVALID_OPTION_ID;
-- 
1.9.1

