From 5972d4e7c1f717c563be72b787b332ad94045f31 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 27 Apr 2016 11:50:40 +1000
Subject: [PATCH 6/8] inetcomm: return S_OK from IMimePropertySchema
 ModifyProperty

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/inetcomm/mimeole.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/inetcomm/mimeole.c b/dlls/inetcomm/mimeole.c
index bcfa2db..851d52c 100644
--- a/dlls/inetcomm/mimeole.c
+++ b/dlls/inetcomm/mimeole.c
@@ -3185,7 +3185,7 @@ static HRESULT WINAPI propschema_ModifyProperty(IMimePropertySchema *iface, cons
 {
     propschema *This = impl_from_IMimePropertySchema(iface);
     FIXME("(%p)->(%s, %x, %d, %d) stub\n", This, debugstr_a(name), flags, rownumber, vtdefault);
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI propschema_GetPropertyId(IMimePropertySchema *iface, const char *name, DWORD *propid)
-- 
1.9.1

