From 603977ec39d5909edaa684caf16383f6d51187c0 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 18 Apr 2016 16:03:00 +1000
Subject: [PATCH 7/7] mimeole.idl: Add more OID values

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/mimeole.idl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/mimeole.idl b/include/mimeole.idl
index dddfc26..a0edbc0 100644
--- a/include/mimeole.idl
+++ b/include/mimeole.idl
@@ -143,8 +143,9 @@ cpp_quote("#define OID_HIDE_TNEF_ATTACHMENTS     (0x000e0000 | VT_BOOL)")
 cpp_quote("#define OID_SECURITY_TYPE             (0x00180000 | VT_UI4)")
 cpp_quote("#define OID_SECURITY_HWND_OWNER       (0x00320000 | VT_UI4)")
 cpp_quote("#define OID_HEADER_RELOAD_TYPE        (0x00370000 | VT_UI4)")
+cpp_quote("#define OID_CAN_INLINE_TEXT_BODIES    (0x00380000 | VT_BOOL)")
 cpp_quote("#define OID_SHOW_MACBINARY            (0x00390000 | VT_BOOL)")
-
+cpp_quote("#define OID_SAVEBODY_KEEPBOUNDARY     (0x00400000 | VT_BOOL)")
 
 [
     uuid(e4b28371-83b0-11d0-8259-00c04fd85ab4),
-- 
1.9.1

