From 2015e1b3caa23aa524064f99ab472cba664347ee Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 10 Oct 2020 15:02:32 +1100
Subject: msado15: Add Open Test


diff --git a/dlls/msado15/tests/msado15.c b/dlls/msado15/tests/msado15.c
index cafb73d64e4..6b4c5222923 100644
--- a/dlls/msado15/tests/msado15.c
+++ b/dlls/msado15/tests/msado15.c
@@ -1477,6 +1477,95 @@ static void test_ConnectionPoint(void)
     ok( !conn_event.refs, "got %d\n", conn_event.refs );
 }
 
+static void test_connection_open(void)
+{
+    HRESULT hr;
+    _Connection *connection;
+    _Recordset  *recordset, *recordset2 = NULL;
+    BSTR str, str2;
+    VARIANT var, var2;
+    VARIANT affected;
+
+    hr = CoCreateInstance(&CLSID_Connection, NULL, CLSCTX_INPROC_SERVER, &IID__Connection, (void**)&connection);
+    ok( hr == S_OK, "got %08x\n", hr );
+
+    hr = CoCreateInstance(&CLSID_Recordset, NULL, CLSCTX_INPROC_SERVER, &IID__Recordset, (void**)&recordset);
+    ok( hr == S_OK, "got %08x\n", hr );
+
+    str = SysAllocString(L"Provider=MSDASQL.1;Persist Security Info=False;Data Source=wine_test");
+    str2 = SysAllocStringLen(NULL, 0);
+
+    hr = _Connection_Open(connection, str, str2, str2, 0);
+    ok( hr == S_OK, "got %08x\n", hr );
+    SysFreeString(str);
+    SysFreeString(str2);
+
+    V_VT(&var2) = VT_BSTR;
+    V_BSTR(&var2) = SysAllocString(L"SELECT * FROM testing");
+
+    //V_VT(&var) = VT_DISPATCH;
+    //V_DISPATCH(&var) = (IDispatch*)connection;
+    //_Connection_AddRef(connection);
+
+    /*ok( 0, "Before _Recordset_Open\n");
+    hr = _Recordset_Open(recordset, var2, var, adOpenDynamic, adLockPessimistic, adCmdText);
+    ok( hr == S_OK, "got %08x\n", hr );
+    ok( 0, "After _Recordset_Open\n");*/
+
+    VariantInit(&affected);
+    ok( 0, "**** Before _Connection_Execute\n");
+    hr = _Connection_Execute(connection, V_BSTR(&var2), &affected, 0, &recordset2);
+    ok( 0, "**** After _Connection_Execute\n");
+    ok( hr == S_OK, "got %08x\n", hr );
+
+    if (recordset2)
+    {
+        VARIANT_BOOL eof, bof;
+
+        ok( 0, "**** Before _Recordset_get_BOF\n");
+        hr = _Recordset_get_BOF(recordset2, &bof);
+        ok( hr == S_OK, "got %08x\n", hr );
+        ok( bof == VARIANT_FALSE, "Failed\n");
+        ok( 0, "**** After _Recordset_get_BOF\n");
+
+        ok( 0, "**** Before _Recordset_MoveFirst\n");
+        hr = _Recordset_MoveFirst(recordset2);
+        ok( hr == S_OK, "got %08x\n", hr );
+        ok( 0, "**** After _Recordset_MoveFirst\n");
+
+        ok( 0, "**** Before _Recordset_MoveLast\n");
+        hr = _Recordset_MoveLast(recordset2);
+        ok( hr == S_OK, "got %08x\n", hr );
+        ok( 0, "**** After _Recordset_MoveLast\n");
+
+        ok( 0, "**** Before _Recordset_MoveNext\n");
+        hr = _Recordset_MoveNext(recordset2);
+        ok( hr == S_OK, "got %08x\n", hr );
+        ok( 0, "**** After _Recordset_MoveNext\n");
+
+        ok( 0, "**** Before _Recordset_get_EOF\n");
+        hr = _Recordset_get_EOF(recordset2, &eof);
+        ok( hr == S_OK, "got %08x\n", hr );
+        ok( eof == VARIANT_FALSE, "Failed\n");
+        ok( 0, "**** After _Recordset_get_EOF\n");
+
+        ok(0, "Affected %d\n", V_VT(&affected));
+        hr = _Recordset_Close(recordset2);
+        ok( hr == S_OK, "got %08x\n", hr );
+
+        _Recordset_Release(recordset2);
+    }
+
+    VariantClear(&var);
+    VariantClear(&var2);
+    SysFreeString(str);
+
+     //hr = _Recordset_Close(recordset);
+     //ok( hr == S_OK, "got %08x\n", hr );
+
+    _Connection_Release(connection);
+}
+
 START_TEST(msado15)
 {
     CoInitialize( NULL );
@@ -1487,5 +1576,6 @@ START_TEST(msado15)
     test_Recordset();
     test_Stream();
     test_Command();
+    test_connection_open();
     CoUninitialize();
 }
