From d1df068b7c0857d2018d1accc05b026df8cbd885 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 7 Oct 2016 09:28:14 +1100
Subject: [PATCH] msado15/tests: Add initial

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/msado15/tests/msado15.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/dlls/msado15/tests/msado15.c b/dlls/msado15/tests/msado15.c
index ed465e2072..976bdcd73f 100644
--- a/dlls/msado15/tests/msado15.c
+++ b/dlls/msado15/tests/msado15.c
@@ -55,9 +55,30 @@ static void test_Stream(void)
     ok( !refs, "got %d\n", refs );
 }
 
+static void test_connection(void)
+{
+    HRESULT hr;
+    _Connection *connection;
+    IRunnableObject *runtime;
+    ISupportErrorInfo *errorinfo;
+
+    hr = CoCreateInstance(&CLSID_Connection, NULL, CLSCTX_INPROC_SERVER, &IID__Connection, (void**)&connection);
+    ok( hr == S_OK, "got %08x\n", hr );
+
+    hr = _Connection_QueryInterface(connection, &IID_IRunnableObject, (void**)&runtime);
+    ok(hr == E_NOINTERFACE, "Unexpected IRunnableObject interface\n");
+
+    hr = _Connection_QueryInterface(connection, &IID_ISupportErrorInfo, (void**)&errorinfo);
+    todo_wine ok(hr == S_OK, "Failed to get ISupportErrorInfo interface\n");
+    if (hr == S_OK)
+        ISupportErrorInfo_Release(errorinfo);
+}
+
 START_TEST(msado15)
 {
     CoInitialize( NULL );
     test_Stream();
+    test_connection();
+
     CoUninitialize();
 }
-- 
2.17.1

