From 457045e387e89b5bc5f9eea5b0ed70c194561055 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 22 Dec 2020 13:25:23 +1100
Subject: [PATCH] msado15: Only support put CursorLocation on Closed Recordset

---
 dlls/msado15/recordset.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/msado15/recordset.c b/dlls/msado15/recordset.c
index 7ccab544dec..e7837a0e5b2 100644
--- a/dlls/msado15/recordset.c
+++ b/dlls/msado15/recordset.c
@@ -1336,6 +1336,8 @@ static HRESULT WINAPI recordset_put_CursorLocation( _Recordset *iface, CursorLoc
 
     TRACE( "%p, %u\n", iface, cursor_loc );
 
+    if (recordset->state == adStateOpen) return MAKE_ADO_HRESULT( adErrObjectOpen );
+
     recordset->cursor_location = cursor_loc;
 
     return S_OK;
-- 
2.29.2

