From 7d82ed197009964abd434dc659123f6ccd925596 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 23 Dec 2020 16:19:13 +1100
Subject: [PATCH] msado15: Implement _Recordset Open

---
 dlls/msado15/recordset.c | 68 ++++++++++++++++++++++++++++++++++++++--
 1 file changed, 65 insertions(+), 3 deletions(-)

diff --git a/dlls/msado15/recordset.c b/dlls/msado15/recordset.c
index e7837a0e5b2..2d1f15564fe 100644
--- a/dlls/msado15/recordset.c
+++ b/dlls/msado15/recordset.c
@@ -1196,15 +1196,77 @@ static HRESULT WINAPI recordset_Open( _Recordset *iface, VARIANT source, VARIANT
                                       CursorTypeEnum cursor_type, LockTypeEnum lock_type, LONG options )
 {
     struct recordset *recordset = impl_from_Recordset( iface );
+    ADOConnectionConstruction15 *construct;
+    IUnknown *session = NULL;
+    IOpenRowset *openrowset = NULL;
+    IDBCreateCommand *create_command = NULL;
+    ICommand *cmd = NULL;
+    ICommandText *comand_text = NULL;
+    DBROWCOUNT affected;
+    IUnknown *rowset = NULL;
+    HRESULT hr;
 
     FIXME( "%p, %s, %s, %d, %d, %d\n", recordset, debugstr_variant(&source), debugstr_variant(&active_connection),
            cursor_type, lock_type, options );
 
-    if (!recordset->fields) return MAKE_ADO_HRESULT( adErrInvalidConnection );
     if (recordset->state == adStateOpen) return MAKE_ADO_HRESULT( adErrObjectOpen );
 
-    recordset->state = adStateOpen;
-    return S_OK;
+    if (V_VT(&active_connection) != VT_DISPATCH)
+    {
+        FIXME("Unsupported Active connection type %d\n", V_VT(&active_connection));
+        return E_FAIL;
+    }
+
+
+    hr = IDispatch_QueryInterface(V_DISPATCH(&active_connection), &IID_ADOConnectionConstruction15, (void**)&construct);
+    if (FAILED(hr))
+        return E_FAIL;
+
+    hr = ADOConnectionConstruction15_get_Session(construct, &session);
+    ADOConnectionConstruction15_Release(construct);
+    if (FAILED(hr))
+        return E_FAIL;
+
+    /* Having to have fields doesn't make sense when the Recordset hasn't been Opened */
+    /*if (!recordset->fields) return MAKE_ADO_HRESULT( adErrInvalidConnection );*/
+
+    hr = IUnknown_QueryInterface(session, &IID_IOpenRowset, (void**)&openrowset);
+    if (FAILED(hr)) return hr;
+
+    hr = IOpenRowset_QueryInterface(openrowset, &IID_IDBCreateCommand, (void**)&create_command);
+    if (FAILED(hr))
+        goto done;
+
+    hr = IDBCreateCommand_CreateCommand(create_command, NULL, &IID_IUnknown, (IUnknown **)&cmd);
+    if (FAILED(hr))
+        goto done;
+
+    hr = ICommand_QueryInterface(cmd, &IID_ICommandText, (void**)&comand_text);
+    if (FAILED(hr))
+    {
+        FIXME("Currently only ICommandText interface is support\n");
+        goto done;
+    }
+
+    hr = ICommandText_SetCommandText(comand_text, &DBGUID_DEFAULT, V_BSTR(&source));
+    if (FAILED(hr))
+        goto done;
+
+    hr = ICommandText_Execute(comand_text, NULL, &IID_IUnknown, NULL, &affected, &rowset);
+    if (SUCCEEDED(hr))
+    {
+        ADORecordsetConstruction_put_Rowset(&recordset->ADORecordsetConstruction_iface, rowset);
+        recordset->state = adStateOpen;
+    }
+
+done:
+    if (session) IUnknown_Release(session);
+    if (rowset) IUnknown_Release(rowset);
+    if (comand_text) ICommandText_Release(comand_text);
+    if (cmd) ICommand_Release(cmd);
+    if (create_command) IDBCreateCommand_Release(create_command);
+    if (openrowset) IOpenRowset_Release(openrowset);
+    return hr;
 }
 
 static HRESULT WINAPI recordset_Requery( _Recordset *iface, LONG options )
-- 
2.29.2

