From a003d70650ae0b90cc372e18208e1ed34afe776e Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 17 Feb 2023 15:50:19 +1100
Subject: [PATCH] msado15: crude _Recordset Find

---
 dlls/msado15/recordset.c | 44 +++++++++++++++++++++++++++++++++++++++-
 1 file changed, 43 insertions(+), 1 deletion(-)

diff --git a/dlls/msado15/recordset.c b/dlls/msado15/recordset.c
index ca30f358b14..8f065b8ab56 100644
--- a/dlls/msado15/recordset.c
+++ b/dlls/msado15/recordset.c
@@ -1980,13 +1980,55 @@ static HRESULT WINAPI recordset_Find( _Recordset *iface, BSTR criteria, LONG ski
                                       SearchDirectionEnum search_direction, VARIANT start )
 {
     struct recordset *recordset = impl_from_Recordset( iface );
+    HRESULT hr;
+    WCHAR column[256];
+    WCHAR value[256];
+    WCHAR *p;
+
     FIXME( "%p, %s, %ld, %d, %s\n", recordset, debugstr_w(criteria), skip_records, search_direction,
            debugstr_variant(&start) );
 
-    if (search_direction == adSearchForward)
+    p = wcsstr(criteria, L"=");
+    if (p)
+    {
+        VARIANT index, rhs;
+        ULONG datacol = 0;
+        ULONG colcnt = get_column_count(recordset);
+
+        lstrcpynW(column, criteria, p - criteria + 1);
+        FIXME("Column %s\n", debugstr_w(column));
+
+        V_VT(&index) = VT_BSTR;
+        V_BSTR(&index) = column;
+
+        if ((hr = map_index( recordset->fields, &index, &datacol )) != S_OK)
+        {
+            FIXME("No Column name");
+            return hr;
+        }
+        FIXME("Column index %d\n", datacol);
+
+        lstrcpyW(value, p+2);
+        value[lstrlenW(value)-1] = '\0';
+        FIXME("value %s\n", debugstr_w(value));
+
+        V_VT(&rhs) = VT_BSTR;
+        V_BSTR(&rhs) = value;
+
+        recordset->index += skip_records;
+        for(; recordset->index < recordset->count; recordset->index++)
+        {
+            FIXME("Data %s\n", debugstr_variant(&recordset->data[recordset->index * colcnt + datacol]));
+            if (lstrcmpW(V_BSTR(&recordset->data[recordset->index * colcnt + datacol]), V_BSTR(&rhs)) == 0)
+                break;
+        }
+    }
+
+/*    if (search_direction == adSearchForward)
         recordset->index = recordset->count;
     else
         recordset->index = -1;
+*/
     return S_OK;
 }
 
-- 
2.39.1

