From 12ab735de0da237d735c2a86ad5556c2ea5c0cc5 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 7 Oct 2016 14:49:58 +1100
Subject: [PATCH 12/12] msado15/tests: Add initial _Command test

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/msado15/tests/connection.c | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/dlls/msado15/tests/connection.c b/dlls/msado15/tests/connection.c
index eca79e0..6bf767f 100644
--- a/dlls/msado15/tests/connection.c
+++ b/dlls/msado15/tests/connection.c
@@ -122,12 +122,47 @@ static void test_recordset_interface(void)
     _Recordset_Release(recordset);
 }
 
+static void test_command_interface(void)
+{
+    HRESULT hr;
+    _Command *command;
+    _ADO *ado;
+    Command15 *command15;
+    Command25 *command25;
+
+    hr = CoCreateInstance(&CLSID_Command, NULL, CLSCTX_INPROC_SERVER,
+            &IID__Command, (void**)&command);
+    if(FAILED(hr))
+    {
+        skip("Failed to create Command instance\n");
+        return;
+    }
+
+    hr = _Command_QueryInterface(command, &IID__ADO, (void**)&ado);
+    ok(hr == S_OK, "Failed to get _ADO interface\n");
+    if (hr == S_OK)
+        _ADO_Release(ado);
+
+    hr = _Command_QueryInterface(command, &IID_Command15, (void**)&command15);
+    ok(hr == S_OK, "Failed to get Recordset15 interface\n");
+    if (hr == S_OK)
+        Command15_Release(command15);
+
+    hr = _Command_QueryInterface(command, &IID_Command25, (void**)&command25);
+    ok(hr == S_OK, "Failed to get Recordset20 interface\n");
+    if (hr == S_OK)
+        Command25_Release(command25);
+
+    _Command_Release(command);
+}
+
 START_TEST(connection)
 {
     CoInitialize(NULL);
 
     test_connection_interface();
     test_recordset_interface();
+    test_command_interface();
 
     CoUninitialize();
 }
-- 
1.9.1

