From 6f9ca9058009d42dfc8b3a00c5519ec734b3ac25 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 20 Feb 2023 13:21:16 +1100
Subject: [PATCH] msado15: Field ActualSize return a default size

---
 dlls/msado15/recordset.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dlls/msado15/recordset.c b/dlls/msado15/recordset.c
index 64a0bc8b139..6fc96d3ee46 100644
--- a/dlls/msado15/recordset.c
+++ b/dlls/msado15/recordset.c
@@ -199,8 +199,10 @@ static HRESULT WINAPI field_get_Properties( Field *iface, Properties **obj )
 
 static HRESULT WINAPI field_get_ActualSize( Field *iface, ADO_LONGPTR *size )
 {
-    FIXME( "%p, %p\n", iface, size );
-    return E_NOTIMPL;
+    struct field *field = impl_from_Field( iface );
+    FIXME( "%p, %p\n", field, size );
+    *size = 0;
+    return S_OK;
 }
 
 static HRESULT WINAPI field_get_Attributes( Field *iface, LONG *attrs )
-- 
2.39.2

