From cebf769dbf6e1ed3d5abcdcb573b37d0e099acf7 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 23 Nov 2020 13:24:19 +1100
Subject: [PATCH] msado15: _Recordset get_EOF - Add rowset support

---
 dlls/msado15/recordset.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/dlls/msado15/recordset.c b/dlls/msado15/recordset.c
index cfd92bc31ce..19c816e7a23 100644
--- a/dlls/msado15/recordset.c
+++ b/dlls/msado15/recordset.c
@@ -24,6 +24,7 @@
 #include "objbase.h"
 #include "msado15_backcompat.h"
 #include "oledb.h"
+#include "oledberr.h"
 
 #include "wine/debug.h"
 #include "wine/heap.h"
@@ -994,7 +995,21 @@ static HRESULT WINAPI recordset_get_EOF( _Recordset *iface, VARIANT_BOOL *eof )
 
     TRACE( "%p, %p\n", recordset, eof );
 
-    *eof = (!recordset->count || recordset->index >= recordset->count) ? VARIANT_TRUE : VARIANT_FALSE;
+    if (recordset->row_set)
+    {
+        DBCOUNTITEM obtained = 0;
+        HROW *rows = NULL;
+        HRESULT hr;
+
+        hr = IRowset_GetNextRows(recordset->row_set, DB_NULL_HCHAPTER, 0, 1, &obtained, &rows);
+
+        *eof = hr == DB_S_ENDOFROWSET ? VARIANT_TRUE : VARIANT_FALSE;
+
+        if (hr == S_OK)
+            IRowset_ReleaseRows(recordset->row_set, 1, rows, NULL, NULL, NULL);
+    }
+    else
+        *eof = (!recordset->count || recordset->index >= recordset->count) ? VARIANT_TRUE : VARIANT_FALSE;
     return S_OK;
 }
 
-- 
2.29.2

