From 2e3feb106dafe5eb0135a44899cd2798547965f7 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 15 Jun 2020 11:09:43 +1000
Subject: [PATCH 5/7] winhttp: Implement WinHttpWebSocketCompleteUpgrade

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/winhttp/request.c | 31 ++++++++++++++++++++++++++++++-
 1 file changed, 30 insertions(+), 1 deletion(-)

diff --git a/dlls/winhttp/request.c b/dlls/winhttp/request.c
index a018b783069..f0553dd6fa1 100644
--- a/dlls/winhttp/request.c
+++ b/dlls/winhttp/request.c
@@ -3054,8 +3054,37 @@ BOOL WINAPI WinHttpWriteData( HINTERNET hrequest, LPCVOID buffer, DWORD to_write
 
 HINTERNET WINAPI WinHttpWebSocketCompleteUpgrade( HINTERNET hrequest, DWORD_PTR context )
 {
+    struct request *request;
+    DWORD size, query, status = 0;
+    HINTERNET hconnect = NULL;
+
     FIXME("%p, %08lx\n", hrequest, context);
-    return NULL;
+
+    if (!(request = (struct request *)grab_object( hrequest )))
+    {
+        SetLastError( ERROR_INVALID_HANDLE );
+        return FALSE;
+    }
+
+    if (!request->upgrade_socket)
+    {
+        release_object( &request->hdr );
+        SetLastError( ERROR_WINHTTP_INCORRECT_HANDLE_STATE );
+        return FALSE;
+    }
+
+    size = sizeof(DWORD);
+    query = WINHTTP_QUERY_STATUS_CODE | WINHTTP_QUERY_FLAG_NUMBER;
+    if (query_headers( request, query, NULL, &status, &size, NULL ) || status != HTTP_STATUS_SWITCH_PROTOCOLS)
+    {
+        release_object( &request->hdr );
+        SetLastError( ERROR_INVALID_OPERATION );
+        return FALSE;
+    }
+
+    hconnect = alloc_handle( &request->hdr );
+
+    return hconnect;
 }
 
 DWORD WINAPI WinHttpWebSocketSend( HINTERNET hsocket, WINHTTP_WEB_SOCKET_BUFFER_TYPE type, void *buf, DWORD len )
-- 
2.27.0

