From 529aa4dbe987d956cfdc340e712cfcb228a4c3b3 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 22 Jun 2020 15:51:23 +1000
Subject: [PATCH 6/6] winhttp: Convert winsock error to internet error

---
 dlls/winhttp/request.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/dlls/winhttp/request.c b/dlls/winhttp/request.c
index 83b478c5e00..92b1bf2079c 100644
--- a/dlls/winhttp/request.c
+++ b/dlls/winhttp/request.c
@@ -122,6 +122,19 @@ static const WCHAR *attribute_table[] =
     NULL                            /* WINHTTP_QUERY_PASSPORT_CONFIG            = 78 */
 };
 
+static DWORD wsa_to_websocket_error(DWORD error)
+{
+    switch(error)
+    {
+        case WSAETIMEDOUT:
+            return ERROR_WINHTTP_TIMEOUT;
+        case WSAECONNRESET:
+            return 12030; /* ERROR_INTERNET_CONNECTION_ABORTED */
+        default:
+            return error;
+    }
+}
+
 static struct task_header *dequeue_task( struct request *request )
 {
     struct task_header *task;
@@ -2838,7 +2851,7 @@ BOOL WINAPI WinHttpReceiveResponse( HINTERNET hrequest, LPVOID reserved )
         ret = receive_response( request, FALSE );
 
     release_object( &request->hdr );
-    SetLastError( ret );
+    SetLastError( wsa_to_websocket_error(ret) );
     return !ret;
 }
 
@@ -3279,7 +3292,7 @@ retry:
     {
         WARN("header recv returned %d\n", ret);
         release_object( &sock->hdr );
-        return ret;
+        return wsa_to_websocket_error(ret);
     }
 
     ptr = buffer;
-- 
2.27.0

