From 2d7b71242621b975571865ff618bf5c4a3309df1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 29 Jun 2017 09:43:38 +1000
Subject: [PATCH 02/10] propsys/tests: Use standard wine_dbgstr_longlong

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/propsys/tests/propsys.c | 26 ++++++++------------------
 1 file changed, 8 insertions(+), 18 deletions(-)

diff --git a/dlls/propsys/tests/propsys.c b/dlls/propsys/tests/propsys.c
index 54b45ac..776ccd0 100644
--- a/dlls/propsys/tests/propsys.c
+++ b/dlls/propsys/tests/propsys.c
@@ -763,16 +763,6 @@ static void test_PropVariantCompare(void)
     SysFreeString(str_b.u.bstrVal);
 }
 
-static inline const char* debugstr_longlong(ULONGLONG ll)
-{
-    static char string[17];
-    if (sizeof(ll) > sizeof(unsigned long) && ll >> 32)
-        sprintf(string, "%lx%08lx", (unsigned long)(ll >> 32), (unsigned long)ll);
-    else
-        sprintf(string, "%lx", (unsigned long)ll);
-    return string;
-}
-
 static void test_intconversions(void)
 {
     PROPVARIANT propvar;
@@ -796,7 +786,7 @@ static void test_intconversions(void)
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == (ULONGLONG)1 << 63, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == (ULONGLONG)1 << 63, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 
     hr = PropVariantToUInt64(&propvar, &ullval);
     ok(hr == HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW), "hr=%x\n", hr);
@@ -818,11 +808,11 @@ static void test_intconversions(void)
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == 5, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == 5, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 
     hr = PropVariantToUInt64(&propvar, &ullval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(ullval == 5, "got wrong value %s\n", debugstr_longlong(ullval));
+    ok(ullval == 5, "got wrong value %s\n", wine_dbgstr_longlong(ullval));
 
     hr = PropVariantToInt32(&propvar, &lval);
     ok(hr == S_OK, "hr=%x\n", hr);
@@ -845,7 +835,7 @@ static void test_intconversions(void)
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == -5, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == -5, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 
     hr = PropVariantToUInt64(&propvar, &ullval);
     ok(hr == HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW), "hr=%x\n", hr);
@@ -869,28 +859,28 @@ static void test_intconversions(void)
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == 6, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == 6, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 
     propvar.vt = VT_I4;
     propvar.u.lVal = -6;
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == -6, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == -6, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 
     propvar.vt = VT_UI2;
     propvar.u.uiVal = 7;
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == 7, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == 7, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 
     propvar.vt = VT_I2;
     propvar.u.iVal = -7;
 
     hr = PropVariantToInt64(&propvar, &llval);
     ok(hr == S_OK, "hr=%x\n", hr);
-    ok(llval == -7, "got wrong value %s\n", debugstr_longlong(llval));
+    ok(llval == -7, "got wrong value %s\n", wine_dbgstr_longlong(llval));
 }
 
 static void test_PropVariantChangeType_LPWSTR(void)
-- 
1.9.1

