From 9b909a4a2a20f149f9955726e766b924258260b1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 29 Jun 2017 09:55:30 +1000
Subject: [PATCH 06/10] dmusic/tests: Use standard wine_dbgstr_longlong

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dmusic/tests/dmusic.c | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/dlls/dmusic/tests/dmusic.c b/dlls/dmusic/tests/dmusic.c
index 7da8aab..961e2cd 100644
--- a/dlls/dmusic/tests/dmusic.c
+++ b/dlls/dmusic/tests/dmusic.c
@@ -29,16 +29,6 @@
 #include "dmusici.h"
 #include "dmksctrl.h"
 
-static inline const char* debugstr_longlong(ULONGLONG ll)
-{
-    static char string[17];
-    if (sizeof(ll) > sizeof(unsigned long) && ll >> 32)
-        sprintf(string, "%lx%08lx", (unsigned long)(ll >> 32), (unsigned long)ll);
-    else
-        sprintf(string, "%lx", (unsigned long)ll);
-    return string;
-}
-
 static void test_dmusic(void)
 {
     IDirectMusic *dmusic = NULL;
@@ -334,7 +324,7 @@ static void test_dmbuffer(void)
         data += sizeof(DMUS_EVENTHEADER);
         ok(header->cbEvent == 3, "cbEvent is %u instead of 3\n", header->cbEvent);
         ok(header->dwChannelGroup == 0, "dwChannelGroup is %u instead of 0\n", header->dwChannelGroup);
-        ok(header->rtDelta == 0, "rtDelta is %s instead of 0\n", debugstr_longlong(header->rtDelta));
+        ok(header->rtDelta == 0, "rtDelta is %s instead of 0\n", wine_dbgstr_longlong(header->rtDelta));
         ok(header->dwFlags == DMUS_EVENT_STRUCTURED, "dwFlags is %x instead of %x\n", header->dwFlags, DMUS_EVENT_STRUCTURED);
         message = *(DWORD*)data & 0xffffff; /* Only 3 bytes are relevant */
         data += sizeof(DWORD);
@@ -345,7 +335,7 @@ static void test_dmbuffer(void)
         data += sizeof(DMUS_EVENTHEADER);
         ok(header->cbEvent == 3, "cbEvent is %u instead of 3\n", header->cbEvent);
         ok(header->dwChannelGroup == 0, "dwChannelGroup is %u instead of 0\n", header->dwChannelGroup);
-        ok(header->rtDelta == 10, "rtDelta is %s instead of 0\n", debugstr_longlong(header->rtDelta));
+        ok(header->rtDelta == 10, "rtDelta is %s instead of 0\n", wine_dbgstr_longlong(header->rtDelta));
         ok(header->dwFlags == DMUS_EVENT_STRUCTURED, "dwFlags is %x instead of %x\n", header->dwFlags, DMUS_EVENT_STRUCTURED);
         message = *(DWORD*)data & 0xffffff; /* Only 3 bytes are relevant */
         ok(message == 0x000080, "Message 2 is %0x instead of 0x000080\n", message);
-- 
1.9.1

