From 473e04bfeb28f1a96fd57e23907a1203b72fdefe Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 3 Jul 2017 08:30:47 +1000
Subject: [PATCH] apphelp/tests: Use standard wine_dbgstr_longlong

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/apphelp/tests/apphelp.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/dlls/apphelp/tests/apphelp.c b/dlls/apphelp/tests/apphelp.c
index 8b264e9..81104d2 100644
--- a/dlls/apphelp/tests/apphelp.c
+++ b/dlls/apphelp/tests/apphelp.c
@@ -64,14 +64,14 @@ static void test_ApphelpCheckShellObject(void)
         flags = 0xdeadbeef;
         SetLastError(0xdeadbeef);
         res = pApphelpCheckShellObject(objects[i], FALSE, &flags);
-        ok(res && (flags == 0), "%s 0: got %d and 0x%x%08x with 0x%x (expected TRUE and 0)\n",
-            wine_dbgstr_guid(objects[i]), res, (ULONG)(flags >> 32), (ULONG)flags, GetLastError());
+        ok(res && (flags == 0), "%s 0: got %d and %s with 0x%x (expected TRUE and 0)\n",
+            wine_dbgstr_guid(objects[i]), res, wine_dbgstr_longlong(flags), GetLastError());
 
         flags = 0xdeadbeef;
         SetLastError(0xdeadbeef);
         res = pApphelpCheckShellObject(objects[i], TRUE, &flags);
-        ok(res && (flags == 0), "%s 1: got %d and 0x%x%08x with 0x%x (expected TRUE and 0)\n",
-            wine_dbgstr_guid(objects[i]), res, (ULONG)(flags >> 32), (ULONG)flags, GetLastError());
+        ok(res && (flags == 0), "%s 1: got %d and %s with 0x%x (expected TRUE and 0)\n",
+            wine_dbgstr_guid(objects[i]), res, wine_dbgstr_longlong(flags), GetLastError());
 
     }
 
@@ -86,7 +86,7 @@ static void test_ApphelpCheckShellObject(void)
         flags = 0xdeadbeef;
         SetLastError(0xdeadbeef);
         res = pApphelpCheckShellObject(NULL, FALSE, &flags);
-        trace("NULL as CLSID*: got %d and 0x%x%08x with 0x%x\n", res, (ULONG)(flags >> 32), (ULONG)flags, GetLastError());
+        trace("NULL as CLSID*: got %d and %s with 0x%x\n", res, wine_dbgstr_longlong(flags), GetLastError());
     }
 }
 
-- 
1.9.1

