From fa62c6dd232ae068eb74f3edfbab91eced7ad6cd Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 16 Oct 2019 11:14:58 +1100
Subject: [PATCH] wmvcore: IWMReader GetOutputCount return 0 zero outputs.

---
 dlls/wmvcore/reader.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/wmvcore/reader.c b/dlls/wmvcore/reader.c
index 77d8ba6f41..ff725c17d9 100644
--- a/dlls/wmvcore/reader.c
+++ b/dlls/wmvcore/reader.c
@@ -205,7 +205,12 @@ static HRESULT WINAPI WMReader_GetOutputCount(IWMReader *iface, DWORD *outputs)
 {
     WMReader *This = impl_from_IWMReader(iface);
     FIXME("(%p)->(%p)\n", This, outputs);
-    return E_NOTIMPL;
+
+    if(!outputs)
+        return E_INVALIDARG;
+
+    *outputs = 0;
+    return S_OK;
 }
 
 static HRESULT WINAPI WMReader_GetOutputProps(IWMReader *iface, DWORD output_num, IWMOutputMediaProps **output)
-- 
2.17.1

