From 999cf37b26efaf8a39f1ab0db359c3eb4f877c1c Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 16 Oct 2019 11:15:44 +1100
Subject: [PATCH] wmvcore: IWMProfile3 GetStreamCount return 0 zero outputs.

---
 dlls/wmvcore/reader.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/wmvcore/reader.c b/dlls/wmvcore/reader.c
index ff725c17d9..0885f3b467 100644
--- a/dlls/wmvcore/reader.c
+++ b/dlls/wmvcore/reader.c
@@ -1785,7 +1785,12 @@ static HRESULT WINAPI profile3_GetStreamCount(IWMProfile3 *iface, DWORD *count)
 {
     WMReader *This = impl_from_IWMProfile3(iface);
     FIXME("%p, %p\n", This, count);
-    return E_NOTIMPL;
+
+    if (!count)
+        return E_INVALIDARG;
+
+    *count = 0;
+    return S_OK;
 }
 
 static HRESULT WINAPI profile3_GetStream(IWMProfile3 *iface, DWORD index, IWMStreamConfig **config)
-- 
2.17.1

