From fdd5cb61980780d50ee94f5e1cf09265f5129859 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 19 Aug 2016 10:22:35 +1000
Subject: [PATCH] comdlg32: Clear the filename text on OPEN_BROWSE

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/comdlg32/filedlg.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/comdlg32/filedlg.c b/dlls/comdlg32/filedlg.c
index 0fbb621..c4bced9 100644
--- a/dlls/comdlg32/filedlg.c
+++ b/dlls/comdlg32/filedlg.c
@@ -2541,6 +2541,7 @@ BOOL FILEDLG95_OnOpen(HWND hwnd)
                 && fodInfos->ofnInfos->Flags & OFN_EXPLORER)
             {
               SendCustomDlgNotificationMessage(hwnd, CDN_FOLDERCHANGE);
+              SendMessageA(fodInfos->DlgInfos.hwndFileName, WM_SETTEXT, 0, (LPARAM)"");
             }
 	  }
 	  else if( nOpenAction == ONOPEN_SEARCH )
-- 
1.9.1

