From 5a7248e371570e525f56da5e6fd44d9240a775c6 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 13 May 2022 10:44:49 +1000
Subject: [PATCH 01/17] d3drm: Correct D3DRMIMAGE validation

---
 dlls/d3drm/texture.c | 27 ++++++++++++++++++++-------
 1 file changed, 20 insertions(+), 7 deletions(-)

diff --git a/dlls/d3drm/texture.c b/dlls/d3drm/texture.c
index c679100c6b7..07e9c4ef066 100644
--- a/dlls/d3drm/texture.c
+++ b/dlls/d3drm/texture.c
@@ -52,14 +52,27 @@ static void d3drm_texture_destroy(struct d3drm_texture *texture)
 
 static BOOL d3drm_validate_image(D3DRMIMAGE *image)
 {
-    if (!image
-            || !image->red_mask
-            || !image->green_mask
-            || !image->blue_mask
-            || !image->buffer1
-            || !(image->rgb || (image->palette && image->palette_size)))
-    {
+    if (!image)
         return FALSE;
+
+    TRACE("size (%d, %d), aspect (%d, %d), depth %d, red %ld, green %ld, blue %ld, "
+          "buffer1 %p, buffer2 %p, rgb %d, pal %p, size %d\n",
+          image->width, image->height, image->aspectx, image->aspecty,
+          image->depth, image->red_mask, image->green_mask, image->blue_mask, image->buffer1,
+          image->buffer2, image->rgb, image->palette, image->palette_size );
+
+    if (!image->buffer1)
+        return FALSE;
+
+    if (image->rgb)
+    {
+        if (!image->red_mask  || !image->green_mask || !image->blue_mask)
+            return FALSE;
+    }
+    else
+    {
+        if (!image->palette || !image->palette_size)
+            return FALSE;
     }
 
     return TRUE;
-- 
2.35.1

