From 796b38d4ccd2c88a2a59f99fa6f8834ac24bc272 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 13 May 2022 13:39:12 +1000
Subject: [PATCH 10/17] d3drm: IDirect3DRMTexture3 SetDecalTransparency

---
 dlls/d3drm/texture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/d3drm/texture.c b/dlls/d3drm/texture.c
index a0a58e08dce..2ee9a1b8567 100644
--- a/dlls/d3drm/texture.c
+++ b/dlls/d3drm/texture.c
@@ -1223,7 +1223,7 @@ static HRESULT WINAPI d3drm_texture3_SetDecalTransparency(IDirect3DRMTexture3 *i
 {
     FIXME("iface %p, transparency %#x stub!\n", iface, transparency);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI d3drm_texture3_SetDecalTransparentColor(IDirect3DRMTexture3 *iface, D3DCOLOR color)
-- 
2.35.1

