From ca13260b9498991535cb5544cd75716055ef3c8d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 13 Jun 2022 17:53:37 +1000
Subject: [PATCH] d3drm: IDirect3DRMFace2 AddVertexAndNormalIndexed

---
 dlls/d3drm/face.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/dlls/d3drm/face.c b/dlls/d3drm/face.c
index 34810894e13..ce75c68bb59 100644
--- a/dlls/d3drm/face.c
+++ b/dlls/d3drm/face.c
@@ -206,9 +206,11 @@ static HRESULT WINAPI d3drm_face1_AddVertex(IDirect3DRMFace *iface, D3DVALUE x,
 static HRESULT WINAPI d3drm_face1_AddVertexAndNormalIndexed(IDirect3DRMFace *iface,
         DWORD vertex, DWORD normal)
 {
-    FIXME("iface %p, vertex %lu, normal %lu stub!\n", iface, vertex, normal);
+    struct d3drm_face *face = impl_from_IDirect3DRMFace(iface);
 
-    return E_NOTIMPL;
+    TRACE("iface %p, vertex %lu, normal %lu stub!\n", iface, vertex, normal);
+
+    return IDirect3DRMFace2_AddVertexAndNormalIndexed(&face->IDirect3DRMFace2_iface, vertex, normal);
 }
 
 static HRESULT WINAPI d3drm_face2_SetColorRGB(IDirect3DRMFace2 *iface, D3DVALUE red, D3DVALUE green, D3DVALUE blue)
@@ -477,7 +479,7 @@ static HRESULT WINAPI d3drm_face2_AddVertexAndNormalIndexed(IDirect3DRMFace2 *if
 {
     FIXME("iface %p, vertex %lu, normal %lu stub!\n", iface, vertex, normal);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI d3drm_face2_SetTexture(IDirect3DRMFace2 *iface, IDirect3DRMTexture3 *texture)
-- 
2.45.2

