From abb220c5767bda4eb0dfe0fb7a76c4488d541c77 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 13 May 2022 15:14:54 +1000
Subject: [PATCH 16/17] d3drm: IDirect3DRMFrame2 GetPosition

---
 dlls/d3drm/frame.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/d3drm/frame.c b/dlls/d3drm/frame.c
index 48af3b355ca..931250803cb 100644
--- a/dlls/d3drm/frame.c
+++ b/dlls/d3drm/frame.c
@@ -1465,7 +1465,7 @@ static HRESULT WINAPI d3drm_frame2_GetPosition(IDirect3DRMFrame2 *iface,
 {
     FIXME("iface %p, reference %p, position %p stub!\n", iface, reference, position);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI d3drm_frame1_GetPosition(IDirect3DRMFrame *iface,
-- 
2.35.1

