From 10867795301b6c67bb0340530f0144b7ffb22d91 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 1 Jun 2022 18:46:13 +1000
Subject: [PATCH] d3drm: IDirect3DRMMeshBuilder3 Load support loading from
 resource

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/d3drm/meshbuilder.c | 25 +++++++++----------------
 1 file changed, 9 insertions(+), 16 deletions(-)

diff --git a/dlls/d3drm/meshbuilder.c b/dlls/d3drm/meshbuilder.c
index 8177d2a2dfd..eda879e0c1a 100644
--- a/dlls/d3drm/meshbuilder.c
+++ b/dlls/d3drm/meshbuilder.c
@@ -1545,8 +1545,10 @@ end:
 static HRESULT WINAPI d3drm_mesh_builder3_Load(IDirect3DRMMeshBuilder3 *iface, void *filename,
         void *name, D3DRMLOADOPTIONS loadflags, D3DRMLOADTEXTURE3CALLBACK cb, void *arg)
 {
+    const DWORD supported_flags = D3DRMLOAD_FROMFILE | D3DRMLOAD_FROMRESOURCE | D3DRMLOAD_FROMMEMORY
+                | D3DRMLOAD_FROMSTREAM | D3DRMLOAD_FROMURL;
     struct d3drm_mesh_builder *mesh_builder = impl_from_IDirect3DRMMeshBuilder3(iface);
-    DXFILELOADOPTIONS load_options;
+    DXFILELOADOPTIONS load_options = loadflags & supported_flags;
     IDirectXFile *dxfile = NULL;
     IDirectXFileEnumObject *enum_object = NULL;
     IDirectXFileData *data = NULL;
@@ -1559,22 +1561,10 @@ static HRESULT WINAPI d3drm_mesh_builder3_Load(IDirect3DRMMeshBuilder3 *iface, v
     TRACE("iface %p, filename %p, name %p, loadflags %#lx, cb %p, arg %p.\n",
             iface, filename, name, loadflags, cb, arg);
 
-    clean_mesh_builder_data(mesh_builder);
+    if (loadflags & ~supported_flags)
+        FIXME("Unsupported flags %#lx\n", loadflags & ~supported_flags);
 
-    if (loadflags == D3DRMLOAD_FROMMEMORY)
-    {
-        load_options = DXFILELOAD_FROMMEMORY;
-    }
-    else if (loadflags == D3DRMLOAD_FROMFILE)
-    {
-        load_options = DXFILELOAD_FROMFILE;
-        TRACE("Loading from file %s\n", debugstr_a(filename));
-    }
-    else
-    {
-        FIXME("Load options %ld not supported yet\n", loadflags);
-        return E_NOTIMPL;
-    }
+    clean_mesh_builder_data(mesh_builder);
 
     hr = DirectXFileCreate(&dxfile);
     if (hr != DXFILE_OK)
@@ -1586,7 +1576,10 @@ static HRESULT WINAPI d3drm_mesh_builder3_Load(IDirect3DRMMeshBuilder3 *iface, v
 
     hr = IDirectXFile_CreateEnumObject(dxfile, filename, load_options, &enum_object);
     if (hr != DXFILE_OK)
+    {
+        WARN("Failed to create object for type %#lx\n", loadflags);
         goto end;
+    }
 
     hr = IDirectXFileEnumObject_GetNextDataObject(enum_object, &data);
     if (hr != DXFILE_OK)
-- 
2.35.1

