From fcfdd3b7696d221171373c33847f60315321c4a8 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 13 Jun 2022 17:57:01 +1000
Subject: [PATCH] d3drm: IDirect3DRMFace SetTexture

---
 dlls/d3drm/face.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/dlls/d3drm/face.c b/dlls/d3drm/face.c
index 9f0ef761464..ba41c05e298 100644
--- a/dlls/d3drm/face.c
+++ b/dlls/d3drm/face.c
@@ -256,9 +256,11 @@ static HRESULT WINAPI d3drm_face1_SetColor(IDirect3DRMFace *iface, D3DCOLOR colo
 
 static HRESULT WINAPI d3drm_face1_SetTexture(IDirect3DRMFace *iface, IDirect3DRMTexture *texture)
 {
-    FIXME("iface %p, texture %p stub!\n", iface, texture);
+    struct d3drm_face *face = impl_from_IDirect3DRMFace(iface);
 
-    return E_NOTIMPL;
+    TRACE("iface %p, texture %p\n", iface, texture);
+
+    return IDirect3DRMFace2_SetTexture(&face->IDirect3DRMFace2_iface, (IDirect3DRMTexture3 *)texture);
 }
 
 static HRESULT WINAPI d3drm_face1_SetTextureCoordinates(IDirect3DRMFace *iface,
@@ -486,7 +488,7 @@ static HRESULT WINAPI d3drm_face2_SetTexture(IDirect3DRMFace2 *iface, IDirect3DR
 {
     FIXME("iface %p, texture %p stub!\n", iface, texture);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI d3drm_face2_SetTextureCoordinates(IDirect3DRMFace2 *iface,
-- 
2.35.1

