From d7b6f24cbc9d76ee73b9284f26a651f1e87c59b5 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 18 Dec 2018 16:14:38 +1100
Subject: [PATCH] mfreadwrite: Create importlib

---
 dlls/mfreadwrite/Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/mfreadwrite/Makefile.in b/dlls/mfreadwrite/Makefile.in
index 7cf3a59..4b8ee82 100644
--- a/dlls/mfreadwrite/Makefile.in
+++ b/dlls/mfreadwrite/Makefile.in
@@ -1,4 +1,5 @@
 MODULE    = mfreadwrite.dll
+IMPORTLIB = mfreadwrite
 
 C_SRCS = \
 	main.c
-- 
1.9.1

