From edb535e28bff0d2a5f373385fc6166e7f1a21a6a Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 3 Jan 2019 13:40:01 +1100
Subject: [PATCH 2/4] mfreadwrite: Fake success in IMFSinkWriterEx AddStream

---
 dlls/mfreadwrite/writer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/mfreadwrite/writer.c b/dlls/mfreadwrite/writer.c
index bf90acdf821..d1df1830f77 100644
--- a/dlls/mfreadwrite/writer.c
+++ b/dlls/mfreadwrite/writer.c
@@ -292,7 +292,7 @@ static HRESULT WINAPI mfsinkwriter_AddStream(IMFSinkWriterEx *iface, IMFMediaTyp
 {
     struct mfsinkwriter *This = impl_from_IMFSinkWriterEx(iface);
     FIXME("%p, %p, %p\n", This, type, index);
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI mfsinkwriter_SetInputMediaType(IMFSinkWriterEx *iface, DWORD index, IMFMediaType *type, IMFAttributes *parameters)
-- 
2.33.0

