From 087765fa3b9c45b84233e3360bdd47ecbf7f052b Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 3 Jan 2019 13:40:14 +1100
Subject: [PATCH 3/4] mfreadwrite: Fake success in IMFSinkWriterEx BeginWriting

---
 dlls/mfreadwrite/writer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/mfreadwrite/writer.c b/dlls/mfreadwrite/writer.c
index d1df1830f77..c3425eb9ff2 100644
--- a/dlls/mfreadwrite/writer.c
+++ b/dlls/mfreadwrite/writer.c
@@ -306,7 +306,7 @@ static HRESULT WINAPI mfsinkwriter_BeginWriting(IMFSinkWriterEx *iface)
 {
     struct mfsinkwriter *This = impl_from_IMFSinkWriterEx(iface);
     FIXME("%p\n", This);
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI mfsinkwriter_WriteSample(IMFSinkWriterEx *iface, DWORD index, IMFSample *sample)
-- 
2.33.0

