From cc6abffd8332bc8244034df4b5244dafd03b95f1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 20 Dec 2018 10:41:20 +1100
Subject: [PATCH] mfreadwrite: Store IMFAttributes interface as part of the
 IMFSourceReader

---
 dlls/mfreadwrite/main.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/dlls/mfreadwrite/main.c b/dlls/mfreadwrite/main.c
index 7732f71..224fdd7 100644
--- a/dlls/mfreadwrite/main.c
+++ b/dlls/mfreadwrite/main.c
@@ -48,6 +48,7 @@ BOOL WINAPI DllMain(HINSTANCE instance, DWORD reason, LPVOID reserved)
 typedef struct _srcreader
 {
     IMFSourceReader IMFSourceReader_iface;
+    IMFAttributes *attributes;
     LONG ref;
 } srcreader;
 
@@ -97,6 +98,8 @@ static ULONG WINAPI src_reader_Release(IMFSourceReader *iface)
 
     if (!ref)
     {
+        if(This->attributes)
+            IMFAttributes_Release(This->attributes);
         HeapFree(GetProcessHeap(), 0, This);
     }
 
@@ -197,7 +200,7 @@ struct IMFSourceReaderVtbl srcreader_vtbl =
     src_reader_GetPresentationAttribute
 };
 
-static HRESULT create_source_reader(srcreader **ret)
+static HRESULT create_source_reader(IMFAttributes *attributes, srcreader **ret)
 {
     srcreader *object;
 
@@ -207,6 +210,9 @@ static HRESULT create_source_reader(srcreader **ret)
 
     object->ref = 1;
     object->IMFSourceReader_iface.lpVtbl = &srcreader_vtbl;
+    object->attributes = attributes;
+    if(attributes)
+        IMFAttributes_AddRef(attributes);
 
     *ret = object;
 
@@ -220,7 +226,7 @@ HRESULT WINAPI MFCreateSourceReaderFromByteStream(IMFByteStream *stream, IMFAttr
 
     TRACE("%p, %p, %p\n", stream, attributes, reader);
 
-    hr = create_source_reader(&object);
+    hr = create_source_reader(attributes, &object);
 
     *reader = &object->IMFSourceReader_iface;
     return hr;
@@ -234,7 +240,7 @@ HRESULT WINAPI MFCreateSourceReaderFromMediaSource(IMFMediaSource *source, IMFAt
 
     TRACE("%p, %p, %p\n", source, attributes, reader);
 
-    hr = create_source_reader(&object);
+    hr = create_source_reader(attributes, &object);
 
     *reader = &object->IMFSourceReader_iface;
     return hr;
-- 
1.9.1

