From 806abcd2dd5e6a1070d95472c74c9721c27ae37b Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 3 Jan 2019 12:13:55 +1100
Subject: [PATCH] mfplat: Implment IMFAttributes SetGUID

---
 dlls/mfplat/main.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/dlls/mfplat/main.c b/dlls/mfplat/main.c
index fc7c7a2..10976c2 100644
--- a/dlls/mfplat/main.c
+++ b/dlls/mfplat/main.c
@@ -823,10 +823,23 @@ static HRESULT WINAPI mfattributes_SetDouble(IMFAttributes *iface, REFGUID key,
 static HRESULT WINAPI mfattributes_SetGUID(IMFAttributes *iface, REFGUID key, REFGUID value)
 {
     mfattributes *This = impl_from_IMFAttributes(iface);
+    struct attribute *entry;
 
-    FIXME("%p, %s, %s\n", This, debugstr_guid(key), debugstr_guid(value));
+    TRACE("%p, %s, %s\n", This, debugstr_guid(key), debugstr_guid(value));
 
-    return E_NOTIMPL;
+    entry = find_attribute_entry(This, key);
+    if(entry)
+    {
+        memcpy(entry->var.puuid, value, sizeof(*(entry->var.puuid)));
+        return S_OK;
+    }
+
+    entry = create_attribute_entry(This, key);
+    if(!entry)
+        return E_OUTOFMEMORY;
+
+    InitPropVariantFromCLSID(value, &entry->var);
+    return S_OK;
 }
 
 static HRESULT WINAPI mfattributes_SetString(IMFAttributes *iface, REFGUID key, const WCHAR *value)
-- 
1.9.1

