From abfa1b54239b3897aef14f80df428a282ee84b37 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 23 Jun 2023 14:38:31 +1000
Subject: [PATCH] msscript.ocx: dump QuickActivate controls

---
 dlls/msscript.ocx/msscript.c | 40 ++++++++++++++++++++++++++++++++----
 1 file changed, 36 insertions(+), 4 deletions(-)

diff --git a/dlls/msscript.ocx/msscript.c b/dlls/msscript.ocx/msscript.c
index b5731a6e622..8803fb0d760 100644
--- a/dlls/msscript.ocx/msscript.c
+++ b/dlls/msscript.ocx/msscript.c
@@ -3381,7 +3381,7 @@ static HRESULT WINAPI PersistStreamInit_Load(IPersistStreamInit *iface, IStream
 
     FIXME("(%p)->(%p)\n", This, stream);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI PersistStreamInit_Save(IPersistStreamInit *iface, IStream *stream, BOOL clear_dirty)
@@ -3471,7 +3471,7 @@ static HRESULT WINAPI OleControl_OnAmbientPropertyChange(IOleControl *iface, DIS
 
     FIXME("(%p)->(%#lx)\n", This, dispid);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI OleControl_FreezeEvents(IOleControl *iface, BOOL freeze)
@@ -3480,7 +3480,7 @@ static HRESULT WINAPI OleControl_FreezeEvents(IOleControl *iface, BOOL freeze)
 
     FIXME("(%p)->(%d)\n", This, freeze);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static const IOleControlVtbl OleControlVtbl = {
@@ -3517,7 +3517,39 @@ static HRESULT WINAPI QuickActivate_QuickActivate(IQuickActivate *iface, QACONTA
 
     FIXME("(%p)->(%p %p)\n", This, container, control);
 
-    return E_NOTIMPL;
+    FIXME("%lu, site %p, adsink %p, propsink %p, unk %p, abient 0x%08lx, forecolor %lu, backcolor %lu,"
+          "font %p, undo %p, apper %lx, lcid %lu, pal %p, bind %p, control_site %p, prov %p\n",
+        container->cbSize, container->pClientSite, container->pAdviseSink,
+        container->pPropertyNotifySink,
+        container->pUnkEventSink,
+        container->dwAmbientFlags,
+        container->colorFore,
+        container->colorBack,
+        container->pFont,
+        container->pUndoMgr,
+        container->dwAppearance,
+        container->lcid,
+        container->hpal,
+        container->pBindHost,
+        container->pOleControlSite,
+        container->pServiceProvider
+        );
+
+    FIXME("%lu, misc %lx, view %lx, event %lx, prop %lx, policy %lx\n",
+        control->cbSize, control->dwMiscStatus,
+        control->dwViewStatus,
+        control->dwEventCookie,
+        control->dwPropNotifyCookie,
+        control->dwPointerActivationPolicy
+        );
+
+    IOleClientSite_AddRef(container->pClientSite);
+    control->dwMiscStatus = OLEMISC_SETCLIENTSITEFIRST|OLEMISC_ACTIVATEWHENVISIBLE|OLEMISC_INSIDEOUT
+        |OLEMISC_CANTLINKINSIDE|OLEMISC_RECOMPOSEONRESIZE;
+    control->dwViewStatus = 0x18;
+    control->dwPropNotifyCookie = 1;
+
+    return S_OK;
 }
 
 static HRESULT WINAPI QuickActivate_SetContentExtent(IQuickActivate *iface, SIZEL *size)
-- 
2.40.1

