From db356277079b16d3bb3fd43f6ff911b78910fbd1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 28 Jun 2023 12:29:36 +1000
Subject: [PATCH] msscript.ocx: Fake success in OnAmbientPropertyChange

---
 dlls/msscript.ocx/msscript.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/msscript.ocx/msscript.c b/dlls/msscript.ocx/msscript.c
index 368e344e1cf..8e70956b001 100644
--- a/dlls/msscript.ocx/msscript.c
+++ b/dlls/msscript.ocx/msscript.c
@@ -3471,7 +3471,7 @@ static HRESULT WINAPI OleControl_OnAmbientPropertyChange(IOleControl *iface, DIS
 
     FIXME("(%p)->(%#lx)\n", This, dispid);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static HRESULT WINAPI OleControl_FreezeEvents(IOleControl *iface, BOOL freeze)
-- 
2.40.1

