From c6c45b1cb1b93fafae8c00117999f01c3c7757a8 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 28 Jun 2023 12:32:29 +1000
Subject: [PATCH] msscript.ocx: Fake success in FreezeEvents

---
 dlls/msscript.ocx/msscript.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/msscript.ocx/msscript.c b/dlls/msscript.ocx/msscript.c
index 3e8fd0f7c72..8803fb0d760 100644
--- a/dlls/msscript.ocx/msscript.c
+++ b/dlls/msscript.ocx/msscript.c
@@ -3480,7 +3480,7 @@ static HRESULT WINAPI OleControl_FreezeEvents(IOleControl *iface, BOOL freeze)
 
     FIXME("(%p)->(%d)\n", This, freeze);
 
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 static const IOleControlVtbl OleControlVtbl = {
-- 
2.40.1

