From 2391868c0d40fd93f5ffdcc0bee6345d1a27fd1d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 3 Feb 2023 14:46:44 +1100
Subject: [PATCH 06/36] odbc32: SQLGetInfo/W support InfoType SQL_ODBC_VER

---
 dlls/odbc32/proxyodbc.c | 36 ++++++++++++++++++++++++++++++------
 1 file changed, 30 insertions(+), 6 deletions(-)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index 1fd2864f78b..6e00661a62e 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -658,12 +658,24 @@ SQLRETURN WINAPI SQLGetFunctions(SQLHDBC ConnectionHandle, SQLUSMALLINT Function
 SQLRETURN WINAPI SQLGetInfo(SQLHDBC ConnectionHandle, SQLUSMALLINT InfoType, SQLPOINTER InfoValue,
                             SQLSMALLINT BufferLength, SQLSMALLINT *StringLength)
 {
-    SQLRETURN ret = SQL_ERROR;
+    char *ptr = InfoValue;
 
-    FIXME("(ConnectionHandle, %p, InfoType %d, InfoValue %p, BufferLength %d, StringLength %p)\n", ConnectionHandle,
+    TRACE("(ConnectionHandle, %p, InfoType %d, InfoValue %p, BufferLength %d, StringLength %p)\n", ConnectionHandle,
           InfoType, InfoValue, BufferLength, StringLength);
 
-    return ret;
+    switch(InfoType)
+    {
+        case SQL_ODBC_VER:
+            lstrcpynA(ptr, "03.80.0000", BufferLength);
+            if (StringLength)
+                *StringLength = strlen(ptr);
+            break;
+        default:
+            FIXME("Unsupported type %d\n", InfoType);
+            return SQL_ERROR;
+    }
+
+    return SQL_SUCCESS;
 }
 
 /*************************************************************************
@@ -1597,12 +1609,24 @@ SQLRETURN WINAPI SQLGetConnectOptionW(SQLHDBC ConnectionHandle, SQLUSMALLINT Opt
 SQLRETURN WINAPI SQLGetInfoW(SQLHDBC ConnectionHandle, SQLUSMALLINT InfoType, SQLPOINTER InfoValue,
                              SQLSMALLINT BufferLength, SQLSMALLINT *StringLength)
 {
-    SQLRETURN ret = SQL_ERROR;
+    WCHAR *ptr = InfoValue;
 
-    FIXME("(ConnectionHandle, %p, InfoType %d, InfoValue %p, BufferLength %d, StringLength %p)\n", ConnectionHandle,
+    TRACE("(ConnectionHandle, %p, InfoType %d, InfoValue %p, BufferLength %d, StringLength %p)\n", ConnectionHandle,
           InfoType, InfoValue, BufferLength, StringLength);
 
-    return ret;
+    switch(InfoType)
+    {
+        case SQL_ODBC_VER:
+            lstrcpynW(ptr, L"03.80.0000", BufferLength);
+            if (StringLength)
+                *StringLength = wcslen(ptr);
+            break;
+        default:
+            FIXME("Unsupported type %d\n", InfoType);
+            return SQL_ERROR;
+    }
+
+    return SQL_SUCCESS;
 }
 
 /*************************************************************************
-- 
2.39.1

