From a6061eab04f7b9432acf1eb86eb47d3286d97f0e Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 3 Feb 2023 15:18:21 +1100
Subject: [PATCH 07/42] odbc32: Support SQL_ATTR_LOGIN_TIMEOUT in
 SQLSetConnectAttrW

---
 dlls/odbc32/proxyodbc.c | 28 +++++++++++++++++++++++++---
 1 file changed, 25 insertions(+), 3 deletions(-)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index c19f0248ef8..b4ec548b09c 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -53,6 +53,8 @@ struct SQLHDBC_data
 {
     int type;
     struct SQLHENV_data *environment;
+
+    SQLUINTEGER login_timeout;
 };
 
 /*************************************************************************
@@ -74,6 +76,7 @@ SQLRETURN WINAPI SQLAllocConnect(SQLHENV EnvironmentHandle, SQLHDBC *ConnectionH
 
     hdbc->type = SQL_HANDLE_DBC;
     hdbc->environment = EnvironmentHandle;
+    hdbc->login_timeout = 0;
 
     *ConnectionHandle = hdbc;
 
@@ -1548,12 +1551,31 @@ SQLRETURN WINAPI SQLGetStmtAttrW(SQLHSTMT StatementHandle, SQLINTEGER Attribute,
 SQLRETURN WINAPI SQLSetConnectAttrW(SQLHDBC ConnectionHandle, SQLINTEGER Attribute, SQLPOINTER Value,
                                     SQLINTEGER StringLength)
 {
-    SQLRETURN ret = SQL_ERROR;
+    struct SQLHDBC_data *hdbc = ConnectionHandle;
 
-    FIXME("(ConnectionHandle %p, Attribute %d, Value %p, StringLength %d)\n", ConnectionHandle, Attribute, Value,
+    TRACE("(ConnectionHandle %p, Attribute %d, Value %p, StringLength %d)\n", ConnectionHandle, Attribute, Value,
           StringLength);
 
-    return ret;
+    if (hdbc->type != SQL_HANDLE_DBC)
+    {
+        WARN("Wrong handle type %d\n", hdbc->type);
+        return SQL_ERROR;
+    }
+
+    switch(Attribute)
+    {
+        case SQL_ATTR_LOGIN_TIMEOUT:
+            if (Value)
+                hdbc->login_timeout = (intptr_t)Value;
+            else
+                hdbc->login_timeout = 0;
+            break;
+        default:
+            FIXME("Unhandle attribute %d\n", Attribute);
+            return SQL_ERROR;
+    }
+
+    return SQL_SUCCESS;
 }
 
 /*************************************************************************
-- 
2.39.1

