From fa035793940555276e5e2eb073d7efc3a8b2502e Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 6 Feb 2023 08:55:12 +1100
Subject: [PATCH 10/36] odbc32: Foward SQLSetConnectAttr requets onto the
 driver

---
 dlls/odbc32/proxyodbc.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index 8afe27da928..59efe6f74a6 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -1809,6 +1809,7 @@ SQLRETURN WINAPI SQLSetConnectAttrW(SQLHDBC ConnectionHandle, SQLINTEGER Attribu
                                     SQLINTEGER StringLength)
 {
     struct SQLHDBC_data *hdbc = ConnectionHandle;
+    SQLRETURN ret = SQL_SUCCESS;
 
     TRACE("(ConnectionHandle %p, Attribute %d, Value %p, StringLength %d)\n", ConnectionHandle, Attribute, Value,
           StringLength);
@@ -1828,11 +1829,18 @@ SQLRETURN WINAPI SQLSetConnectAttrW(SQLHDBC ConnectionHandle, SQLINTEGER Attribu
                 hdbc->login_timeout = 0;
             break;
         default:
-            FIXME("Unhandle attribute %d\n", Attribute);
-            return SQL_ERROR;
+            if (hdbc->pSQLSetConnectAttrW)
+                ret = hdbc->pSQLSetConnectAttrW(hdbc->driver_hdbc, Attribute, Value, StringLength);
+            else
+            {
+                FIXME("Unsupported Attribute %d\n", Attribute);
+                ret = SQL_ERROR;
+            }
     }
 
-    return SQL_SUCCESS;
+    TRACE("ret %d\n", ret);
+
+    return ret;
 }
 
 /*************************************************************************
-- 
2.39.1

