From 330f869873e589ca627a48450044d6194c9df5d4 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 6 Feb 2023 11:12:20 +1100
Subject: [PATCH 21/36] odbc32: Forward SQLMoreResults request onto driver

---
 dlls/odbc32/proxyodbc.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index b760dcf0a70..139832a681f 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -1511,10 +1511,23 @@ SQLRETURN WINAPI SQLForeignKeys(SQLHSTMT hstmt, SQLCHAR *szPkCatalogName, SQLSMA
  */
 SQLRETURN WINAPI SQLMoreResults(SQLHSTMT StatementHandle)
 {
+    struct SQLHSTMT_data *statement = StatementHandle;
     SQLRETURN ret = SQL_ERROR;
 
-    FIXME("(%p)\n", StatementHandle);
+    TRACE("(%p)\n", StatementHandle);
 
+    if (statement->type != SQL_HANDLE_STMT)
+    {
+        WARN("Wrong handle type %d\n", statement->type);
+        return SQL_ERROR;
+    }
+
+    if (statement->connection->pSQLMoreResults)
+    {
+        ret = statement->connection->pSQLMoreResults(statement->driver_stmt);
+    }
+
+    TRACE("ret %d\n", ret);
     return ret;
 }
 
-- 
2.39.1

