From 208f81f0423137a41da86ab5684455eb611d61f6 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 6 Feb 2023 13:14:19 +1100
Subject: [PATCH 29/36] odbc32: Forward SQLExecute request onto driver

---
 dlls/odbc32/proxyodbc.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index db21317def4..e932c23fc07 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -738,10 +738,23 @@ SQLRETURN WINAPI SQLExecute(SQLHSTMT StatementHandle)
  */
 SQLRETURN WINAPI SQLFetch(SQLHSTMT StatementHandle)
 {
+    struct SQLHSTMT_data *statement = StatementHandle;
     SQLRETURN ret = SQL_ERROR;
 
-    FIXME("(StatementHandle %p)\n", StatementHandle);
+    TRACE("(StatementHandle %p)\n", StatementHandle);
+
+    if (statement->type != SQL_HANDLE_STMT)
+    {
+        WARN("Wrong handle type %d\n", statement->type);
+        return SQL_ERROR;
+    }
+
+    if (statement->connection->pSQLFetch)
+    {
+        ret = statement->connection->pSQLFetch(statement->driver_stmt);
+    }
 
+    TRACE("ret %d\n", ret);
     return ret;
 }
 
-- 
2.39.1

