From 99696a933b6393903df0aa314190a220cea635ac Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 8 Feb 2023 08:39:26 +1100
Subject: [PATCH 38/42] odbc32: SQLGetEnvAttr: Support SQL_ATTR_ODBC_VERSION
 attribute

---
 dlls/odbc32/proxyodbc.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index bdd8361d221..46872add8b0 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -47,6 +47,7 @@ struct SQLHENV_data
 {
     int type;
     SQLUINTEGER pooling;
+    SQLUINTEGER version;
 };
 
 struct SQLHDBC_data
@@ -368,6 +369,7 @@ SQLRETURN WINAPI SQLAllocEnv(SQLHENV *EnvironmentHandle)
 
     henv->type = SQL_HANDLE_ENV;
     henv->pooling = SQL_CP_OFF;
+    henv->version = SQL_OV_ODBC2;
 
     *EnvironmentHandle = henv;
 
@@ -1067,6 +1069,14 @@ SQLRETURN WINAPI SQLGetEnvAttr(SQLHENV EnvironmentHandle, SQLINTEGER Attribute,
             }
             *(SQLUINTEGER*)Value = data->pooling;
             break;
+        case SQL_ATTR_ODBC_VERSION:
+            if (BufferLength != sizeof(data->version))
+            {
+                WARN("Invalid buffer size\n");
+                return SQL_ERROR;
+            }
+            *(SQLUINTEGER*)Value = data->version;
+            break;
         default:
             FIXME("Unhandle attribute %d\n", Attribute);
             return SQL_ERROR;
@@ -1426,6 +1436,12 @@ SQLRETURN WINAPI SQLSetEnvAttr(SQLHENV EnvironmentHandle, SQLINTEGER Attribute,
             else
                 data->pooling = SQL_CP_OFF;
             break;
+        case SQL_ATTR_ODBC_VERSION:
+            if (Value)
+                data->version = (uintptr_t)Value;
+            else
+                data->version = SQL_OV_ODBC2;
+            break;
         default:
             FIXME("Unhandle attribute %d\n", Attribute);
             return SQL_ERROR;
-- 
2.39.1

