From e73aae5f26b1af29ce14f3edaa84569bd3c86bc4 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 8 Feb 2023 14:16:24 +1100
Subject: [PATCH 40/42] odbc32: SQLBindCol convert to ODBC2 types if required

---
 dlls/odbc32/proxyodbc.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/dlls/odbc32/proxyodbc.c b/dlls/odbc32/proxyodbc.c
index 851d2535ca1..c6d19b58e1c 100644
--- a/dlls/odbc32/proxyodbc.c
+++ b/dlls/odbc32/proxyodbc.c
@@ -474,6 +474,20 @@ SQLRETURN WINAPI SQLBindCol(SQLHSTMT StatementHandle, SQLUSMALLINT ColumnNumber,
 
     if (statement->connection->pSQLBindCol)
     {
+        /*
+         * Map ODBC3 Datatype back to ODBC2 types when the application has asked for SQL_OV_ODBC2.
+         * Some drivers rely on this (PostgreSQL odbc driver).
+         */
+        if (statement->connection->environment->version == SQL_OV_ODBC2)
+        {
+            if(TargetType == SQL_C_TYPE_TIME)
+                TargetType = SQL_C_TIME;
+            else if(TargetType == SQL_C_TYPE_DATE)
+                TargetType = SQL_C_DATE;
+            else if(TargetType == SQL_C_TYPE_TIMESTAMP)
+                TargetType = SQL_C_TIMESTAMP;
+        }
+
         ret = statement->connection->pSQLBindCol(statement->driver_stmt, ColumnNumber, TargetType,
                             TargetValue, BufferLength, StrLen_or_Ind);
     }
-- 
2.39.1

