From bac8ab96f4e39c01ad8b1663e83315144a6332af Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 3 Mar 2017 16:00:07 +1100
Subject: [PATCH 2/2] dplayx/test: Free IDirectPlayLobby object

---
 dlls/dplayx/tests/dplayx.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/dlls/dplayx/tests/dplayx.c b/dlls/dplayx/tests/dplayx.c
index bb97199..2b4fda3 100644
--- a/dlls/dplayx/tests/dplayx.c
+++ b/dlls/dplayx/tests/dplayx.c
@@ -700,7 +700,7 @@ static void init_TCPIP_provider( IDirectPlay4 *pDP, LPCSTR strIPAddressString, W
     checkHR( DP_OK, hr );
 
     HeapFree( GetProcessHeap(), 0, pAddress );
-
+    IDirectPlayLobby_Release(pDPL);
 }
 
 static BOOL CALLBACK EnumSessions_cb_join( LPCDPSESSIONDESC2 lpThisSD,
@@ -940,6 +940,8 @@ static BOOL CALLBACK EnumConnections_cb( LPCGUID lpguidSP,
 
     callbackData->dwCounter1++;
 
+    IDirectPlayLobby_Release(pDPL);
+
     return TRUE;
 }
 
@@ -1170,8 +1172,8 @@ static void test_EnumAddressTypes(void)
     }
 
     IDirectPlayX_Close(pDP);
-
     IDirectPlayX_Release(pDP);
+    IDirectPlayLobby_Release(pDPL);
 
     HeapFree( GetProcessHeap(), 0, pAddress );
 }
@@ -3075,7 +3077,7 @@ static void test_GetPlayerAddress(void)
     if ( hr == DP_OK )
     {
         todo_wine win_skip( "GetPlayerAddress not implemented\n" );
-        return;
+        goto cleanup;
     }
 
     init_TCPIP_provider( pDP[0], "127.0.0.1", 0 );
@@ -3171,9 +3173,11 @@ static void test_GetPlayerAddress(void)
 
 
     HeapFree( GetProcessHeap(), 0, lpData );
+
+cleanup:
     IDirectPlayX_Release( pDP[0] );
     IDirectPlayX_Release( pDP[1] );
-
+    IDirectPlayLobby_Release(pDPL);
 }
 
 /* GetPlayerFlags */
-- 
1.9.1

