From e9f6d7e0e0b348f8b43359607a476cd03389b5cc Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 27 Nov 2015 11:34:08 +1100
Subject: [PATCH 2/9] d3dx10core.idl: Add ID3DX10Sprite interface

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/d3dx10core.idl | 53 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 53 insertions(+)

diff --git a/include/d3dx10core.idl b/include/d3dx10core.idl
index f0d8ff6..bcaff03 100644
--- a/include/d3dx10core.idl
+++ b/include/d3dx10core.idl
@@ -20,6 +20,23 @@ import "oaidl.idl";
 import "ocidl.idl";
 import "dxgi.idl";
 import "d3dcommon.idl";
+import "d3d10.idl";
+
+cpp_quote("#if 0")
+typedef struct D3DXMATRIX D3DXMATRIX;
+typedef struct D3DXVECTOR2 D3DXVECTOR2;
+typedef struct D3DXCOLOR D3DXCOLOR;
+cpp_quote("#endif")
+
+typedef struct _D3DX10_SPRITE
+{
+    D3DXMATRIX  matWorld;
+    D3DXVECTOR2 TexCoord;
+    D3DXVECTOR2 TexSize;
+    D3DXCOLOR   ColorModulate;
+    ID3D10ShaderResourceView *pTexture;
+    UINT        TextureIndex;
+} D3DX10_SPRITE;
 
 [
     object,
@@ -64,3 +81,39 @@ interface ID3DX10ThreadPump : IUnknown
     HRESULT PurgeAllItems();
     HRESULT GetQueueStatus([in] UINT *queue, [in] UINT *processqueue, [in] UINT *devicequeue);
 };
+
+[
+    object,
+    local,
+    pointer_default(unique),
+    uuid(ba0b762d-8d28-43ec-b9dc-2f84443b0614)
+]
+interface ID3DX10Sprite : IUnknown
+{
+    HRESULT Begin(UINT flags);
+
+    HRESULT DrawSpritesBuffered(D3DX10_SPRITE *sprites, UINT sprites_cnt);
+    HRESULT Flush();
+
+    HRESULT DrawSpritesImmediate(D3DX10_SPRITE *sprites, UINT sprites_cnt, UINT sprite, UINT flags);
+    HRESULT End();
+
+    HRESULT GetViewTransform(D3DXMATRIX *view_transform);
+    HRESULT SetViewTransform(D3DXMATRIX *view_transform);
+    HRESULT GetProjectionTransform(D3DXMATRIX *projection_transform);
+    HRESULT SetProjectionTransform(D3DXMATRIX *projection_transform);
+
+    HRESULT GetDevice(ID3D10Device** device);
+};
+
+cpp_quote("#ifdef __cplusplus")
+cpp_quote("extern \"C\" {")
+cpp_quote("#endif")
+cpp_quote("HRESULT WINAPI D3DX10CreateSprite(ID3D10Device *device, UINT size, ID3DX10Sprite **sprite);")
+cpp_quote("#ifdef __cplusplus")
+cpp_quote("}")
+cpp_quote("#endif")
+cpp_quote("")
+cpp_quote("#define _FACD3D  0x876")
+cpp_quote("#define MAKE_D3DHRESULT( code )                 MAKE_HRESULT( 1, _FACD3D, code )")
+cpp_quote("#define D3DERR_INVALIDCALL                      MAKE_D3DHRESULT(2156)")
-- 
1.9.1

