From 199fae10c01b2c3f67d609ebe8513b651e135346 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 27 Nov 2015 14:17:34 +1100
Subject: [PATCH 05/10] configure: Create d3dx10.lib

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2d2a168..187c963 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2791,7 +2791,7 @@ WINE_CONFIG_DLL(d3dx10_39)
 WINE_CONFIG_DLL(d3dx10_40)
 WINE_CONFIG_DLL(d3dx10_41)
 WINE_CONFIG_DLL(d3dx10_42)
-WINE_CONFIG_DLL(d3dx10_43)
+WINE_CONFIG_DLL(d3dx10_43,,[implib])
 WINE_CONFIG_DLL(d3dx11_42)
 WINE_CONFIG_DLL(d3dx11_43)
 WINE_CONFIG_DLL(d3dx9_24)
-- 
1.9.1

