From 83a9592f0d35ad1f8b8ad1432009b68185656792 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 27 Feb 2026 10:54:38 +1100
Subject: [PATCH 02/14] include: Add IDataTransferManagerInterop interface

---
 include/shobjidl_core.idl | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/shobjidl_core.idl b/include/shobjidl_core.idl
index 89a06bed47c..e86b666537a 100644
--- a/include/shobjidl_core.idl
+++ b/include/shobjidl_core.idl
@@ -15,6 +15,7 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
  */
+import "objidl.idl";
 
 cpp_quote("#define FOFX_NOSKIPJUNCTIONS 0x00010000")
 cpp_quote("#define FOFX_PREFERHARDLINK 0x00020000")
@@ -33,6 +34,17 @@ cpp_quote("#define FOFX_ADDUNDORECORD 0x20000000")
 cpp_quote("#define FOFX_COPYASDOWNLOAD 0x40000000")
 cpp_quote("#define FOFX_DONTDISPLAYLOCATIONS 0x80000000")
 
+[
+    object,
+    uuid(3a3dcd6c-3eab-43dc-bcde-45671ce800c8),
+    pointer_default(unique)
+]
+interface IDataTransferManagerInterop : IUnknown
+{
+    HRESULT GetForWindow( [in] HWND hwnd, [in] REFIID riid, [out, retval, iid_is(riid)] void **manager );
+    HRESULT ShowShareUIForWindow( [in] HWND hwnd );
+}
+
 [
     uuid(56f9f44f-f74c-4e38-99bc-9f3ebd3d696a)
 ]
-- 
2.51.0

