From 1b3a30ba57c07a9a97598c12a37c95c6ef44b49b Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 4 Mar 2026 14:04:59 +1100
Subject: [PATCH] wintypes: IApiInformationStatics return FALSE

---
 dlls/wintypes/main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/dlls/wintypes/main.c b/dlls/wintypes/main.c
index 287e5992225..2ea34734517 100644
--- a/dlls/wintypes/main.c
+++ b/dlls/wintypes/main.c
@@ -162,7 +162,9 @@ static HRESULT STDMETHODCALLTYPE api_information_statics_IsMethodPresent(
     if (!type_name)
         return E_INVALIDARG;
 
-    return E_NOTIMPL;
+    /* Add interface IDataTransferManagerStatics2, then this can return TRUE */
+    *value = FALSE;
+    return S_OK;
 }
 
 static HRESULT STDMETHODCALLTYPE api_information_statics_IsMethodPresentWithArity(
-- 
2.51.0

