From 832487b5f2a1826b2ccd50dc5112f1046f1266f1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 12 Mar 2026 12:01:06 +1100
Subject: [PATCH] propsys: Only trace Properties that cannot be found

---
 dlls/propsys/propsys_main.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/dlls/propsys/propsys_main.c b/dlls/propsys/propsys_main.c
index b8e800fbbc4..00c4336d2e1 100644
--- a/dlls/propsys/propsys_main.c
+++ b/dlls/propsys/propsys_main.c
@@ -166,14 +166,17 @@ static HRESULT WINAPI propsys_GetPropertyDescription(IPropertySystem *iface,
     HRESULT hr;
     IPropertyDescription *desc;
 
-    FIXME("(%p, %s, %s, %p): semi-stub!\n", iface, debugstr_propkey(propkey), debugstr_guid(riid), ppv);
+    TRACE("(%p, %s, %s, %p)\n", iface, debugstr_propkey(propkey), debugstr_guid(riid), ppv);
 
     if (!ppv)
         return E_INVALIDARG;
     *ppv = NULL;
     hr = propdesc_get_by_key( propkey, &desc );
     if (FAILED( hr ))
+    {
+        FIXME("propkey not found %s\n", debugstr_propkey(propkey));
         return hr;
+    }
 
     hr = IPropertyDescription_QueryInterface( desc, riid, ppv );
     IPropertyDescription_Release( desc );
@@ -186,14 +189,17 @@ static HRESULT WINAPI propsys_GetPropertyDescriptionByName(IPropertySystem *ifac
     HRESULT hr;
     IPropertyDescription *desc;
 
-    FIXME("(%p, %s, %s, %p): stub\n", iface, debugstr_w(canonical_name), debugstr_guid(riid), ppv);
+    TRACE("(%p, %s, %s, %p)\n", iface, debugstr_w(canonical_name), debugstr_guid(riid), ppv);
 
     if (!ppv)
         return E_INVALIDARG;
     *ppv = NULL;
     hr = propdesc_get_by_name( canonical_name, &desc );
     if (FAILED( hr ))
+    {
+        FIXME("canonical_name not found %s\n", debugstr_w(canonical_name));
         return hr;
+    }
     hr = IPropertyDescription_QueryInterface( desc, riid, ppv );
     IPropertyDescription_Release( desc );
     return hr;
-- 
2.51.0

