From 861a54e76ffbd9e252d476525627a94d7d47840d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 18 Mar 2026 16:23:25 +1100
Subject: [PATCH] GamingServer - To NOT merge

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index b54d478419f..1afa7a062fd 100755
--- a/configure
+++ b/configure
@@ -1748,6 +1748,7 @@ enable_fc
 enable_find
 enable_findstr
 enable_fsutil
+enable_gameinputsvc
 enable_hh
 enable_hostname
 enable_icacls
@@ -22340,6 +22341,7 @@ enable_icucommon=${enable_icucommon:-$enable_cplusplus}
 enable_icui18n=${enable_icui18n:-$enable_cplusplus}
 
 enable_conhost=${enable_conhost:-$HOST_ARCH}
+enable_gameinput=${enable_gameinput:-$HOST_ARCH}
 enable_lsass=${enable_lsass:-$HOST_ARCH}
 enable_plugplay=${enable_plugplay:-$HOST_ARCH}
 enable_rpcss=${enable_rpcss:-$HOST_ARCH}
@@ -23531,6 +23533,7 @@ wine_fn_config_makefile programs/findstr enable_findstr
 wine_fn_config_makefile programs/findstr/tests enable_tests
 wine_fn_config_makefile programs/fsutil enable_fsutil
 wine_fn_config_makefile programs/fsutil/tests enable_tests
+wine_fn_config_makefile programs/gameinputsvc enable_gameinputsvc
 wine_fn_config_makefile programs/hh enable_hh
 wine_fn_config_makefile programs/hostname enable_hostname
 wine_fn_config_makefile programs/icacls enable_icacls
-- 
2.53.0

