From 8fb2d949a72be8bb4c094054a15d62de60b0f85f Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 18 Mar 2026 16:23:25 +1100
Subject: [PATCH] GamingServer - To NOT merge

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index a076fd81269..9fbdbd07e5e 100755
--- a/configure
+++ b/configure
@@ -1745,6 +1745,7 @@ enable_fc
 enable_find
 enable_findstr
 enable_fsutil
+enable_gameinputsvc
 enable_hh
 enable_hostname
 enable_icacls
@@ -22882,6 +22883,7 @@ enable_icucommon=${enable_icucommon:-$enable_cplusplus}
 enable_icui18n=${enable_icui18n:-$enable_cplusplus}
 
 enable_conhost=${enable_conhost:-$HOST_ARCH}
+enable_gameinput=${enable_gameinput:-$HOST_ARCH}
 enable_plugplay=${enable_plugplay:-$HOST_ARCH}
 enable_rpcss=${enable_rpcss:-$HOST_ARCH}
 enable_services=${enable_services:-$HOST_ARCH}
@@ -24062,6 +24064,7 @@ wine_fn_config_makefile programs/findstr enable_findstr
 wine_fn_config_makefile programs/findstr/tests enable_tests
 wine_fn_config_makefile programs/fsutil enable_fsutil
 wine_fn_config_makefile programs/fsutil/tests enable_tests
+wine_fn_config_makefile programs/gameinputsvc enable_gameinputsvc
 wine_fn_config_makefile programs/hh enable_hh
 wine_fn_config_makefile programs/hostname enable_hostname
 wine_fn_config_makefile programs/icacls enable_icacls
-- 
2.53.0

