From a082eb1efb774b9d3a22b0cb450165ca6a81cd26 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 6 Apr 2020 15:14:31 +1000
Subject: [PATCH 2/2] dmime: Parse lyric data

---
 dlls/dmime/lyricstrack.c | 124 ++++++++++++++++++++++++++++++++++++++-
 1 file changed, 123 insertions(+), 1 deletion(-)

diff --git a/dlls/dmime/lyricstrack.c b/dlls/dmime/lyricstrack.c
index 49f44f52a34..e12350fedbf 100644
--- a/dlls/dmime/lyricstrack.c
+++ b/dlls/dmime/lyricstrack.c
@@ -21,6 +21,7 @@
 #include "dmobject.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(dmime);
+WINE_DECLARE_DEBUG_CHANNEL(dmfile);
 
 /*****************************************************************************
  * IDirectMusicLyricsTrack implementation
@@ -37,6 +38,11 @@ static inline IDirectMusicLyricsTrack *impl_from_IDirectMusicTrack8(IDirectMusic
     return CONTAINING_RECORD(iface, IDirectMusicLyricsTrack, IDirectMusicTrack8_iface);
 }
 
+static inline IDirectMusicLyricsTrack *impl_from_IPersistStream(IPersistStream *iface)
+{
+    return CONTAINING_RECORD(iface, IDirectMusicLyricsTrack, dmobj.IPersistStream_iface);
+}
+
 static HRESULT WINAPI lyrics_track_QueryInterface(IDirectMusicTrack8 *iface, REFIID riid,
         void **ret_iface)
 {
@@ -236,9 +242,125 @@ static const IDirectMusicTrack8Vtbl dmtrack8_vtbl = {
     lyrics_track_Join
 };
 
+static void parse_lyric_track_list(IDirectMusicLyricsTrack *This, IStream *stream)
+{
+    HRESULT hr;
+    LARGE_INTEGER liMove;
+    DMUS_PRIVATE_CHUNK chunk;
+    hr = IStream_Read (stream, &chunk.fccID, sizeof(FOURCC), NULL);
+    while(hr == S_OK) {
+        TRACE_(dmfile)(": LIST chunk of type %s\n", debugstr_fourcc(chunk.fccID));
+        switch (chunk.fccID) {
+            case FOURCC_LIST: {
+                IStream_Read(stream, &chunk.dwSize, sizeof(chunk.dwSize), NULL);
+                TRACE_(dmfile)("     - size %d\n", chunk.dwSize);
+                break;
+            }
+            case DMUS_FOURCC_LYRICSTRACKEVENTHEADER_CHUNK: {
+                DMUS_IO_LYRICSTRACK_EVENTHEADER header;
+
+                IStream_Read (stream, &chunk.dwSize, sizeof(chunk.dwSize), NULL);
+                TRACE_(dmfile)("     - size %d\n", chunk.dwSize);
+
+                IStream_Read (stream, &header, sizeof(header), NULL);
+                TRACE_(dmfile)("     - dwFlags 0x%08x\n", header.dwFlags);
+                TRACE_(dmfile)("     - dwTimingFlags 0x%08x\n", header.dwTimingFlags);
+                TRACE_(dmfile)("     - lTimeLogical %g\n", header.lTimeLogical);
+                TRACE_(dmfile)("     - lTimePhysical %g\n", header.lTimePhysical);
+                break;
+            }
+            case DMUS_FOURCC_LYRICSTRACKEVENTTEXT_CHUNK: {
+                WCHAR name[256];
+                IStream_Read (stream, &chunk.dwSize, sizeof(chunk.dwSize), NULL);
+                TRACE_(dmfile)("     - size %d\n", chunk.dwSize);
+
+                IStream_Read (stream, &name, chunk.dwSize, NULL);
+                TRACE_(dmfile)("     - name %s\n", debugstr_w(name));
+
+                return;
+            }
+            default:
+                TRACE_(dmfile)(": unknown (skipping)\n");
+                liMove.QuadPart = chunk.dwSize - sizeof(FOURCC);
+                IStream_Seek (stream, liMove, STREAM_SEEK_CUR, NULL);
+        }
+
+        hr = IStream_Read (stream, &chunk.fccID, sizeof(FOURCC), NULL);
+    }
+
+}
+
+static void parse_lyricstrack_list(IDirectMusicLyricsTrack *This, IStream *stream)
+{
+    HRESULT hr;
+    LARGE_INTEGER liMove;
+    DMUS_PRIVATE_CHUNK chunk;
+    hr = IStream_Read (stream, &chunk.fccID, sizeof(FOURCC), NULL);
+    while(hr == S_OK)
+    {
+        TRACE_(dmfile)(": LIST chunk of type %s\n", debugstr_fourcc(chunk.fccID));
+        switch (chunk.fccID) {
+            case FOURCC_LIST:
+            {
+                IStream_Read (stream, &chunk.dwSize, sizeof(chunk.dwSize), NULL);
+                TRACE_(dmfile)("     - size %d\n", chunk.dwSize);
+                break;
+            }
+            case DMUS_FOURCC_LYRICSTRACKEVENTS_LIST:
+            case DMUS_FOURCC_LYRICSTRACK_LIST: {
+                IStream_Read (stream, &chunk.fccID, sizeof(FOURCC), NULL);
+                continue;
+            }
+            case DMUS_FOURCC_LYRICSTRACKEVENT_LIST: {
+                parse_lyric_track_list(This, stream);
+                break;
+            }
+            case DMUS_FOURCC_TRACK_EXTRAS_CHUNK: {
+                DMUS_IO_TRACK_EXTRAS_HEADER txhdr = {0};
+
+                IStream_Read (stream, &txhdr, sizeof(txhdr), NULL);
+
+                TRACE("     - dwFlags: %#x, dwPriority: %u\n", txhdr.dwFlags, txhdr.dwPriority);
+                break;
+            }
+            default:
+                TRACE_(dmfile)(": unknown (skipping)\n");
+                liMove.QuadPart = chunk.dwSize - sizeof(FOURCC);
+                IStream_Seek (stream, liMove, STREAM_SEEK_CUR, NULL);
+        }
+
+        hr = IStream_Read (stream, &chunk.fccID, sizeof(FOURCC), NULL);
+    }
+}
+
 static HRESULT WINAPI lyrics_IPersistStream_Load(IPersistStream *iface, IStream *stream)
 {
-	FIXME(": Loading not implemented yet\n");
+    IDirectMusicLyricsTrack *This = impl_from_IPersistStream(iface);
+    HRESULT hr;
+    DMUS_PRIVATE_CHUNK chunk;
+    LARGE_INTEGER liMove;
+
+    TRACE("%p, %p\n", This, stream);
+
+    hr = IStream_Read (stream, &chunk, sizeof(FOURCC) + sizeof(DWORD), NULL);
+
+	while (hr == S_OK)
+    {
+	    TRACE_(dmfile)(": %s chunk (size = %d)\n", debugstr_fourcc (chunk.fccID), chunk.dwSize);
+        switch (chunk.fccID) {
+            case FOURCC_LIST: {
+                parse_lyricstrack_list(This, stream);
+                break;
+            }
+            default:
+                TRACE_(dmfile)(": Unknown LIST chunk of type %s\n", debugstr_fourcc(chunk.fccID));
+                liMove.QuadPart = chunk.dwSize - sizeof(FOURCC);
+                IStream_Seek (stream, liMove, STREAM_SEEK_CUR, NULL);
+        }
+
+        hr = IStream_Read (stream, &chunk, sizeof(FOURCC) + sizeof(DWORD), NULL);
+	}
+
 	return S_OK;
 }
 
-- 
2.26.1

