From c08c9072e24b6aef74a4136329cbeac7ab43e63c Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 24 Mar 2017 16:12:22 +1100
Subject: [PATCH] include/lmserver.h: Add missing SV_TYPE_* defines

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/lmserver.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/lmserver.h b/include/lmserver.h
index 785cae3..08b09a7 100644
--- a/include/lmserver.h
+++ b/include/lmserver.h
@@ -69,6 +69,10 @@ BOOL WINAPI SetServiceBits(SERVICE_STATUS_HANDLE,DWORD,BOOL,BOOL);
 #define SV_TYPE_SERVER_VMS        0x00200000
 #define SV_TYPE_WINDOWS           0x00400000
 #define SV_TYPE_DFS               0x00800000
+#define SV_TYPE_CLUSTER_NT        0x01000000
+#define SV_TYPE_TERMINALSERVER    0x02000000
+#define SV_TYPE_CLUSTER_VS_NT     0x04000000
+#define SV_TYPE_DCE               0x10000000
 #define SV_TYPE_ALTERNATE_XPORT   0x20000000
 #define SV_TYPE_LOCAL_LIST_ONLY   0x40000000
 #define SV_TYPE_DOMAIN_ENUM       0x80000000
-- 
1.9.1

