From 26115864fe14e18bbf65d4be116a8e62ba47d35d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 18 Jul 2019 13:40:01 +1000
Subject: include: Added d2d1helper.h


diff --git a/include/Makefile.in b/include/Makefile.in
index a4d3c9f7501..8ead7cdccb4 100644
--- a/include/Makefile.in
+++ b/include/Makefile.in
@@ -111,6 +111,7 @@ SOURCES = \
 	d2d1effects.idl \
 	d2d1effects_1.idl \
 	d2d1effects_2.idl \
+	d2d1helper.h \
 	d2dbasetypes.h \
 	d2derr.h \
 	d3d.h \
diff --git a/include/d2d1helper.h b/include/d2d1helper.h
new file mode 100644
index 00000000000..3430b5b04ee
--- /dev/null
+++ b/include/d2d1helper.h
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 Alistair Leslie-Hughes
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#ifndef _D2D1_HELPER_H_
+#define _D2D1_HELPER_H_
+
+#include <d2d1.h>
+
+#ifndef D2D_USE_C_DEFINITIONS
+
+/* TODO */
+
+#endif /* D2D_USE_C_DEFINITIONS */
+
+#endif /* _D2D1_HELPER_H_ */
