From 11ced8800171edef179793dba13b21906bda0244 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 20 Jan 2020 11:17:50 +1100
Subject: [PATCH 7/8] include: Add more atldef defines

---
 include/atldef.h | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/include/atldef.h b/include/atldef.h
index a93a53c09b2..f61264c99e6 100644
--- a/include/atldef.h
+++ b/include/atldef.h
@@ -44,6 +44,30 @@
 #define ATLAPI HRESULT __stdcall
 #define ATLAPI_(x) x __stdcall
 
+#define ATL_MAKEINTRESOURCEA(i) ((LPSTR)((ULONG_PTR)((WORD)(i))))
+#define ATL_MAKEINTRESOURCEW(i) ((LPWSTR)((ULONG_PTR)((WORD)(i))))
+#ifndef RC_INVOKED
+# ifdef WINE_NO_UNICODE_MACROS /* force using a cast */
+#  define ATL_MAKEINTRESOURCE(i) ((ULONG_PTR)((WORD)(i)))
+# else
+#  define MAKEINTRESOURCE WINELIB_NAME_AW(MAKEINTRESOURCE)
+# endif
+#endif
+
+#define ATL_IS_INTRESOURCE(x) (((ULONG_PTR)(x) >> 16) == 0)
+
+#define ATL_RT_CURSOR           ATL_MAKEINTRESOURCE(1)
+#define ATL_RT_BITMAP           ATL_MAKEINTRESOURCE(2)
+#define ATL_RT_ICON             ATL_MAKEINTRESOURCE(3)
+#define ATL_RT_MENU             ATL_MAKEINTRESOURCE(4)
+#define ATL_RT_DIALOG           ATL_MAKEINTRESOURCE(5)
+#define ATL_RT_STRING           ATL_MAKEINTRESOURCE(6)
+#define ATL_RT_FONTDIR          ATL_MAKEINTRESOURCE(7)
+#define ATL_RT_FONT             ATL_MAKEINTRESOURCE(8)
+#define ATL_RT_ACCELERATOR      ATL_MAKEINTRESOURCE(9)
+#define ATL_RT_RCDATA           ATL_MAKEINTRESOURCE(10)
+#define ATL_RT_MESSAGETABLE     ATL_MAKEINTRESOURCE(11)
+
 #define AtlThrow ATL::AtlThrowImpl
 
 #ifdef __cplusplus
-- 
2.17.1

