From da505c6e909235eacedd7a1d5b8e646c2fc6e8ab Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 30 Jul 2021 11:12:26 +1000
Subject: [PATCH 1/2] include: Add more sal.h defines

---
 include/sal.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/sal.h b/include/sal.h
index 5e51719d35b..f19e5fa735e 100644
--- a/include/sal.h
+++ b/include/sal.h
@@ -23,6 +23,10 @@
 #define __drv_allocatesMem(ptr)
 #define __drv_freesMem(ptr)
 
+#define _Acquires_lock_(a)
+#define _Releases_lock_(a)
+#define _Analysis_assume_(a)
+
 #define _Always_(exp)
 
 #define _Analysis_noreturn_
-- 
2.17.1

