From f2718d245a4bc9c8279c85122d1ec455ab7096b9 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 3 Dec 2024 09:50:58 +1100
Subject: [PATCH] include: Add _WIN32_WINNT_ version defines

---
 include/sdkddkver.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/sdkddkver.h b/include/sdkddkver.h
index baffe7a4a28..91057085295 100644
--- a/include/sdkddkver.h
+++ b/include/sdkddkver.h
@@ -79,4 +79,18 @@
 #define NTDDI_VERSION NTDDI_VERSION_FROM_WIN32_WINNT(_WIN32_WINNT)
 #endif
 
+#define _WIN32_WINNT_NT4          0x0400
+#define _WIN32_WINNT_WIN2K        0x0500
+#define _WIN32_WINNT_WINXP        0x0501
+#define _WIN32_WINNT_WS03         0x0502
+#define _WIN32_WINNT_WIN6         0x0600
+#define _WIN32_WINNT_VISTA        0x0600
+#define _WIN32_WINNT_WS08         0x0600
+#define _WIN32_WINNT_LONGHORN     0x0600
+#define _WIN32_WINNT_WIN7         0x0601
+#define _WIN32_WINNT_WIN8         0x0602
+#define _WIN32_WINNT_WINBLUE      0x0603
+#define _WIN32_WINNT_WINTHRESHOLD 0x0a00
+#define _WIN32_WINNT_WIN10        0x0a00
+
 #endif /* _INC_SDKDDKVER */
-- 
2.45.2

