From de75744be244f0e2d0d858b2163ac9234bc05ea2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 27 May 2024 04:41:40 +1000
Subject: [PATCH] include: Add IFACEMETHOD macros

---
 include/objbase.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/objbase.h b/include/objbase.h
index bae8de54bab..024d96ca7da 100644
--- a/include/objbase.h
+++ b/include/objbase.h
@@ -210,6 +210,11 @@
 #define DECLARE_INTERFACE_(iface,ibase) interface DECLSPEC_NOVTABLE iface : public ibase
 #define DECLARE_INTERFACE_IID_(iface, ibase, iid) interface DECLSPEC_UUID(iid) DECLSPEC_NOVTABLE iface : public ibase
 
+#define IFACEMETHOD(method)         STDMETHOD(method)
+#define IFACEMETHOD_(type,method)   STDMETHOD_(type,method)
+#define IFACEMETHODV(method)        STDMETHODV(method)
+#define IFACEMETHODV_(type,method)  STDMETHODV_(type,method)
+
 #define BEGIN_INTERFACE
 #define END_INTERFACE
 
@@ -253,6 +258,11 @@
 #define BEGIN_INTERFACE
 #define END_INTERFACE
 
+#define IFACEMETHOD(method)         STDMETHOD(method)
+#define IFACEMETHOD_(type,method)   STDMETHOD_(type,method)
+#define IFACEMETHODV(method)        STDMETHODV(method)
+#define IFACEMETHODV_(type,method)  STDMETHODV_(type,method)
+
 #endif  /* __cplusplus && !CINTERFACE */
 
 #ifndef __IRpcStubBuffer_FWD_DEFINED__
-- 
2.43.0

