From 47490951fd1683311e87e0b274b5df779b00cb63 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 18 Mar 2025 13:17:27 +1100
Subject: [PATCH] include: Add ApplicationRecoveryInProgress/Finished prototype

---
 include/winbase.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/winbase.h b/include/winbase.h
index e4eec9532e4..896b1bd42b2 100644
--- a/include/winbase.h
+++ b/include/winbase.h
@@ -1498,6 +1498,8 @@ WINBASEAPI VOID        WINAPI AcquireSRWLockShared(PSRWLOCK);
 WINADVAPI  BOOL        WINAPI AdjustTokenPrivileges(HANDLE,BOOL,PTOKEN_PRIVILEGES,DWORD,PTOKEN_PRIVILEGES,PDWORD);
 WINADVAPI  BOOL        WINAPI AllocateAndInitializeSid(PSID_IDENTIFIER_AUTHORITY,BYTE,DWORD,DWORD,DWORD,DWORD,DWORD,DWORD,DWORD,DWORD,PSID *);
 WINADVAPI  BOOL        WINAPI AllocateLocallyUniqueId(PLUID);
+WINBASEAPI HRESULT     WINAPI ApplicationRecoveryInProgress(PBOOL);
+WINBASEAPI VOID        WINAPI ApplicationRecoveryFinished(BOOL);
 WINADVAPI  BOOL        WINAPI AreAllAccessesGranted(DWORD,DWORD);
 WINADVAPI  BOOL        WINAPI AreAnyAccessesGranted(DWORD,DWORD);
 WINBASEAPI BOOL        WINAPI AreFileApisANSI(void);
@@ -2295,6 +2297,8 @@ WINBASEAPI BOOL        WINAPI ReadFile(HANDLE,LPVOID,DWORD,LPDWORD,LPOVERLAPPED)
 WINBASEAPI BOOL        WINAPI ReadFileEx(HANDLE,LPVOID,DWORD,LPOVERLAPPED,LPOVERLAPPED_COMPLETION_ROUTINE);
 WINBASEAPI BOOL        WINAPI ReadFileScatter(HANDLE,FILE_SEGMENT_ELEMENT*,DWORD,LPDWORD,LPOVERLAPPED);
 WINBASEAPI BOOL        WINAPI ReadProcessMemory(HANDLE,LPCVOID,LPVOID,SIZE_T,SIZE_T*);
+WINBASEAPI HRESULT     WINAPI RegisterApplicationRecoveryCallback(APPLICATION_RECOVERY_CALLBACK,PVOID,DWORD,DWORD);
+WINBASEAPI HRESULT     WINAPI RegisterApplicationRestart(PCWSTR,DWORD);
 WINADVAPI  HANDLE      WINAPI RegisterEventSourceA(LPCSTR,LPCSTR);
 WINADVAPI  HANDLE      WINAPI RegisterEventSourceW(LPCWSTR,LPCWSTR);
 #define                       RegisterEventSource WINELIB_NAME_AW(RegisterEventSource)
@@ -2445,6 +2449,7 @@ WINBASEAPI BOOL        WINAPI UnmapViewOfFile(LPCVOID);
 WINBASEAPI BOOL        WINAPI UnmapViewOfFile2(HANDLE,PVOID,ULONG);
 WINBASEAPI BOOL        WINAPI UnmapViewOfFileEx(PVOID,ULONG);
 WINBASEAPI BOOL        WINAPI UmsThreadYield(void *);
+WINBASEAPI HRESULT     WINAPI UnregisterApplicationRecoveryCallback(void);
 WINBASEAPI HRESULT     WINAPI UnregisterApplicationRestart(void);
 WINBASEAPI BOOL        WINAPI UnregisterWait(HANDLE);
 WINBASEAPI BOOL        WINAPI UnregisterWaitEx(HANDLE,HANDLE);
-- 
2.47.2

