From 6546267d6694dea2b573156bca604da364bbea8d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 13 Oct 2025 12:14:56 +1100
Subject: [PATCH] include: Add missing AddrBStore to STACKFRAME in imagehlp.h

---
 include/imagehlp.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/imagehlp.h b/include/imagehlp.h
index 3c7bb90b511..9ec53ec2968 100644
--- a/include/imagehlp.h
+++ b/include/imagehlp.h
@@ -278,6 +278,7 @@ typedef struct _STACKFRAME {
   BOOL    Virtual;
   DWORD     Reserved[3];
   KDHELP  KdHelp;
+  ADDRESS AddrBStore;
 } STACKFRAME, *LPSTACKFRAME;
 
 typedef struct _STACKFRAME64 {
-- 
2.51.0

