From 564490af9459f877551ee401fb1992ecbdbc2554 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 13 Oct 2025 12:14:56 +1100
Subject: [PATCH] include: Correct STACKFRAME for 64bits in imagehlp.h

---
 include/imagehlp.h | 31 +++++++++++++++++++------------
 1 file changed, 19 insertions(+), 12 deletions(-)

diff --git a/include/imagehlp.h b/include/imagehlp.h
index 3c7bb90b511..3df6e0730a6 100644
--- a/include/imagehlp.h
+++ b/include/imagehlp.h
@@ -267,18 +267,25 @@ typedef struct _KDHELP64 {
     DWORD64 Reserved[8];
 } KDHELP64, *PKDHELP64;
 
-typedef struct _STACKFRAME {
-  ADDRESS AddrPC;
-  ADDRESS AddrReturn;
-  ADDRESS AddrFrame;
-  ADDRESS AddrStack;
-  PVOID     FuncTableEntry;
-  DWORD     Params[4];
-  BOOL    Far;
-  BOOL    Virtual;
-  DWORD     Reserved[3];
-  KDHELP  KdHelp;
-} STACKFRAME, *LPSTACKFRAME;
+#if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
+#define STACKFRAME STACKFRAME64
+#define LPSTACKFRAME LPSTACKFRAME64
+#else
+  typedef struct _STACKFRAME
+  {
+    ADDRESS     AddrPC;
+    ADDRESS     AddrReturn;
+    ADDRESS     AddrFrame;
+    ADDRESS     AddrStack;
+    PVOID       FuncTableEntry;
+    DWORD       Params[4];
+    BOOL        Far;
+    BOOL        Virtual;
+    DWORD       Reserved[3];
+    KDHELP      KdHelp;
+    ADDRESS     AddrBStore;
+  } STACKFRAME, *LPSTACKFRAME;
+#endif
 
 typedef struct _STACKFRAME64 {
     ADDRESS64 AddrPC;
-- 
2.51.0

