From 955a881fe019cfcdc8883075da463b3ff2790bdd Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 27 May 2024 14:39:58 +1000
Subject: [PATCH] include: Add IID_PPV_ARGS macro

---
 include/combaseapi.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/combaseapi.h b/include/combaseapi.h
index 4b8cb457328..831b2a33053 100644
--- a/include/combaseapi.h
+++ b/include/combaseapi.h
@@ -46,6 +46,8 @@ typedef struct tagServerInformation
 
 HRESULT WINAPI CoDecodeProxy(DWORD client_pid, UINT64 proxy_addr, ServerInformation *server_info);
 
+#define IID_PPV_ARGS(type) __uuidof(**(type))
+
 #ifdef __cplusplus
 }
 #endif
-- 
2.43.0

