From 20d13a1f6ea7e108cf87b958ff2c1487c0f34b67 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 27 May 2024 14:39:58 +1000
Subject: [PATCH] include: Add IID_PPV_ARGS macro

---
 include/combaseapi.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/combaseapi.h b/include/combaseapi.h
index 41461f22e98..d3aa4a08bd9 100644
--- a/include/combaseapi.h
+++ b/include/combaseapi.h
@@ -57,4 +57,17 @@ HRESULT WINAPI RoGetAgileReference(enum AgileReferenceOptions options, REFIID ri
 }
 #endif
 
+#ifdef __cplusplus
+
+extern "C++" template<typename T> void **IID_PPV_ARGS_Helper(T **obj)
+{
+    (void)static_cast<IUnknown *>(*obj);
+    return reinterpret_cast<void **>(obj);
+}
+
+
+#define IID_PPV_ARGS(type) __uuidof(**(type)), IID_PPV_ARGS_Helper(type)
+
+#endif
+
 #endif /* _COMBASEAPI_H_ */
-- 
2.47.2

