From 752adf68d35566b1a8b1b6f6a40fb39ff5b842a8 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 18 Mar 2025 13:36:56 +1100
Subject: [PATCH] include: Add InitNetworkAddressControl prototype

---
 include/shellapi.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/shellapi.h b/include/shellapi.h
index ff36803fa19..2fccf37802d 100644
--- a/include/shellapi.h
+++ b/include/shellapi.h
@@ -687,6 +687,7 @@ WINSHELLAPI UINT        WINAPI ExtractIconExW(LPCWSTR,INT,HICON*,HICON*,UINT);
 WINSHELLAPI HINSTANCE   WINAPI FindExecutableA(LPCSTR,LPCSTR,LPSTR);
 WINSHELLAPI HINSTANCE   WINAPI FindExecutableW(LPCWSTR,LPCWSTR,LPWSTR);
 #define                        FindExecutable WINELIB_NAME_AW(FindExecutable)
+WINSHELLAPI BOOL        WINAPI InitNetworkAddressControl(void);
 WINSHELLAPI BOOL        WINAPI ShellAboutA(HWND,LPCSTR,LPCSTR,HICON);
 WINSHELLAPI BOOL        WINAPI ShellAboutW(HWND,LPCWSTR,LPCWSTR,HICON);
 #define                        ShellAbout WINELIB_NAME_AW(ShellAbout)
-- 
2.47.2

