From b6f34c3ceb02b23717954cbc921720e4e5f7ccec Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 11 Mar 2025 10:01:44 +1100
Subject: [PATCH] include: Add _wcsinc define

---
 include/tchar.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/tchar.h b/include/tchar.h
index e0436c51c6e..0d0bd100167 100644
--- a/include/tchar.h
+++ b/include/tchar.h
@@ -36,6 +36,7 @@ extern "C" {
 #define _strnextc(str)          ((unsigned int)*(str))
 #define _strninc(str,n)         (((char*)(str))+(n))
 #define _strspnp(s1,s2)         (*((s1)+=strspn((s1),(s2))) ? (s1) : NULL)
+#define _wcsinc(current)        (((wchar_t*)(current))+1)
 
 
 /*****************************************************************************
-- 
2.47.2

