From e3685f5e5445cc643a68d6d41ac7b5ff3c4588b2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 3 Dec 2024 12:37:40 +1100
Subject: [PATCH] include: Add DB_VARNUMERIC struct

---
 include/dbs.idl | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/dbs.idl b/include/dbs.idl
index aea304d4e62..4c6781dddbb 100644
--- a/include/dbs.idl
+++ b/include/dbs.idl
@@ -828,6 +828,15 @@ typedef struct tagDBTIMESTAMP {
 	ULONG fraction;
 } DBTIMESTAMP;
 
+typedef signed char SBYTE;
+
+typedef struct tagDB_VARNUMERIC {
+	BYTE  precision;
+	SBYTE scale;
+	BYTE  sign;
+	BYTE  val[];
+} DB_VARNUMERIC;
+
 typedef DWORD DBREASON;
 typedef DWORD DBEVENTPHASE;
 
-- 
2.45.2

