From 282e4d61952194e0206b80737d3894a3fed1d9fd Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 25 Oct 2024 15:59:01 +1100
Subject: [PATCH] include: Add DB_S_ROWLIMITEXCEEDED define

---
 include/oledberr.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/oledberr.h b/include/oledberr.h
index 1ca5446d06e..b3445d50c10 100644
--- a/include/oledberr.h
+++ b/include/oledberr.h
@@ -93,6 +93,7 @@
 
 #define DB_E_MISMATCHEDPROVIDER             0x80040e75
 
+#define DB_S_ROWLIMITEXCEEDED               0x00040ec0
 #define DB_S_BOOKMARKSKIPPED                0x00040ec3
 #define DB_S_NONEXTROWSET                   0x00040ec5
 #define DB_S_ENDOFROWSET                    0x00040ec6
-- 
2.45.2

