From 7a9a78cde6625c8348e05623e7a2f82083adc468 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 25 Oct 2024 16:03:23 +1100
Subject: [PATCH] include: dbs.idl: Added DBVECTOR/DB_VARNUMERIC types

---
 include/dbs.idl | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/dbs.idl b/include/dbs.idl
index 334c6daff80..c81fe5d4372 100644
--- a/include/dbs.idl
+++ b/include/dbs.idl
@@ -784,6 +784,11 @@ typedef struct tagRMTPACK
     [size_is(cArray)] VARIANT *rgArray;
 } RMTPACK;
 
+typedef struct tagDBVECTOR {
+	DBLENGTH size;
+	[size_is((ULONG)size)] void *ptr;
+} DBVECTOR;
+
 typedef struct tagDBDATE {
 	SHORT year;
 	USHORT month;
-- 
2.45.2

