From 573113609eb4313f29bfd047ceadcb2f63aadfdc Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 30 Oct 2024 11:01:00 +1100
Subject: [PATCH] include: Move ISAXXMLFilter interface to after base
 ISAXXMLReader

---
 include/msxml6.idl | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/include/msxml6.idl b/include/msxml6.idl
index b2d8bd3b337..a02f7b2db14 100644
--- a/include/msxml6.idl
+++ b/include/msxml6.idl
@@ -1831,19 +1831,6 @@ interface ISAXLocator : IUnknown
         [out, retval] const WCHAR ** systemId);
 }
 
-[
-    local,
-    object,
-    uuid(70409222-ca09-4475-acb8-40312fe8d145)
-]
-interface ISAXXMLFilter : ISAXXMLReader
-{
-    HRESULT getParent(
-        [out, retval] ISAXXMLReader ** pReader);
-    HRESULT putParent(
-        [in] ISAXXMLReader * reader);
-}
-
 [
 local,
 object,
@@ -1901,6 +1888,19 @@ interface ISAXXMLReader : IUnknown
         [in] const WCHAR * url);
 }
 
+[
+    local,
+    object,
+    uuid(70409222-ca09-4475-acb8-40312fe8d145)
+]
+interface ISAXXMLFilter : ISAXXMLReader
+{
+    HRESULT getParent(
+        [out, retval] ISAXXMLReader ** pReader);
+    HRESULT putParent(
+        [in] ISAXXMLReader * reader);
+}
+
 [
     object,
     dual,
-- 
2.45.2

