From 4b3fefc8d7e6de729cce82c07314f54a9fabdeba Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 16 Mar 2026 05:26:10 +1100
Subject: [PATCH] include: Add define WC_NETADDRESS

---
 include/shellapi.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/shellapi.h b/include/shellapi.h
index 1516e7a7ac5..43fd9f6d5f9 100644
--- a/include/shellapi.h
+++ b/include/shellapi.h
@@ -521,6 +521,13 @@ typedef struct tagNC_ADDRESS
    BYTE                     PrefixLength;
 } NC_ADDRESS, *PNC_ADDRESS;
 
+#if defined(_MSC_VER) || defined(__MINGW32__)
+# define WC_NETADDRESS       L"msctls_netaddress"
+#else
+static const WCHAR WC_NETADDRESS[] = { 'm','s','c','t','l','s','_',
+  'n','e','t','a','d','d','r','e','s','s',0 };
+#endif
+
 #define NCM_GETADDRESS      (WM_USER+1)
 #define NCM_SETALLOWTYPE    (WM_USER+2)
 #define NCM_GETALLOWTYPE    (WM_USER+3)
-- 
2.53.0

