From 205d9aea474afda1ef15302adcd36c2c943e3aca Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 5 May 2025 16:14:07 +1000
Subject: [PATCH] include: Add IHTMLOptionButtonElement interface

---
 include/mshtmdid.h | 12 +++++++++
 include/mshtml.idl | 66 +++++++++++++++++++++++++++++++++++++++++++++-
 2 files changed, 77 insertions(+), 1 deletion(-)

diff --git a/include/mshtmdid.h b/include/mshtmdid.h
index a28d706903f..8bb1252c50f 100644
--- a/include/mshtmdid.h
+++ b/include/mshtmdid.h
@@ -4830,4 +4830,16 @@
 #define DISPID_IHTMLNAMESPACECOLLECTION_ITEM    DISPID_VALUE
 #define DISPID_IHTMLNAMESPACECOLLECTION_ADD     DISPID_NAMESPACE_COLLECTION+1
 
+/* IHTMLOptionButtonElement */
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_VALUE                     DISPID_A_VALUE
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_TYPE                      DISPID_INPUT
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_NAME                      STDPROPID_XOBJ_NAME
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_CHECKED                   DISPID_INPUT+9
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_DEFAULTCHECKED            DISPID_INPUT+8
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_ONCHANGE                  DISPID_EVPROP_ONCHANGE
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_DISABLED                  STDPROPID_XOBJ_DISABLED
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_STATUS                    DISPID_INPUT+1
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_INDETERMINATE             DISPID_INPUT+7
+#define DISPID_IHTMLOPTIONBUTTONELEMENT_FORM                      DISPID_SITE+4
+
 #endif /* __MSHTMDID_H__ */
diff --git a/include/mshtml.idl b/include/mshtml.idl
index 2a1fccba6be..9f168202d3c 100644
--- a/include/mshtml.idl
+++ b/include/mshtml.idl
@@ -16226,6 +16226,70 @@ methods:
     void setSelectionRange([in] long start, [in] long end);
 }
 
+[
+    odl,
+    oleautomation,
+    dual,
+    uuid(3050f2bc-98b5-11cf-bb82-00aa00bdce0b)
+]
+interface IHTMLOptionButtonElement : IDispatch
+{
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_VALUE), displaybind, bindable]
+    HRESULT value([in] BSTR v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_VALUE), displaybind, bindable]
+    HRESULT value([retval, out] BSTR *p);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_TYPE)]
+    HRESULT type([retval, out] BSTR *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_NAME), displaybind, bindable]
+    HRESULT name([in] BSTR v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_NAME), displaybind, bindable]
+    HRESULT name([retval, out] BSTR *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_CHECKED), bindable]
+    HRESULT checked([in] VARIANT_BOOL v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_CHECKED), bindable]
+    HRESULT checked([retval, out] VARIANT_BOOL *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_DEFAULTCHECKED), bindable]
+    HRESULT defaultChecked([in] VARIANT_BOOL v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_DEFAULTCHECKED), bindable]
+    HRESULT defaultChecked([retval, out] VARIANT_BOOL *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_ONCHANGE), displaybind, bindable]
+    HRESULT onchange([in] VARIANT v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_ONCHANGE), displaybind, bindable]
+    HRESULT onchange([retval, out] VARIANT *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_DISABLED), displaybind, bindable]
+    HRESULT disabled([in] VARIANT_BOOL v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_DISABLED), displaybind, bindable]
+    HRESULT disabled([retval, out] VARIANT_BOOL *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_STATUS)]
+    HRESULT status([in] VARIANT_BOOL v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_STATUS)]
+    HRESULT status([retval, out] VARIANT_BOOL *p);
+
+    [propput, id(DISPID_IHTMLOPTIONBUTTONELEMENT_INDETERMINATE), bindable]
+    HRESULT indeterminate([in] VARIANT_BOOL v);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_INDETERMINATE), bindable]
+    HRESULT indeterminate([retval, out] VARIANT_BOOL *p);
+
+    [propget, id(DISPID_IHTMLOPTIONBUTTONELEMENT_FORM)]
+    HRESULT form([retval, out] IHTMLFormElement* *p);
+};
+
+
 /*****************************************************************************
  *    HTMLInputElement coclass
  */
@@ -16249,7 +16313,7 @@ coclass HTMLInputElement
  /* interface IHTMLInputHiddenElement; */
  /* interface IHTMLInputButtonElement; */
  /* interface IHTMLInputFileElement; */
- /* interface IHTMLOptionButtonElement; */
+    interface IHTMLOptionButtonElement;
  /* interface IHTMLInputImage; */
  /* interface IHTMLInputElement3; */
  /* interface IHTMLInputRangeElement; */
-- 
2.47.2

