From aeb6be955a57fcbe5ed20ea7e6c9fb51928b07b1 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 17 Dec 2019 15:42:32 +1100
Subject: [PATCH] msdasql: Implement MSDASQL provider

---
 dlls/msdasql/msdasql_main.c | 97 +++++++++++++++++++++++++++++++++++++
 1 file changed, 97 insertions(+)

diff --git a/dlls/msdasql/msdasql_main.c b/dlls/msdasql/msdasql_main.c
index 7b1ad8f53c..6a9ace3153 100644
--- a/dlls/msdasql/msdasql_main.c
+++ b/dlls/msdasql/msdasql_main.c
@@ -24,8 +24,10 @@
 #include "winbase.h"
 #include "objbase.h"
 #include "rpcproxy.h"
+#include "wine/heap.h"
 #include "wine/debug.h"
 
+#include "initguid.h"
 #include "msdasql_classes.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(msdasql);
@@ -103,10 +105,17 @@ static ULONG WINAPI ClassFactory_Release(IClassFactory *iface)
     return 1;
 }
 
+static HRESULT create_msdasql_provider(void **ppv);
+
 HRESULT WINAPI ClassFactory_CreateInstance(IClassFactory *iface, IUnknown *outer, REFIID riid, void **ppv)
 {
     FIXME("(%p %s %p)\n", outer, debugstr_guid(riid), ppv);
 
+    if (IsEqualGUID(&CLSID_MSDASQL, riid))
+    {
+        return create_msdasql_provider(ppv);
+    }
+
     return CLASS_E_CLASSNOTAVAILABLE;
 }
 
@@ -131,3 +140,91 @@ HRESULT WINAPI DllGetClassObject( REFCLSID rclsid, REFIID riid, void **ppv )
     FIXME("%s %s %p\n", debugstr_guid(rclsid), debugstr_guid(riid), ppv);
     return IClassFactory_QueryInterface(&cffactory, riid, ppv);
 }
+
+struct msdasql
+{
+    IUnknown         MSDASQL_iface;
+
+/*    IDBCreateSession IDBCreateSession_iface;
+    IDBInitialize    IDBInitialize_iface;
+    IDBProperties    IDBProperties_iface;*/
+
+    LONG     ref;
+};
+
+static inline struct msdasql *impl_from_IUnknown(IUnknown *iface)
+{
+    return CONTAINING_RECORD(iface, struct msdasql, MSDASQL_iface);
+}
+
+static HRESULT WINAPI msdsql_QueryInterface(IUnknown *iface, REFIID riid, void **out)
+{
+    struct msdasql *provider = impl_from_IUnknown(iface);
+
+    TRACE("(%p)->(%s %p)\n", provider, debugstr_guid(riid), out);
+
+    if(IsEqualGUID(riid, &IID_IUnknown) ||
+       IsEqualGUID(riid, &CLSID_MSDASQL))
+    {
+        *out = &provider->MSDASQL_iface;
+    }
+    else
+    {
+        FIXME("(%s, %p)\n", debugstr_guid(riid), out);
+        *out = NULL;
+        return E_NOINTERFACE;
+    }
+
+    IUnknown_AddRef((IUnknown*)*out);
+    return S_OK;
+}
+
+static ULONG WINAPI msdsql_AddRef(IUnknown *iface)
+{
+    struct msdasql *provider = impl_from_IUnknown(iface);
+    ULONG ref = InterlockedIncrement(&provider->ref);
+
+    TRACE("(%p) ref=%u\n", provider, ref);
+
+    return ref;
+}
+
+static ULONG  WINAPI msdsql_Release(IUnknown *iface)
+{
+    struct msdasql *provider = impl_from_IUnknown(iface);
+    ULONG ref = InterlockedDecrement(&provider->ref);
+
+    TRACE("(%p) ref=%u\n", provider, ref);
+
+    if (!ref)
+    {
+        heap_free(provider);
+    }
+
+    return ref;
+}
+
+static const IUnknownVtbl msdsql_vtbl =
+{
+    msdsql_QueryInterface,
+    msdsql_AddRef,
+    msdsql_Release
+};
+
+static HRESULT create_msdasql_provider(void **ppv)
+{
+    struct msdasql *provider;
+
+    provider = heap_alloc(sizeof(struct msdasql));
+    if (!provider)
+        return E_OUTOFMEMORY;
+
+    provider->MSDASQL_iface.lpVtbl = &msdsql_vtbl;
+    provider->ref = 1;
+
+    *ppv = (void*)provider;
+
+    return S_OK;
+}
+
+
-- 
2.17.1

