From 1a86b1825734e5dd60c01f5715c19ed9e15132ff Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 22 Nov 2021 16:44:51 +1100
Subject: [PATCH 04/15] msdasql: Implement ICommandPrepare Prepare

---
 dlls/msdasql/session.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index f9538e5d4d9..14ae853bb64 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -335,6 +335,7 @@ struct command
     LONG refs;
     WCHAR *query;
     IUnknown *session;
+    SQLHSTMT hstmt;
 
     struct command_props properties[79];
 };
@@ -456,6 +457,10 @@ static ULONG WINAPI command_Release(ICommandText *iface)
         TRACE( "destroying %p\n", command );
         if (command->session)
             IUnknown_Release(command->session);
+
+        if (command->hstmt)
+            SQLFreeHandle(SQL_HANDLE_STMT, command->hstmt);
+
         heap_free( command->query );
         heap_free( command );
     }
@@ -1230,7 +1235,24 @@ static ULONG WINAPI commandprepare_Release(ICommandPrepare *iface)
 static HRESULT WINAPI commandprepare_Prepare(ICommandPrepare *iface, ULONG runs)
 {
     struct command *command = impl_from_ICommandPrepare( iface );
+    RETCODE ret;
+
     TRACE("%p, %u\n", command, runs);
+
+    if (!command->query)
+        return DB_E_NOCOMMAND;
+
+    if (command->hstmt)
+        SQLFreeHandle(SQL_HANDLE_STMT, command->hstmt);
+
+    SQLAllocHandle(SQL_HANDLE_STMT, command->hdbc, &command->hstmt);
+
+    ret = SQLPrepareW(command->hstmt, command->query, lstrlenW(command->query));
+    if (ret != SQL_SUCCESS)
+    {
+        dump_sql_diag_records(SQL_HANDLE_STMT, command->hstmt);
+        return E_FAIL;
+    }
     return S_OK;
 }
 
@@ -1327,6 +1349,7 @@ static HRESULT WINAPI createcommand_CreateCommand(IDBCreateCommand *iface, IUnkn
     command->ICommandWithParameters_iface.lpVtbl = &command_with_params_vtbl;
     command->refs = 1;
     command->query = NULL;
+    command->hstmt = NULL;
 
     for(i = 0; i < ARRAY_SIZE(command->properties); i++)
     {
-- 
2.33.0

