From 2a33b940e9b5a6d41d4cea55ec05e21395ed4674 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 19 Dec 2019 09:37:40 +1100
Subject: [PATCH] msdasql: Support IDBProperties interface.

---
 dlls/msdasql/msdasql_main.c | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/dlls/msdasql/msdasql_main.c b/dlls/msdasql/msdasql_main.c
index 677eb157313..b55f9b6d023 100644
--- a/dlls/msdasql/msdasql_main.c
+++ b/dlls/msdasql/msdasql_main.c
@@ -24,6 +24,7 @@
 #include "winbase.h"
 #include "objbase.h"
 #include "rpcproxy.h"
+#include "msdasc.h"
 #include "wine/heap.h"
 #include "wine/debug.h"
 
@@ -143,10 +144,11 @@ HRESULT WINAPI DllGetClassObject( REFCLSID rclsid, REFIID riid, void **ppv )
 struct msdasql
 {
     IUnknown         MSDASQL_iface;
+    IDBProperties    IDBProperties_iface;
 
 /*    IDBCreateSession IDBCreateSession_iface;
     IDBInitialize    IDBInitialize_iface;
-    IDBProperties    IDBProperties_iface;*/
+    */
 
     LONG     ref;
 };
@@ -156,6 +158,11 @@ static inline struct msdasql *impl_from_IUnknown(IUnknown *iface)
     return CONTAINING_RECORD(iface, struct msdasql, MSDASQL_iface);
 }
 
+static inline struct msdasql *impl_from_IDBProperties(IDBProperties *iface)
+{
+    return CONTAINING_RECORD(iface, struct msdasql, IDBProperties_iface);
+}
+
 static HRESULT WINAPI msdsql_QueryInterface(IUnknown *iface, REFIID riid, void **out)
 {
     struct msdasql *provider = impl_from_IUnknown(iface);
@@ -167,6 +174,10 @@ static HRESULT WINAPI msdsql_QueryInterface(IUnknown *iface, REFIID riid, void *
     {
         *out = &provider->MSDASQL_iface;
     }
+    else if ( IsEqualGUID(riid, &IID_IDBProperties))
+    {
+        *out = &provider->IDBProperties_iface;
+    }
     else
     {
         FIXME("(%s, %p)\n", debugstr_guid(riid), out);
@@ -246,10 +257,19 @@ static HRESULT WINAPI dbprops_GetPropertyInfo(IDBProperties *iface, ULONG cPrope
             DBPROPINFOSET **prgPropertyInfoSets, OLECHAR **ppDescBuffer)
 {
     struct msdasql *provider = impl_from_IDBProperties(iface);
+    int i;
 
     FIXME("(%p)->(%d %p %p %p %p)\n", provider, cPropertyIDSets, rgPropertyIDSets, pcPropertyInfoSets,
                 prgPropertyInfoSets, ppDescBuffer);
 
+    for (i=0; i < cPropertyIDSets; i++)
+    {
+        TRACE("Property Set %d\n", i);
+        TRACE("  rgPropertyIDs: %p\n", rgPropertyIDSets[i].rgPropertyIDs);
+        TRACE("  cPropertyIDs:  %d\n", rgPropertyIDSets[i].cPropertyIDs);
+        TRACE("  guidPropertySet: %s\n", debugstr_guid(&rgPropertyIDSets[i].guidPropertySet) );
+    }
+
     return E_NOTIMPL;
 }
 
@@ -283,6 +303,7 @@ static HRESULT create_msdasql_provider(REFIID riid, void **ppv)
         return E_OUTOFMEMORY;
 
     provider->MSDASQL_iface.lpVtbl = &msdsql_vtbl;
+    provider->IDBProperties_iface.lpVtbl = &dbprops_vtbl;
     provider->ref = 1;
 
     hr = IUnknown_QueryInterface(&provider->MSDASQL_iface, riid, ppv);
-- 
2.29.2

