From 1d621433ce86fe83ddbc94eacc888f1ec72e8e46 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 30 May 2020 13:44:12 +1000
Subject: [PATCH 08/11] msdasql: Add IDBInitialize support

---
 dlls/msdasql/msdasql_main.c | 61 ++++++++++++++++++++++++++++++++++++-
 1 file changed, 60 insertions(+), 1 deletion(-)

diff --git a/dlls/msdasql/msdasql_main.c b/dlls/msdasql/msdasql_main.c
index 28baf27a9f..c1929192b6 100644
--- a/dlls/msdasql/msdasql_main.c
+++ b/dlls/msdasql/msdasql_main.c
@@ -243,9 +243,9 @@ struct msdasql
 {
     IUnknown         MSDASQL_iface;
     IDBProperties    IDBProperties_iface;
+    IDBInitialize    IDBInitialize_iface;
 
 /*    IDBCreateSession IDBCreateSession_iface;
-    IDBInitialize    IDBInitialize_iface;
     */
 
     LONG     ref;
@@ -261,6 +261,11 @@ static inline struct msdasql *impl_from_IDBProperties(IDBProperties *iface)
     return CONTAINING_RECORD(iface, struct msdasql, IDBProperties_iface);
 }
 
+static inline struct msdasql *impl_from_IDBInitialize(IDBInitialize *iface)
+{
+    return CONTAINING_RECORD(iface, struct msdasql, IDBInitialize_iface);
+}
+
 static HRESULT WINAPI msdsql_QueryInterface(IUnknown *iface, REFIID riid, void **out)
 {
     struct msdasql *provider = impl_from_IUnknown(iface);
@@ -276,6 +281,10 @@ static HRESULT WINAPI msdsql_QueryInterface(IUnknown *iface, REFIID riid, void *
     {
         *out = &provider->IDBProperties_iface;
     }
+    else if ( IsEqualGUID(riid, &IID_IDBInitialize))
+    {
+        *out = &provider->IDBInitialize_iface;
+    }
     else
     {
         FIXME("(%s, %p)\n", debugstr_guid(riid), out);
@@ -449,6 +458,55 @@ static const struct IDBPropertiesVtbl dbprops_vtbl =
     dbprops_SetProperties
 };
 
+static HRESULT WINAPI dbinit_QueryInterface(IDBInitialize *iface, REFIID riid, void **ppvObject)
+{
+    struct msdasql *provider = impl_from_IDBInitialize(iface);
+
+    return IUnknown_QueryInterface(&provider->MSDASQL_iface, riid, ppvObject);
+}
+
+static ULONG WINAPI dbinit_AddRef(IDBInitialize *iface)
+{
+    struct msdasql *provider = impl_from_IDBInitialize(iface);
+
+    return IUnknown_AddRef(&provider->MSDASQL_iface);
+}
+
+static ULONG WINAPI dbinit_Release(IDBInitialize *iface)
+{
+    struct msdasql *provider = impl_from_IDBInitialize(iface);
+
+    return IUnknown_Release(&provider->MSDASQL_iface);
+}
+
+static HRESULT WINAPI dbinit_Initialize(IDBInitialize *iface)
+{
+    struct msdasql *provider = impl_from_IDBInitialize(iface);
+
+    FIXME("%p stub\n", provider);
+
+    return S_OK;
+}
+
+static HRESULT WINAPI dbinit_Uninitialize(IDBInitialize *iface)
+{
+    struct msdasql *provider = impl_from_IDBInitialize(iface);
+
+    FIXME("%p stub\n", provider);
+
+    return S_OK;
+}
+
+
+static const struct IDBInitializeVtbl dbinit_vtbl =
+{
+    dbinit_QueryInterface,
+    dbinit_AddRef,
+    dbinit_Release,
+    dbinit_Initialize,
+    dbinit_Uninitialize
+};
+
 static HRESULT create_msdasql_provider(REFIID riid, void **ppv)
 {
     struct msdasql *provider;
@@ -460,6 +518,7 @@ static HRESULT create_msdasql_provider(REFIID riid, void **ppv)
 
     provider->MSDASQL_iface.lpVtbl = &msdsql_vtbl;
     provider->IDBProperties_iface.lpVtbl = &dbprops_vtbl;
+    provider->IDBInitialize_iface.lpVtbl = &dbinit_vtbl;
     provider->ref = 1;
 
     hr = IUnknown_QueryInterface(&provider->MSDASQL_iface, riid, ppv);
-- 
2.26.2

