From f63b1e647456dac484d97bacd3e6466136ba4aad Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 30 May 2020 15:33:18 +1000
Subject: [PATCH 03/12] msdasql: Fake return from IDBProperties SetProperties

---
 dlls/msdasql/msdasql_main.c | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/dlls/msdasql/msdasql_main.c b/dlls/msdasql/msdasql_main.c
index c1929192b65..7fd30721c4a 100644
--- a/dlls/msdasql/msdasql_main.c
+++ b/dlls/msdasql/msdasql_main.c
@@ -441,11 +441,33 @@ static HRESULT WINAPI dbprops_GetPropertyInfo(IDBProperties *iface, ULONG cPrope
 static HRESULT WINAPI dbprops_SetProperties(IDBProperties *iface, ULONG cPropertySets,
             DBPROPSET rgPropertySets[])
 {
+    int i;
     struct msdasql *provider = impl_from_IDBProperties(iface);
 
     FIXME("(%p)->(%d %p)\n", provider, cPropertySets, rgPropertySets);
 
-    return E_NOTIMPL;
+    FIXME("count %d guidPropertySet: %s\n", rgPropertySets->cProperties, debugstr_guid(&rgPropertySets->guidPropertySet));
+    for (i=0; i < rgPropertySets->cProperties; i++)
+    {
+        FIXME("Property Set %d\n", i);
+        FIXME("  dwPropertyID: %x\n", rgPropertySets->rgProperties[i].dwPropertyID);
+        if (V_VT(&rgPropertySets->rgProperties[i].vValue) == VT_BSTR)
+            FIXME("  vValue: %s\n", debugstr_w(V_BSTR(&rgPropertySets->rgProperties[i].vValue)));
+        else if (V_VT(&rgPropertySets->rgProperties[i].vValue) == VT_BOOL)
+            FIXME("  vValue: %d\n", V_BOOL(&rgPropertySets->rgProperties[i].vValue));
+        else
+            FIXME("  vValue: %d\n", V_I4(&rgPropertySets->rgProperties[i].vValue));
+        //FIXME("  cPropertyIDs:  %d\n", rgPropertySets->rgProperties[i].cPropertyIDs);
+        //FIXME("  guidPropertySet: %s\n", rgPropertySets(&rgPropertyIDSets[i].guidPropertySet) );
+    }
+
+    // DBPROP_AUTH_PERSIST_SENSITIVE_AUTHINFO  (0x0b)
+    // DBPROP_INIT_DATASOURCE (0x3b)
+
+    // DBPROP_INIT_TIMEOUT (0x42)
+    // DBPROP_INIT_OLEDBSERVICES (0xf8)
+
+    return S_OK;
 }
 
 static const struct IDBPropertiesVtbl dbprops_vtbl =
-- 
2.28.0

