From bc8bb574f615e46309d1fdeebec5268ab10b7111 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 18 Nov 2021 10:07:11 +1100
Subject: [PATCH 12/13] msdasql: Skip DBTYPE_IUNKNOWN columns

---
 dlls/msdasql/session.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 6386a8ce9bb..840e506e8d8 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -813,6 +813,31 @@ static HRESULT WINAPI msdasql_rowset_GetData(IRowset *iface, HROW row, HACCESSOR
             FIXME("DBOBJECT 0x%08x, %s currently not supported\n", rowset->bindings[i].pObject->dwFlags,
                     debugstr_guid(&rowset->bindings[i].pObject->iid));
 
+        if (rowset->bindings[i].wType == DBTYPE_IUNKNOWN)
+        {
+            IStream *steam;
+            //ISequentialStream* pISequentialStream = NULL;
+            IUnknown *pIUnknown = NULL; //*((IUnknown**)(pSrc);
+            //LONG_PTR val;
+
+            hr = CreateStreamOnHGlobal(NULL, TRUE, &steam);
+            FIXME("CreateStreamOnHGlobal 0x%08x\n", hr);
+
+            hr = IStream_QueryInterface(steam, &IID_IUnknown, (void**)&pIUnknown);
+            //*(IUnknown **)pSrc = &pIUnknown;
+
+            *(IUnknown**)(pSrc) = *(IUnknown**)&pIUnknown;
+            //hr = IStream_QueryInterface(steam, &IID_IUnknown, &pSrc);
+
+            FIXME("IStream_QueryInterface 0x%08x (%x, %p)\n", hr, *(LONG_PTR*)(pSrc) ,pIUnknown);
+//FIXME("pIUnknown %p\n", pIUnknown);
+  //          if (pIUnknown)
+    //            hr = IUnknown_QueryInterface(pIUnknown, &IID_ISequentialStream, (void**)&pISequentialStream);
+
+            FIXME("Skipping streaming data %x\n", hr);
+            continue;
+        }
+
         dwPart = rowset->bindings[i].dwPart;
         pDst         = dwPart & DBPART_VALUE ? ((BYTE *)data + rowset->bindings[i].obValue) : NULL;
         pulDstLength = dwPart & DBPART_LENGTH ? (DBLENGTH *) ((BYTE*)data + rowset->bindings[i].obLength) : NULL;
@@ -1207,6 +1232,10 @@ static HRESULT WINAPI rowset_accessor_CreateAccessor(IAccessor *iface, DBACCESSO
             FIXME("DBOBJECT 0x%08x, %s currently not supported\n", bindings[i].pObject->dwFlags,
                     debugstr_guid(&bindings[i].pObject->iid));
 
+        /* Don't bind Columns that will route though an interface, eg Blobs, Large text field */
+        if (bindings[i].wType == DBTYPE_IUNKNOWN)
+            continue;
+
         type = bindtype_to_sqltype(bindings[i].wType);
         ret = SQLBindCol(rowset->hstmt, bindings[i].iOrdinal, type, &ptr[bindings[i].obValue],
                             bindings[i].cbMaxLen, (SQLLEN*)&ptr[bindings[i].obLength]);
-- 
2.33.0

