From d8902f9cf7a22dd2b6da30e522e8f8a227e0beae Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 21 Oct 2020 16:09:36 +1100
Subject: [PATCH] msdasql: Dump OpenRowset properties

---
 dlls/msdasql/session.c | 51 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 51 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index adcd558a28d..212b0373233 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -173,13 +173,64 @@ ULONG WINAPI openrowset_Release(IOpenRowset *iface)
     return IUnknown_Release(&session->session_iface);
 }
 
+static const char *get_property_value(DWORD property)
+{
+    switch(property)
+    {
+#define FMT_TO_STR(property) case property: return #property
+
+        FMT_TO_STR(DBPROP_AUTH_INTEGRATED);
+        FMT_TO_STR(DBPROP_AUTH_MASK_PASSWORD);
+        FMT_TO_STR(DBPROP_BLOCKINGSTORAGEOBJECTS);
+        FMT_TO_STR(DBPROP_CANFETCHBACKWARDS);
+        FMT_TO_STR(DBPROP_CANSCROLLBACKWARDS);
+        FMT_TO_STR(DBPROP_OTHERINSERT);
+        FMT_TO_STR(DBPROP_OWNUPDATEDELETE);
+        FMT_TO_STR(DBPROP_REMOVEDELETED);
+        FMT_TO_STR(DBPROP_UPDATABILITY);
+
+        FMT_TO_STR(DBPROP_IColumnsRowset);
+        FMT_TO_STR(DBPROP_IRowset);
+        FMT_TO_STR(DBPROP_IRowsetCurrentIndex);
+        FMT_TO_STR(DBPROP_IRowsetChange);
+        FMT_TO_STR(DBPROP_IRowsetIndex);
+        FMT_TO_STR(DBPROP_IRowsetUpdate);
+        FMT_TO_STR(DBPROP_ISequentialStream);
+        default:
+            FIXME("Unknown value 0x%08x\n", property);
+            return "Unknown";
+    }
+}
+
 HRESULT WINAPI openrowset_OpenRowset(IOpenRowset *iface, IUnknown *pUnkOuter, DBID *table,
             DBID *index, REFIID riid, ULONG count, DBPROPSET propertysets[], IUnknown **rowset)
 {
     struct msdasql_session *session = impl_from_IOpenRowset( iface );
+    int i, j;
+
     FIXME("%p, %p, %p %p %s, %d %p %p stub\n", session, pUnkOuter, table, index, debugstr_guid(riid),
             count, propertysets, rowset);
 
+    if (table)
+        FIXME("table eKind %d %s %s\n", table->eKind, debugstr_guid(&table->uGuid.guid), debugstr_w(table->uName.pwszName));
+
+    if (index)
+        FIXME("index eKind %d %s %s\n", index->eKind, debugstr_guid(&index->uGuid.guid), debugstr_w(index->uName.pwszName));
+
+    for (j=0; j < count; j++)
+    {
+        FIXME(" Property Set %d (%s)\n", j, debugstr_guid(&propertysets[j].guidPropertySet));
+        for (i=0; i < propertysets[j].cProperties; i++)
+        {
+            FIXME(" Property %d, dwOptions 0x%08x\n", i, propertysets[j].rgProperties[i].dwOptions);
+            FIXME("  dwPropertyID: %s (%d)\n", get_property_value(propertysets[j].rgProperties[i].dwPropertyID),
+                    V_VT(&propertysets[j].rgProperties[i].vValue));
+            FIXME("  vValue: %s\n", debugstr_variant(&propertysets[j].rgProperties[i].vValue));
+            //FIXME("  cPropertyIDs:  %d\n", rgPropertySets->rgProperties[i].cPropertyIDs);
+            //FIXME("  guidPropertySet: %s\n", rgPropertySets(&rgPropertyIDSets[i].guidPropertySet) );
+        }
+    }
+
     return E_NOTIMPL;
 }
 
-- 
2.29.2

