From 153cb78456c0a90d261663a4a1a6a9435f92003d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sun, 25 Oct 2020 16:50:06 +1100
Subject: [PATCH] msdasql: Add ICommandStream to Command

---
 dlls/msdasql/session.c | 55 +++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 54 insertions(+), 1 deletion(-)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index bce2cf125b9..537e9f0ad63 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -317,6 +317,7 @@ static ULONG WINAPI createcommand_Release(IDBCreateCommand *iface)
 struct command
 {
     ICommand ICommand_iface;
+    ICommandStream ICommandStream_iface;
     LONG refs;
 };
 
@@ -325,6 +326,11 @@ static inline struct command *impl_from_ICommand( ICommand *iface )
     return CONTAINING_RECORD( iface, struct command, ICommand_iface );
 }
 
+static inline struct command *impl_from_ICommandStream( ICommandStream *iface )
+{
+    return CONTAINING_RECORD( iface, struct command, ICommandStream_iface );
+}
+
 static HRESULT WINAPI command_QueryInterface(ICommand *iface, REFIID riid, void **ppv)
 {
     struct command *command = impl_from_ICommand( iface );
@@ -336,7 +342,10 @@ static HRESULT WINAPI command_QueryInterface(ICommand *iface, REFIID riid, void
        IsEqualGUID(&IID_ICommand, riid)) {
         *ppv = &command->ICommand_iface;
     }
-    // ICommandStream must.
+    else if(IsEqualGUID(&IID_ICommandStream, riid))
+    {
+        *ppv = &command->ICommandStream_iface;
+    }
 
     if(*ppv) {
         IUnknown_AddRef((IUnknown*)*ppv);
@@ -401,6 +410,49 @@ static const ICommandVtbl commandVtbl = {
     command_GetDBSession
 };
 
+static HRESULT WINAPI commandstream_QueryInterface(ICommandStream *iface, REFIID riid, void **out)
+{
+    struct command *command = impl_from_ICommandStream( iface );
+    return ICommand_QueryInterface(&command->ICommand_iface, riid, out);
+}
+
+static ULONG WINAPI commandstream_AddRef(ICommandStream *iface)
+{
+    struct command *command = impl_from_ICommandStream( iface );
+    return ICommand_AddRef(&command->ICommand_iface);
+}
+
+static ULONG WINAPI commandstream_Release(ICommandStream *iface)
+{
+    struct command *command = impl_from_ICommandStream( iface );
+    return ICommand_Release(&command->ICommand_iface);
+}
+
+static HRESULT WINAPI commandstream_GetCommandStream(ICommandStream *iface, IID *iid,
+        GUID *dialect, IUnknown **commandstream)
+{
+    struct command *command = impl_from_ICommandStream( iface );
+    FIXME("%p, %p, %p, %p\n", command, iid, dialect, commandstream);
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI commandstream_SetCommandStream(ICommandStream *iface, REFIID riid,
+        REFGUID dialect, IUnknown *commandstream)
+{
+    struct command *command = impl_from_ICommandStream( iface );
+    FIXME("%p, %s, %s, %p\n", command, debugstr_guid(riid), debugstr_guid(dialect), commandstream);
+    return E_NOTIMPL;
+}
+
+static const ICommandStreamVtbl commandStreamVtbl =
+{
+    commandstream_QueryInterface,
+    commandstream_AddRef,
+    commandstream_Release,
+    commandstream_GetCommandStream,
+    commandstream_SetCommandStream
+};
+
 static HRESULT WINAPI createcommand_CreateCommand(IDBCreateCommand *iface, IUnknown *outer, REFIID riid,
                                             IUnknown **out)
 {
@@ -415,6 +467,7 @@ static HRESULT WINAPI createcommand_CreateCommand(IDBCreateCommand *iface, IUnkn
         return E_OUTOFMEMORY;
 
     command->ICommand_iface.lpVtbl = &commandVtbl;
+    command->ICommandStream_iface.lpVtbl = &commandStreamVtbl;
     command->refs = 1;
 
     hr = ICommand_QueryInterface(&command->ICommand_iface, riid, (void**)out);
-- 
2.28.0

