From a4841a147fcbcff2781ccde9277eaab0256d18bf Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 27 Oct 2020 14:18:01 +1100
Subject: [PATCH] msdasql: Add ICommandProperties to session

---
 dlls/msdasql/session.c | 53 +++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 52 insertions(+), 1 deletion(-)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 35a43634ef7..07fb1b5bb92 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -317,6 +317,7 @@ static ULONG WINAPI createcommand_Release(IDBCreateCommand *iface)
 struct command
 {
     ICommandText ICommandText_iface;
+    ICommandProperties ICommandProperties_iface;
     LONG refs;
 };
 
@@ -325,6 +326,11 @@ static inline struct command *impl_from_ICommandText( ICommandText *iface )
     return CONTAINING_RECORD( iface, struct command, ICommandText_iface );
 }
 
+static inline struct command *impl_from_ICommandProperties( ICommandProperties *iface )
+{
+    return CONTAINING_RECORD( iface, struct command, ICommandProperties_iface );
+}
+
 static HRESULT WINAPI command_QueryInterface(ICommandText *iface, REFIID riid, void **ppv)
 {
     struct command *command = impl_from_ICommandText( iface );
@@ -337,7 +343,8 @@ static HRESULT WINAPI command_QueryInterface(ICommandText *iface, REFIID riid, v
        IsEqualGUID(&IID_ICommandText, riid)) {
         *ppv = &command->ICommandText_iface;
     }
-    // ICommandStream must.
+    else if(IsEqualGUID(&IID_ICommandProperties, riid))
+         *ppv = &command->ICommandProperties_iface;
 
     if(*ppv) {
         IUnknown_AddRef((IUnknown*)*ppv);
@@ -418,6 +425,49 @@ static const ICommandTextVtbl commandVtbl = {
     command_SetCommandText
 };
 
+static HRESULT WINAPI command_prop_QueryInterface(ICommandProperties *iface, REFIID riid, void **out)
+{
+    struct command *command = impl_from_ICommandProperties( iface );
+    return ICommandText_QueryInterface(&command->ICommandText_iface, riid, out);
+}
+
+static ULONG WINAPI command_prop_AddRef(ICommandProperties *iface)
+{
+    struct command *command = impl_from_ICommandProperties( iface );
+    return ICommandText_AddRef(&command->ICommandText_iface);
+}
+
+static ULONG WINAPI command_prop_Release(ICommandProperties *iface)
+{
+    struct command *command = impl_from_ICommandProperties( iface );
+    return ICommandText_Release(&command->ICommandText_iface);
+}
+
+static HRESULT WINAPI command_prop_GetProperties(ICommandProperties *iface, ULONG count,
+        const DBPROPIDSET propertyidsets[], ULONG *sets_cnt, DBPROPSET **propertyset)
+{
+    struct command *command = impl_from_ICommandProperties( iface );
+    FIXME("%p %d %p %p %p\n", command, count, propertyidsets, sets_cnt, propertyset);
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI command_prop_SetProperties(ICommandProperties *iface, ULONG count,
+        DBPROPSET propertyset[])
+{
+    struct command *command = impl_from_ICommandProperties( iface );
+    FIXME("%p %p\n", command, propertyset);
+    return S_OK;
+}
+
+static const ICommandPropertiesVtbl commonpropsVtbl =
+{
+    command_prop_QueryInterface,
+    command_prop_AddRef,
+    command_prop_Release,
+    command_prop_GetProperties,
+    command_prop_SetProperties
+};
+
 static HRESULT WINAPI createcommand_CreateCommand(IDBCreateCommand *iface, IUnknown *outer, REFIID riid,
                                             IUnknown **out)
 {
@@ -432,6 +482,7 @@ static HRESULT WINAPI createcommand_CreateCommand(IDBCreateCommand *iface, IUnkn
         return E_OUTOFMEMORY;
 
     command->ICommandText_iface.lpVtbl = &commandVtbl;
+    command->ICommandProperties_iface.lpVtbl = &commonpropsVtbl;
     command->refs = 1;
 
     hr = ICommandText_QueryInterface(&command->ICommandText_iface, riid, (void**)out);
-- 
2.29.2

