From b57fca5b7be2b91bd0cfdb3aa263fda219dee575 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 24 Nov 2023 15:48:36 +1100
Subject: [PATCH 5/5] include: Add more DBCOLUMN_ defines

---
 include/dbs.idl | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/dbs.idl b/include/dbs.idl
index f72204b1e90..1850ab19439 100644
--- a/include/dbs.idl
+++ b/include/dbs.idl
@@ -543,8 +543,19 @@ cpp_quote("DEFINE_DBGUID(DBPROPSET_DBINITALL,         0xc8b522ca, 0x5cf3, 0x11ce
 cpp_quote("DEFINE_DBGUID(DBPROPSET_PROPERTIESINERROR, 0xc8b522d4, 0x5cf3, 0x11ce, 0xad, 0xe5, 0x00, 0xaa, 0x00, 0x44, 0x77, 0x3d);")
 
 cpp_quote("DEFINE_DBID(DB_NULLID,               DB_NULLGUID, 0,                   0);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_IDNAME,         DBCIDGUID,   DBKIND_GUID_PROPID,  2);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_NAME,           DBCIDGUID,   DBKIND_GUID_PROPID,  3);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_NUMBER,         DBCIDGUID,   DBKIND_GUID_PROPID,  4);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_TYPE,           DBCIDGUID,   DBKIND_GUID_PROPID,  5);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_PRECISION,      DBCIDGUID,   DBKIND_GUID_PROPID,  7);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_SCALE,          DBCIDGUID,   DBKIND_GUID_PROPID,  8);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_FLAGS,          DBCIDGUID,   DBKIND_GUID_PROPID,  9);")
 cpp_quote("DEFINE_DBID(DBCOLUMN_BASECOLUMNNAME, DBCIDGUID,   DBKIND_GUID_PROPID, 10);")
 cpp_quote("DEFINE_DBID(DBCOLUMN_BASETABLENAME,  DBCIDGUID,   DBKIND_GUID_PROPID, 11);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_GUID,           DBCIDGUID,   DBKIND_GUID_PROPID, 29);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_PROPID,         DBCIDGUID,   DBKIND_GUID_PROPID, 30);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_TYPEINFO,       DBCIDGUID,   DBKIND_GUID_PROPID, 31);")
+cpp_quote("DEFINE_DBID(DBCOLUMN_COLUMNSIZE,     DBCIDGUID,   DBKIND_GUID_PROPID, 37);")
 cpp_quote("DEFINE_DBID(DBCOLUMN_KEYCOLUMN,      DBCIDGUID,   DBKIND_GUID_PROPID, 41);")
 
 cpp_quote("#define DBPROMPT_PROMPT              0x01")
-- 
2.42.0

