From 5781ae6159ed4f8c3464000a8c9d2494671c20a2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 7 Dec 2023 07:24:46 +1100
Subject: [PATCH] msdasql: Support IAccessor for metadata rowset

---
 dlls/msdasql/session.c | 71 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 71 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 5d77232db3b..fe811dd7396 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -1417,6 +1417,7 @@ struct metadata_rowset
 {
     IRowset IRowset_iface;
     IColumnsInfo IColumnsInfo_iface;
+    IAccessor IAccessor_iface;
     LONG refs;
 
     DBORDINAL dbid_cnt;
@@ -1433,6 +1434,11 @@ static inline struct metadata_rowset *meta_impl_from_IColumnsInfo( IColumnsInfo
     return CONTAINING_RECORD( iface, struct metadata_rowset, IColumnsInfo_iface );
 }
 
+static inline struct metadata_rowset *meta_impl_from_IAccessor ( IAccessor *iface )
+{
+    return CONTAINING_RECORD( iface, struct metadata_rowset, IAccessor_iface );
+}
+
 static HRESULT WINAPI metadata_rowset_QueryInterface(IRowset *iface, REFIID riid,  void **ppv)
 {
     struct metadata_rowset *rowset = meta_impl_from_IRowset( iface );
@@ -1449,6 +1455,10 @@ static HRESULT WINAPI metadata_rowset_QueryInterface(IRowset *iface, REFIID riid
     {
         *ppv = &rowset->IColumnsInfo_iface;
     }
+    else if(IsEqualGUID(&IID_IAccessor, riid))
+    {
+        *ppv = &rowset->IAccessor_iface;
+    }
 
     if(*ppv)
     {
@@ -1682,6 +1692,66 @@ static struct IColumnsInfoVtbl meta_columninfo_vtbll =
     meta_colsinfo_MapColumnIDs
 };
 
+static HRESULT WINAPI metadata_rowset_accessor_QueryInterface(IAccessor *iface, REFIID riid, void **out)
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    return IRowset_QueryInterface(&rowset->IRowset_iface, riid, out);
+}
+
+static ULONG WINAPI metadata_rowset_accessor_AddRef(IAccessor *iface)
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    return IRowset_AddRef(&rowset->IRowset_iface);
+}
+
+static ULONG  WINAPI metadata_rowset_accessor_Release(IAccessor *iface)
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    return IRowset_Release(&rowset->IRowset_iface);
+}
+
+static HRESULT WINAPI metadata_rowset_accessor_AddRefAccessor(IAccessor *iface, HACCESSOR accessor, DBREFCOUNT *count)
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    FIXME("%p, %Id, %p\n", rowset, accessor, count);
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI metadata_rowset_accessor_CreateAccessor(IAccessor *iface, DBACCESSORFLAGS flags,
+        DBCOUNTITEM count, const DBBINDING bindings[], DBLENGTH row_size, HACCESSOR *accessor,
+        DBBINDSTATUS status[])
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    FIXME("%p 0x%08lx, %lu, %p, %lu, %p, %p\n", rowset, flags, count, bindings, row_size, accessor, status);
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI metadata_rowset_accessor_GetBindings(IAccessor *iface, HACCESSOR accessor,
+        DBACCESSORFLAGS *flags, DBCOUNTITEM *count, DBBINDING **bindings)
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    FIXME("%p, %Id, %p, %p, %p\n", rowset, accessor, flags, count, bindings);
+    return E_NOTIMPL;
+}
+
+static HRESULT WINAPI metadata_rowset_accessor_ReleaseAccessor(IAccessor *iface, HACCESSOR accessor, DBREFCOUNT *count)
+{
+    struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
+    FIXME("%p, %Id, %p\n", rowset, accessor, count);
+    return E_NOTIMPL;
+}
+
+struct IAccessorVtbl meta_accessor_vtbl =
+{
+    metadata_rowset_accessor_QueryInterface,
+    metadata_rowset_accessor_AddRef,
+    metadata_rowset_accessor_Release,
+    metadata_rowset_accessor_AddRefAccessor,
+    metadata_rowset_accessor_CreateAccessor,
+    metadata_rowset_accessor_GetBindings,
+    metadata_rowset_accessor_ReleaseAccessor
+};
+
 static HRESULT WINAPI column_rs_GetColumnsRowset(IColumnsRowset *iface, IUnknown *outer, DBORDINAL count,
         const DBID columns[], REFIID riid, ULONG property_cnt, DBPROPSET property_sets[], IUnknown **unk_rs)
 {
@@ -1710,6 +1780,7 @@ static HRESULT WINAPI column_rs_GetColumnsRowset(IColumnsRowset *iface, IUnknown
 
     metadata_rs->IRowset_iface.lpVtbl = &metadata_rowset_vtbl;
     metadata_rs->IColumnsInfo_iface.lpVtbl = &meta_columninfo_vtbll;
+    metadata_rs->IAccessor_iface.lpVtbl = &meta_accessor_vtbl;
     metadata_rs->refs = 1;
     metadata_rs->dbid_cnt = count;
     if (count)
-- 
2.43.0

