From 12ccf31adf064b81a7214cf04c2689d3fa08a4cf Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 7 Dec 2023 07:50:34 +1100
Subject: [PATCH] msdasql: HACK - Fake success from CreateAccessor metadata

---
 dlls/msdasql/session.c | 22 ++++++++++++++++++++--
 1 file changed, 20 insertions(+), 2 deletions(-)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index fe811dd7396..9289f4694ac 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -1721,9 +1721,27 @@ static HRESULT WINAPI metadata_rowset_accessor_CreateAccessor(IAccessor *iface,
         DBCOUNTITEM count, const DBBINDING bindings[], DBLENGTH row_size, HACCESSOR *accessor,
         DBBINDSTATUS status[])
 {
+    struct msdasql_accessor *access;
+    int i;
+
     struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
     FIXME("%p 0x%08lx, %lu, %p, %lu, %p, %p\n", rowset, flags, count, bindings, row_size, accessor, status);
-    return E_NOTIMPL;
+
+
+    access = malloc( sizeof(struct msdasql_accessor));
+    access->size = 0;
+    //access->data = malloc(4096);
+    if (status)
+    {
+        for(i=0; i < count; i++)
+        {
+            status[i] = DBBINDSTATUS_OK;
+        }
+    }
+
+    *accessor = (HACCESSOR)access;
+
+    return S_OK;
 }
 
 static HRESULT WINAPI metadata_rowset_accessor_GetBindings(IAccessor *iface, HACCESSOR accessor,
@@ -1738,7 +1756,7 @@ static HRESULT WINAPI metadata_rowset_accessor_ReleaseAccessor(IAccessor *iface,
 {
     struct metadata_rowset *rowset = meta_impl_from_IAccessor( iface );
     FIXME("%p, %Id, %p\n", rowset, accessor, count);
-    return E_NOTIMPL;
+    return S_OK;
 }
 
 struct IAccessorVtbl meta_accessor_vtbl =
-- 
2.43.0

