From 7bf2b50e63b17c59b3e1d63037ce22b9c4cdc1a9 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 23 Nov 2020 15:45:17 +1100
Subject: [PATCH 25/42] msdasql: Explicity list interfaces queried

---
 dlls/msdasql/session.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 82637678fcf..626a5cd1a4b 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -417,6 +417,21 @@ static HRESULT WINAPI msdasql_rowset_QueryInterface(IRowset *iface, REFIID riid,
     {
         *ppv = &rowset->IRowsetInfo_iface;
     }
+    else if (IsEqualGUID(&IID_IRowsetChange, riid))
+    {
+        TRACE("IID_IRowsetChange not supported\n");
+        return E_NOINTERFACE;
+    }
+    else if (IsEqualGUID(&IID_IRowsetUpdate, riid))
+    {
+        TRACE("IID_IRowsetUpdate not supported\n");
+        return E_NOINTERFACE;
+    }
+    else if (IsEqualGUID(&IID_IRowsetLocate, riid))
+    {
+        TRACE("IID_IRowsetLocate not supported\n");
+        return E_NOINTERFACE;
+    }
 
     if(*ppv) {
         IUnknown_AddRef((IUnknown*)*ppv);
-- 
2.30.2

