From ef72e7e0db7cb7c0c8b4959d336ab39449a6b485 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 23 Nov 2020 15:45:17 +1100
Subject: [PATCH] msdasql: Explicity list interfaces queried

---
 dlls/msdasql/session.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 45ceaa6b4a1..a98c3c23709 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -417,6 +417,21 @@ static HRESULT WINAPI msdasql_rowset_QueryInterface(IRowset *iface, REFIID riid,
     {
         *ppv = &rowset->IRowsetInfo_iface;
     }
+    else if (IsEqualGUID(&IID_IRowsetChange, riid))
+    {
+        FIXME("Asking for IID_IRowsetChange\n");
+        return E_NOINTERFACE;
+    }
+    else if (IsEqualGUID(&IID_IRowsetUpdate, riid))
+    {
+        FIXME("Asking for IID_IRowsetUpdaten\n");
+        return E_NOINTERFACE;
+    }
+    else if (IsEqualGUID(&IID_IRowsetLocate, riid))
+    {
+        FIXME("Asking for IID_IRowsetLocate\n");
+        return E_NOINTERFACE;
+    }
 
     if(*ppv) {
         IUnknown_AddRef((IUnknown*)*ppv);
-- 
2.29.2

