From 44a4bc605f9eccb378018e08546e710438ab3769 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 23 Dec 2020 10:27:24 +1100
Subject: [PATCH 31/42] msdasql: Explicit trace unsupported interfaces

---
 dlls/msdasql/session.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 6013f1e09fb..57050d185f1 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -370,6 +370,16 @@ static HRESULT WINAPI command_QueryInterface(ICommandText *iface, REFIID riid, v
          *ppv = &command->IConvertType_iface;
     else if(IsEqualGUID(&IID_ICommandPrepare, riid))
          *ppv = &command->ICommandPrepare_iface;
+    else if(IsEqualGUID(&IID_ICommandStream, riid))
+    {
+        TRACE("ICommandStream not support\n");
+        return E_NOINTERFACE;
+    }
+    else if(IsEqualGUID(&IID_IMultipleResults, riid))
+    {
+        TRACE("IMultipleResults not support\n");
+        return E_NOINTERFACE;
+    }
 
     if(*ppv) {
         IUnknown_AddRef((IUnknown*)*ppv);
-- 
2.30.2

