From e3ecc6179a3777970bee111ac84c367a2788e090 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 30 Dec 2020 19:17:14 +1100
Subject: [PATCH] include: Add DBACCESSORFLAGSENUM/DBBINDSTATUSENUM values

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/access.idl | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/include/access.idl b/include/access.idl
index 5ade09b85ce..fe1f3baddd7 100644
--- a/include/access.idl
+++ b/include/access.idl
@@ -32,6 +32,27 @@ interface IAccessor : IUnknown
 
     typedef DWORD DBBINDSTATUS;
 
+    enum DBACCESSORFLAGSENUM
+    {
+        DBACCESSOR_INVALID       = 0x00,
+        DBACCESSOR_PASSBYREF     = 0x01,
+        DBACCESSOR_ROWDATA       = 0x02,
+        DBACCESSOR_PARAMETERDATA = 0x04,
+        DBACCESSOR_OPTIMIZED     = 0x08,
+        DBACCESSOR_INHERITED     = 0x10,
+    };
+
+    enum DBBINDSTATUSENUM
+    {
+        DBBINDSTATUS_OK                    = 0,
+        DBBINDSTATUS_BADORDINAL            = 1,
+        DBBINDSTATUS_UNSUPPORTEDCONVERSION = 2,
+        DBBINDSTATUS_BADBINDINFO           = 3,
+        DBBINDSTATUS_BADSTORAGEFLAGS       = 4,
+        DBBINDSTATUS_NOINTERFACE           = 5,
+        DBBINDSTATUS_MULTIPLESTORAGE       = 6,
+    };
+
     [local]
     HRESULT AddRefAccessor([in] HACCESSOR hAccessor,
                            [in, out, unique, annotation("__out_opt")] DBREFCOUNT *pcRefCount);
-- 
2.29.2

