From 0f18127dde28766783ab29be3c8584e23ab3461e Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 11 Mar 2021 12:00:25 +1100
Subject: [PATCH 39/42] msdasql: Implement IRowsetInfo GetSpecification

---
 dlls/msdasql/session.c | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index ff67c324579..26ac9d4afea 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -456,6 +456,7 @@ struct msdasql_rowset
     IColumnsInfo IColumnsInfo_iface;
     IAccessor IAccessor_iface;
     IColumnsRowset IColumnsRowset_iface;
+    ICommandText *command_text;
     LONG refs;
     DBBINDING *bindings;
     DBCOUNTITEM bind_cnt;
@@ -629,6 +630,8 @@ static ULONG WINAPI msdasql_rowset_Release(IRowset *iface)
     if (!refs)
     {
         TRACE( "destroying %p\n", rowset );
+
+        ICommandText_Release(rowset->command_text);
         heap_free(rowset->bindings);
         heap_free( rowset );
     }
@@ -848,8 +851,19 @@ static HRESULT WINAPI rowset_info_GetSpecification(IRowsetInfo *iface, REFIID ri
         IUnknown **specification)
 {
     struct msdasql_rowset *rowset = impl_from_IRowsetInfo( iface );
-    FIXME("%p, %s, %p\n", rowset, debugstr_guid(riid), specification);
-    return E_NOTIMPL;
+
+    TRACE("%p, %s, %p\n", rowset, debugstr_guid(riid), specification);
+
+    if (IsEqualGUID(&IID_ICommand, riid))
+        ICommandText_QueryInterface(rowset->command_text, &IID_IUnknown, (void**)specification);
+
+    if(*specification)
+    {
+        return S_OK;
+    }
+
+    FIXME("(%p)->(%s %p)\n", rowset, debugstr_guid(riid), specification);
+    return E_NOINTERFACE;
 }
 
 struct IRowsetInfoVtbl rowset_info_vtbl =
@@ -1204,6 +1218,9 @@ static HRESULT WINAPI command_Execute(ICommandText *iface, IUnknown *outer, REFI
     msrowset->bind_cnt = 0;
     msrowset->hstmt = hstmt;
 
+    msrowset->command_text = iface;
+    ICommandText_AddRef(msrowset->command_text);
+
     if (affected) *affected = results;
 
     hr = IRowset_QueryInterface(&msrowset->IRowset_iface, &IID_IUnknown, (void**)rowset);
-- 
2.30.2

