From f0d9517ffc978140ad78043692eea1a067befa8c Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 26 May 2020 16:47:25 +1000
Subject: Great Hack


diff --git a/dlls/oledb32/dslocator.c b/dlls/oledb32/dslocator.c
index c74fb5133d2..2f91dbcb8f8 100644
--- a/dlls/oledb32/dslocator.c
+++ b/dlls/oledb32/dslocator.c
@@ -39,6 +39,9 @@
 #include "wine/debug.h"
 #include "wine/heap.h"
 
+#include "initguid.h"
+#include "msado15_backcompat.h"
+
 WINE_DEFAULT_DEBUG_CHANNEL(oledb);
 
 struct datasource
@@ -624,6 +627,7 @@ static HRESULT WINAPI dslocator_PromptNew(IDataSourceLocator *iface, IDispatch *
     PROPSHEETHEADERW hdr;
     PROPSHEETPAGEW pages[4];
     INT_PTR ret;
+    HRESULT hr;
 
     FIXME("(%p, %p) Semi-stub\n", iface, connection);
 
@@ -663,8 +667,21 @@ static HRESULT WINAPI dslocator_PromptNew(IDataSourceLocator *iface, IDispatch *
     hdr.u3.ppsp = pages;
     hdr.nPages = ARRAY_SIZE(pages);
     ret = PropertySheetW(&hdr);
+    FIXME("PropertySheetW return %ld\n", ret);
+    if (ret == -1)
+        return S_FALSE;
+
+    hr = CoCreateInstance( &CLSID_Connection, NULL, CLSCTX_INPROC_SERVER, &IID__Connection, (void**)connection );
+    if (hr == S_OK)
+    {
+        BSTR str = SysAllocString(L"Provider=MSDASQL.1;Persist Security Info=False;Data "
+                              "Source=wine_test");
+
+        ret = _Connection_put_ConnectionString( (_Connection*)*connection, str);
+        SysFreeString(str);
+    }
 
-    return ret ? S_OK : S_FALSE;
+    return hr == S_OK ? S_OK : S_FALSE;
 }
 
 static HRESULT WINAPI dslocator_PromptEdit(IDataSourceLocator *iface, IDispatch **ppADOConnection, VARIANT_BOOL *success)
