From 29c33d675e02fe91d679ae3658f5c6f737f83027 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 16 Jul 2016 18:12:12 +1000
Subject: [PATCH] xaudio: Stop nbufs becoming out of range.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/xaudio2_7/xaudio_dll.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/dlls/xaudio2_7/xaudio_dll.c b/dlls/xaudio2_7/xaudio_dll.c
index b2a5e3b..8e46611 100644
--- a/dlls/xaudio2_7/xaudio_dll.c
+++ b/dlls/xaudio2_7/xaudio_dll.c
@@ -2273,7 +2273,10 @@ static void update_source_state(XA2SourceImpl *src)
 
                 src->first_buf++;
                 src->first_buf %= XAUDIO2_MAX_QUEUED_BUFFERS;
-                src->nbufs--;
+                if(src->nbufs)
+                    src->nbufs--;
+                else
+                    WARN("Leaving src->nbufs at 0\n");
 
                 TRACE("%p: done with buffer %u\n", src, old_buf);
 
-- 
2.8.1

