From 939b1a832aa4955fb6184eeb2bad263046a20f65 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 26 Jan 2016 18:00:38 +1100
Subject: [PATCH] include: Add missed D3D11_FORMAT_SUPPORT enum values

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/d3d11.idl | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/d3d11.idl b/include/d3d11.idl
index df6ea08..4b78b1f 100644
--- a/include/d3d11.idl
+++ b/include/d3d11.idl
@@ -1016,7 +1016,11 @@ typedef enum D3D11_FORMAT_SUPPORT
     D3D11_FORMAT_SUPPORT_SHADER_GATHER                = 0x00800000,
     D3D11_FORMAT_SUPPORT_BACK_BUFFER_CAST             = 0x01000000,
     D3D11_FORMAT_SUPPORT_TYPED_UNORDERED_ACCESS_VIEW  = 0x02000000,
-    D3D11_FORMAT_SUPPORT_SHADER_GATHER_COMPARISON     = 0x04000000
+    D3D11_FORMAT_SUPPORT_SHADER_GATHER_COMPARISON     = 0x04000000,
+    D3D11_FORMAT_SUPPORT_DECODER_OUTPUT               = 0x08000000,
+    D3D11_FORMAT_SUPPORT_VIDEO_PROCESSOR_OUTPUT       = 0x10000000,
+    D3D11_FORMAT_SUPPORT_VIDEO_PROCESSOR_INPUT        = 0x20000000,
+    D3D11_FORMAT_SUPPORT_VIDEO_ENCODER                = 0x40000000
 } D3D11_FORMAT_SUPPORT;
 
 typedef enum D3D11_CLEAR_FLAG
-- 
2.7.0.rc3

