From 03116e6e89ff5dbcaf43d3978c9efa6d6a232ffc Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 13 Jul 2016 09:14:14 +1000
Subject: [PATCH 1/4] include/icftypes.idl: Add more typedefs

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/icftypes.idl | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/include/icftypes.idl b/include/icftypes.idl
index 9cc0b6a..4b44fde 100644
--- a/include/icftypes.idl
+++ b/include/icftypes.idl
@@ -29,6 +29,14 @@ typedef enum NET_FW_PROFILE_TYPE_
     NET_FW_PROFILE_TYPE_MAX
 }   NET_FW_PROFILE_TYPE;
 
+typedef enum NET_FW_PROFILE_TYPE2_
+{
+   NET_FW_PROFILE2_DOMAIN   = 0x0001,
+   NET_FW_PROFILE2_PRIVATE  = 0x0002,
+   NET_FW_PROFILE2_PUBLIC   = 0x0004,
+   NET_FW_PROFILE2_ALL      = 0x7FFFFFFF
+} NET_FW_PROFILE_TYPE2;
+
 typedef enum NET_FW_IP_VERSION_
 {
     NET_FW_IP_VERSION_V4,
@@ -67,3 +75,24 @@ typedef enum NET_FW_IP_PROTOCOL_
     NET_FW_IP_PROTOCOL_TCP =  6,
     NET_FW_IP_PROTOCOL_UDP = 17
 }   NET_FW_IP_PROTOCOL;
+
+typedef enum NET_FW_RULE_DIRECTION_
+{
+    NET_FW_RULE_DIR_IN = 1,
+    NET_FW_RULE_DIR_OUT,
+    NET_FW_RULE_DIR_MAX
+} NET_FW_RULE_DIRECTION;
+
+typedef enum NET_FW_ACTION_
+{
+    NET_FW_ACTION_BLOCK,
+    NET_FW_ACTION_ALLOW,
+    NET_FW_ACTION_MAX
+} NET_FW_ACTION;
+
+typedef enum NET_FW_MODIFY_STATE_
+{
+    NET_FW_MODIFY_STATE_OK,
+    NET_FW_MODIFY_STATE_GP_OVERRIDE,
+    NET_FW_MODIFY_STATE_INBOUND_BLOCKED
+} NET_FW_MODIFY_STATE;
-- 
1.9.1

