From 8395680ca71790edd0ead4dae77cb6da4d69ab39 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 13 Jul 2016 09:22:14 +1000
Subject: [PATCH 3/4] hnetcfg: Register NetFwPolicy2 interface

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/hnetcfg/hnetcfg.idl | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/dlls/hnetcfg/hnetcfg.idl b/dlls/hnetcfg/hnetcfg.idl
index 9934f6a..763529a 100644
--- a/dlls/hnetcfg/hnetcfg.idl
+++ b/dlls/hnetcfg/hnetcfg.idl
@@ -43,3 +43,11 @@ coclass NetFwAuthorizedApplication { interface INetFwAuthorizedApplications; }
     uuid(0ca545c6-37ad-4a6c-bf92-9f7610067ef5)
 ]
 coclass NetFwOpenPort { interface INetFwOpenPort; }
+
+[
+    helpstring("HNetCfg.FwPolicy2"),
+    progid("HNetCfg.FwPolicy2"),
+    threading(both),
+    uuid(e2b3c97f-6ae1-41ac-817a-f6f92166d7dd)
+]
+coclass NetFwPolicy2 { interface INetFwPolicy2; }
-- 
1.9.1

