From 9786ba09360c93a76e29de0112e127eae9b988ea Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 11 Nov 2019 10:42:23 +1100
Subject: [PATCH] dmime: Free list when Sig trigger track is destroyed.

---
 dlls/dmime/segtriggertrack.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/dlls/dmime/segtriggertrack.c b/dlls/dmime/segtriggertrack.c
index f2e8b5fe52..daf28e204e 100644
--- a/dlls/dmime/segtriggertrack.c
+++ b/dlls/dmime/segtriggertrack.c
@@ -21,6 +21,8 @@
 #include "dmime_private.h"
 #include "dmobject.h"
 
+#include "wine/heap.h"
+
 WINE_DEFAULT_DEBUG_CHANNEL(dmime);
 WINE_DECLARE_DEBUG_CHANNEL(dmfile);
 
@@ -81,7 +83,19 @@ static ULONG WINAPI track_sigtrigger_Release(IDirectMusicTrack8 *iface)
     TRACE("(%p) ref=%d\n", This, ref);
 
     if (!ref) {
-        HeapFree(GetProcessHeap(), 0, This);
+        struct list* cursor, *cursor2;
+        LPDMUS_PRIVATE_SEGMENT_ITEM item;
+
+        LIST_FOR_EACH_SAFE (cursor, cursor2, &This->Items) {
+            item = LIST_ENTRY(cursor, DMUS_PRIVATE_SEGMENT_ITEM, entry);
+            list_remove(cursor);
+
+            if (item->pObject)
+                IDirectMusicObject_Release(item->pObject);
+            heap_free(item);
+        }
+
+        heap_free(This);
         DMIME_UnlockModule();
     }
 
-- 
2.17.1

