From b252ae01ce80419e4f879efee5de106db89581f0 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 19 Jul 2019 12:02:07 +1000
Subject: [PATCH] include: Add COM_DECLSPEC_NOTHROW define

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/basetyps.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/basetyps.h b/include/basetyps.h
index 94875334b0..6e4f9e6d16 100644
--- a/include/basetyps.h
+++ b/include/basetyps.h
@@ -39,6 +39,8 @@
 #define STDMETHODIMPV      HRESULT STDMETHODVCALLTYPE
 #define STDMETHODIMPV_(t)  t STDMETHODVCALLTYPE
 
+#define COM_DECLSPEC_NOTHROW
+
 #if defined(__cplusplus) && !defined(CINTERFACE)
 # define interface struct
 # define STDMETHOD(m) virtual HRESULT STDMETHODCALLTYPE m
-- 
2.17.1

