From 469627fa7f28820b95a6e3aea56adde05829d8c6 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 20 Jul 2019 16:47:29 +1000
Subject: [PATCH] include/ws2ipdef.h: Add define SS_PORT

---
 include/ws2ipdef.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/ws2ipdef.h b/include/ws2ipdef.h
index 52c80c552e..4cae361e5f 100644
--- a/include/ws2ipdef.h
+++ b/include/ws2ipdef.h
@@ -280,6 +280,8 @@ typedef struct WS(in_pktinfo) {
 #define WS_IN6ADDR_LOOPBACK_INIT { 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1 }
 #endif /* USE_WS_PREFIX */
 
+#define SS_PORT(ssp) (((PSOCKADDR_IN)(ssp))->sin_port)
+
 #ifdef __cplusplus
 extern "C" {
 #endif
-- 
2.17.1

