From 79db7d7a4304805a4d79d09efd55b94c4bea2515 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 20 Jul 2019 20:26:34 +1000
Subject: [PATCH] include/winuser.h: Add struct tagMENUGETOBJECTINFO.

---
 include/winuser.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/winuser.h b/include/winuser.h
index 60bc473225..abb6d5b5e9 100644
--- a/include/winuser.h
+++ b/include/winuser.h
@@ -3425,6 +3425,15 @@ typedef struct tagAUDIODESCRIPTION
     LCID Locale;
 } AUDIODESCRIPTION, *LPAUDIODESCRIPTION;
 
+typedef struct tagMENUGETOBJECTINFO
+{
+    DWORD dwFlags;
+    UINT  uPos;
+    HMENU hmenu;
+    void  *riid;
+    void  *pvObj;
+} MENUGETOBJECTINFO, *PMENUGETOBJECTINFO;
+
 #if defined(_WINGDI_) && !defined(NOGDI)
 WINUSERAPI LONG        WINAPI ChangeDisplaySettingsA(LPDEVMODEA,DWORD);
 WINUSERAPI LONG        WINAPI ChangeDisplaySettingsW(LPDEVMODEW,DWORD);
-- 
2.17.1

