From 2ac556a10d52b31e7e06dbd9102a26031a0952b2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 22 Jul 2019 12:05:41 +1000
Subject: [PATCH] include: Add UI_ANIMATION_KEYFRAME_STORYBOARD_START constant

---
 include/uianimation.idl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uianimation.idl b/include/uianimation.idl
index e8b366e8fb..631af11899 100644
--- a/include/uianimation.idl
+++ b/include/uianimation.idl
@@ -30,6 +30,7 @@ interface IUIAnimationTransitionLibrary;
 
 typedef DOUBLE UI_ANIMATION_SECONDS;
 typedef struct { int _; } *UI_ANIMATION_KEYFRAME;
+const UI_ANIMATION_KEYFRAME UI_ANIMATION_KEYFRAME_STORYBOARD_START = (UI_ANIMATION_KEYFRAME)(-1);
 
 const UI_ANIMATION_SECONDS UI_ANIMATION_SECONDS_EVENTUALLY = -1.0;
 
-- 
2.17.1

