From e1b309bb3a7ca82641ac167c2aa28e108ac64543 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 22 Jul 2019 12:13:47 +1000
Subject: [PATCH] include: Add UI_E_* defines

---
 include/winerror.h | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/include/winerror.h b/include/winerror.h
index 32680e097c..6da5c2a7aa 100644
--- a/include/winerror.h
+++ b/include/winerror.h
@@ -3098,6 +3098,33 @@ static inline HRESULT HRESULT_FROM_WIN32(unsigned int x)
 #define DWM_E_ADAPTER_NOT_FOUND                            _HRESULT_TYPEDEF_(0x80263005)
 #define DWM_S_GDI_REDIRECTION_SURFACE                      _HRESULT_TYPEDEF_(0x00263005)
 
+#define UI_E_CREATE_FAILED                                 _HRESULT_TYPEDEF_(0x802a0001)
+#define UI_E_SHUTDOWN_CALLED                               _HRESULT_TYPEDEF_(0x802a0002)
+#define UI_E_ILLEGAL_REENTRANCY                            _HRESULT_TYPEDEF_(0x802a0003)
+#define UI_E_OBJECT_SEALED                                 _HRESULT_TYPEDEF_(0x802a0004)
+#define UI_E_OBJECT_SEALED                                 _HRESULT_TYPEDEF_(0x802a0004)
+#define UI_E_VALUE_NOT_SET                                 _HRESULT_TYPEDEF_(0x802a0005)
+#define UI_E_VALUE_NOT_DETERMINED                          _HRESULT_TYPEDEF_(0x802a0006)
+#define UI_E_INVALID_OUTPUT                                _HRESULT_TYPEDEF_(0x802a0007)
+#define UI_E_BOOLEAN_EXPECTED                              _HRESULT_TYPEDEF_(0x802a0008)
+#define UI_E_DIFFERENT_OWNER                               _HRESULT_TYPEDEF_(0x802a0009)
+#define UI_E_AMBIGUOUS_MATCH                               _HRESULT_TYPEDEF_(0x802a000a)
+#define UI_E_FP_OVERFLOW                                   _HRESULT_TYPEDEF_(0x802a000b)
+#define UI_E_WRONG_THREAD                                  _HRESULT_TYPEDEF_(0x802a000c)
+#define UI_E_STORYBOARD_ACTIVE                             _HRESULT_TYPEDEF_(0x802a0101)
+#define UI_E_STORYBOARD_NOT_PLAYING                        _HRESULT_TYPEDEF_(0x802a0102)
+#define UI_E_START_KEYFRAME_AFTER_END                      _HRESULT_TYPEDEF_(0x802a0103)
+#define UI_E_END_KEYFRAME_NOT_DETERMINED                   _HRESULT_TYPEDEF_(0x802a0104)
+#define UI_E_LOOPS_OVERLAP                                 _HRESULT_TYPEDEF_(0x802a0105)
+#define UI_E_TRANSITION_ALREADY_USED                       _HRESULT_TYPEDEF_(0x802a0106)
+#define UI_E_TRANSITION_NOT_IN_STORYBOARD                  _HRESULT_TYPEDEF_(0x802a0107)
+#define UI_E_TRANSITION_ECLIPSED                           _HRESULT_TYPEDEF_(0x802a0108)
+#define UI_E_TIME_BEFORE_LAST_UPDATE                       _HRESULT_TYPEDEF_(0x802a0109)
+#define UI_E_TIMER_CLIENT_ALREADY_CONNECTED                _HRESULT_TYPEDEF_(0x802a010a)
+#define UI_E_INVALID_DIMENSION                             _HRESULT_TYPEDEF_(0x802a010b)
+#define UI_E_PRIMITIVE_OUT_OF_BOUNDS                       _HRESULT_TYPEDEF_(0x802a010c)
+#define UI_E_WINDOW_CLOSED                                 _HRESULT_TYPEDEF_(0x802a0201)
+
 #define D3D11_ERROR_TOO_MANY_UNIQUE_STATE_OBJECTS          _HRESULT_TYPEDEF_(0x887c0001)
 #define D3D11_ERROR_FILE_NOT_FOUND                         _HRESULT_TYPEDEF_(0x887c0002)
 #define D3D11_ERROR_TOO_MANY_UNIQUE_VIEW_OBJECTS           _HRESULT_TYPEDEF_(0x887c0003)
-- 
2.17.1

