From f221b3118faad6fb105f35f126ee90111c7374f6 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 22 Jul 2019 14:33:37 +1000
Subject: [PATCH] include: Add IPreviewHandlerFrame interface declaration.

---
 include/shobjidl.idl | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/include/shobjidl.idl b/include/shobjidl.idl
index d459272a1d..42ef264ed8 100644
--- a/include/shobjidl.idl
+++ b/include/shobjidl.idl
@@ -3742,6 +3742,23 @@ interface IApplicationActivationManager : IUnknown
                                 [out] DWORD *processid);
 }
 
+[
+    object,
+    uuid(fec87aaf-35f9-447a-adb7-20234491401a),
+    pointer_default(unique)
+]
+interface IPreviewHandlerFrame : IUnknown
+{
+    typedef struct PREVIEWHANDLERFRAMEINFO
+    {
+        HACCEL haccel;
+        UINT   cAccelEntries;
+    } PREVIEWHANDLERFRAMEINFO;
+
+    HRESULT GetWindowContext([out] PREVIEWHANDLERFRAMEINFO *info);
+    HRESULT TranslateAccelerator([in] MSG *msg);
+}
+
 /*****************************************************************************
  * ShellObjects typelibrary
  */
-- 
2.17.1

