From d73358887b886b335d4dc5b61c62cf86d45fdac3 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 23 Jul 2019 10:38:32 +1000
Subject: [PATCH] include: Add recovery defines

---
 include/winbase.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/winbase.h b/include/winbase.h
index 3e75a4baf5..086a8b3645 100644
--- a/include/winbase.h
+++ b/include/winbase.h
@@ -1053,6 +1053,16 @@ typedef DWORD (CALLBACK *LPPROGRESS_ROUTINE)(LARGE_INTEGER, LARGE_INTEGER, LARGE
 
 typedef DWORD (WINAPI *APPLICATION_RECOVERY_CALLBACK)(PVOID);
 
+#define RECOVERY_DEFAULT_PING_INTERVAL  5000
+#define RECOVERY_MAX_PING_INTERVAL      (5*60*1000)
+
+#define RESTART_MAX_CMD_LINE    1024
+
+#define RESTART_NO_CRASH        1
+#define RESTART_NO_HANG         2
+#define RESTART_NO_PATCH        4
+#define RESTART_NO_REBOOT       8
+
 typedef enum _COPYFILE2_MESSAGE_TYPE
 {
     COPYFILE2_CALLBACK_NONE = 0,
-- 
2.17.1

