From 95daad9e565ab410b63f04c5d3aa8c2459d42893 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 23 Jul 2019 12:20:49 +1000
Subject: [PATCH] msvcrt: Add missing declaration

---
 include/msvcrt/wchar.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/msvcrt/wchar.h b/include/msvcrt/wchar.h
index 78df974f9a..a5dddb4247 100644
--- a/include/msvcrt/wchar.h
+++ b/include/msvcrt/wchar.h
@@ -431,6 +431,7 @@ int      __cdecl _wcsicmp(const wchar_t*,const wchar_t*);
 int      __cdecl _wcsicoll(const wchar_t*,const wchar_t*);
 int      __cdecl _wcsicoll_l(const wchar_t*,const wchar_t*,_locale_t);
 wchar_t* __cdecl _wcslwr(wchar_t*);
+errno_t  __cdecl _wcslwr_s(wchar_t*, size_t);
 int      __cdecl _wcsncoll(const wchar_t*,const wchar_t*,size_t);
 int      __cdecl _wcsncoll_l(const wchar_t*,const wchar_t*,size_t,_locale_t);
 int      __cdecl _wcsnicmp(const wchar_t*,const wchar_t*,size_t);
@@ -440,6 +441,7 @@ wchar_t* __cdecl _wcsnset(wchar_t*,wchar_t,size_t);
 wchar_t* __cdecl _wcsrev(wchar_t*);
 wchar_t* __cdecl _wcsset(wchar_t*,wchar_t);
 wchar_t* __cdecl _wcsupr(wchar_t*);
+errno_t  __cdecl _wcsupr_s(wchar_t*, size_t);
 
 wchar_t* __cdecl wcscat(wchar_t*,const wchar_t*);
 errno_t  __cdecl wcscat_s(wchar_t*,size_t,const wchar_t*);
@@ -451,9 +453,11 @@ errno_t  __cdecl wcscpy_s(wchar_t*,size_t,const wchar_t*);
 size_t   __cdecl wcscspn(const wchar_t*,const wchar_t*);
 size_t   __cdecl wcslen(const wchar_t*);
 wchar_t* __cdecl wcsncat(wchar_t*,const wchar_t*,size_t);
+errno_t  __cdecl wcsncat_s(wchar_t *, size_t, const wchar_t *, size_t);
 int      __cdecl wcsncmp(const wchar_t*,const wchar_t*,size_t);
 wchar_t* __cdecl wcsncpy(wchar_t*,const wchar_t*,size_t);
 errno_t  __cdecl wcsncpy_s(wchar_t*,size_t,const wchar_t*,size_t);
+size_t   __cdecl wcsnlen(const size_t*,size_t);
 wchar_t* __cdecl wcspbrk(const wchar_t*,const wchar_t*);
 wchar_t* __cdecl wcsrchr(const wchar_t*,wchar_t wcFor);
 size_t   __cdecl wcsspn(const wchar_t*,const wchar_t*);
@@ -515,6 +519,7 @@ static inline wchar_t* __cdecl wmemcpy(wchar_t *dst, const wchar_t *src, size_t
 {
     return memcpy(dst, src, n * sizeof(wchar_t));
 }
+errno_t __cdecl wmemcpy_s(wchar_t *, size_t, const wchar_t *, size_t);
 
 static inline wchar_t* __cdecl wmemmove(wchar_t *dst, const wchar_t *src, size_t n)
 {
-- 
2.17.1

