From c8c4eb7761f1f37c3a78d4e8a8bf93c77921c1b2 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 14 Jul 2025 10:15:14 +1000
Subject: [PATCH] msado15: ADOConnectionConstruction15::WrapDSOandSession semi
 stub

---
 dlls/msado15/connection.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/dlls/msado15/connection.c b/dlls/msado15/connection.c
index 278ef1e76f4..b491c624aa6 100644
--- a/dlls/msado15/connection.c
+++ b/dlls/msado15/connection.c
@@ -931,7 +931,17 @@ static HRESULT WINAPI adoconstruct_WrapDSOandSession(ADOConnectionConstruction15
 {
     struct connection *connection = impl_from_ADOConnectionConstruction15( iface );
     FIXME("%p, %p, %p\n", connection, dso, session);
-    return E_NOTIMPL;
+
+    if (connection->session)
+        IUnknown_Release( connection->session );
+    connection->session = session;
+    if (connection->session)
+        IUnknown_AddRef(connection->session);
+
+    if (dso && session)
+        connection->state = adStateOpen;
+
+    return S_OK;
 }
 
 struct ADOConnectionConstruction15Vtbl ado_construct_vtbl =
-- 
2.47.2

