From dcd0a8c549d9b277fdfe63e04170c4caca115c10 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 4 Dec 2017 15:29:21 +1100
Subject: [PATCH 3/3] dpnet: IDirectPlay8Server_Host start dpsvr.exe once

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/server.c    | 51 ++++++++++++++++++++++++++++++++++++++++++
 programs/dpnsvr/main.c | 12 ++++++++++
 2 files changed, 63 insertions(+)

diff --git a/dlls/dpnet/server.c b/dlls/dpnet/server.c
index 977771697ac..b21d952d45b 100644
--- a/dlls/dpnet/server.c
+++ b/dlls/dpnet/server.c
@@ -45,6 +45,8 @@ typedef struct IDirectPlay8ServerImpl
     WCHAR *servername;
     void  *data;
     DWORD datasize;
+
+    HANDLE dpnsrv;
 } IDirectPlay8ServerImpl;
 
 WINE_DEFAULT_DEBUG_CHANNEL(dpnet);
@@ -95,6 +97,8 @@ static ULONG WINAPI IDirectPlay8ServerImpl_Release(IDirectPlay8Server *iface)
 
     if (!ref)
     {
+        CloseHandle(This->dpnsrv);
+
         heap_free(This->servername);
         heap_free(This->data);
         heap_free(This);
@@ -246,6 +250,49 @@ static HRESULT WINAPI IDirectPlay8ServerImpl_Host(IDirectPlay8Server *iface, con
     IDirectPlay8ServerImpl *This = impl_from_IDirectPlay8Server(iface);
     FIXME("(%p)->(%p %p %d %p %p %p %d)\n", This, pdnAppDesc, prgpDeviceInfo, cDeviceInfo, pdnSecurity, pdnCredentials,
                     pvPlayerContext, dwFlags);
+
+    if(This->dpnsrv)
+    {
+        TRACE("Attempting to Host twice\n");
+        return DPNERR_ALREADYINITIALIZED;
+    }
+
+    This->dpnsrv = CreateEventA(NULL, TRUE, FALSE, "wine_dpnsvr_started");
+
+    if(WaitForSingleObject(This->dpnsrv,0) != WAIT_OBJECT_0)
+    {
+        static WCHAR dpnsvr[] = {'d','p','n','s','v','r','.','e','x','e', 0};
+        STARTUPINFOW si;
+        PROCESS_INFORMATION info;
+
+        memset(&si, 0, sizeof(si));
+        si.cb=sizeof(si);
+        si.dwFlags=STARTF_USESHOWWINDOW;
+
+        memset(&info, 0, sizeof(info));
+
+        TRACE("Starting dpnsvr.exe\n");
+
+        if(!CreateProcessW(NULL, dpnsvr, NULL, NULL, FALSE, 0, NULL, NULL, &si, &info))
+        {
+            WARN("Failed to start dpnsvr (%d)\n", GetLastError() );
+            CloseHandle(This->dpnsrv);
+            This->dpnsrv = NULL;
+
+            return DPNERR_DPNSVRNOTAVAILABLE;
+        }
+
+        CloseHandle(info.hThread);
+        CloseHandle(info.hProcess);
+    }
+
+    if(WaitForSingleObject(This->dpnsrv,20000) != WAIT_OBJECT_0)
+    {
+        CloseHandle(This->dpnsrv);
+        This->dpnsrv = NULL;
+        return DPNERR_DPNSVRNOTAVAILABLE;
+    }
+
     return E_NOTIMPL;
 }
 
@@ -324,6 +371,10 @@ static HRESULT WINAPI IDirectPlay8ServerImpl_Close(IDirectPlay8Server *iface, DW
 {
     IDirectPlay8ServerImpl *This = impl_from_IDirectPlay8Server(iface);
     FIXME("(%p)->(%d)\n", This, dwFlags);
+
+    CloseHandle(This->dpnsrv);
+    This->dpnsrv = NULL;
+
     return E_NOTIMPL;
 }
 
diff --git a/programs/dpnsvr/main.c b/programs/dpnsvr/main.c
index 0180f121a10..716c6c3886d 100644
--- a/programs/dpnsvr/main.c
+++ b/programs/dpnsvr/main.c
@@ -172,6 +172,14 @@ int __cdecl wmain(int argc, WCHAR *argv[])
     SOCKET remote;
     struct sockaddr_in addr;
     int addrlen;
+    HANDLE started;
+
+    started = CreateEventA(NULL, TRUE, FALSE, "wine_dpnsvr_started");
+    if(WaitForSingleObject(started, 0) == WAIT_OBJECT_0)
+    {
+        WINE_WARN("Attempting to start dpnsvr twice\n");
+        return 0;
+    }
 
     if (WSAStartup(MAKEWORD(2, 2), &data))
         return 1;
@@ -181,6 +189,8 @@ int __cdecl wmain(int argc, WCHAR *argv[])
     /* Use a socket to comminicate with applications that have loaded us. */
     setup_local_network();
 
+    SetEvent(started);
+
     while(loop < 30)
     {
         for(i=0; i < sock_cnt; i++)
@@ -212,6 +222,8 @@ int __cdecl wmain(int argc, WCHAR *argv[])
         loop++;
     }
 
+    CloseHandle(started);
+
     for(i=0; i < sock_cnt; i++)
     {
         closesocket(socks[i]);
-- 
2.30.2

