From 957a30cfe5495ec06663a86d0f398ccb8232ccee Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 9 Jul 2026 11:27:57 +1000
Subject: [PATCH] include: Add DRIVER_REGKEY_TYPE

---
 include/ddk/wdm.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/ddk/wdm.h b/include/ddk/wdm.h
index 406870d8f06..aff4877d43e 100644
--- a/include/ddk/wdm.h
+++ b/include/ddk/wdm.h
@@ -800,6 +800,11 @@ typedef enum _INTERFACE_TYPE {
   MaximumInterfaceType
 } INTERFACE_TYPE, *PINTERFACE_TYPE;
 
+typedef enum _DRIVER_REGKEY_TYPE {
+  DriverRegKeyParameters,
+  DriverRegKeyPersistentState,
+} DRIVER_REGKEY_TYPE, *PDRIVER_REGKEY_TYPE;
+
 typedef LARGE_INTEGER PHYSICAL_ADDRESS, *PPHYSICAL_ADDRESS;
 
 #define IO_RESOURCE_PREFERRED             0x01
-- 
2.53.0

