From baec8faa4e09f07460dfb57034a3f3c520f32dc0 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 9 Jul 2026 11:36:38 +1000
Subject: [PATCH] include: fltkernel.h - Add more defines.

---
 include/ddk/fltkernel.h | 56 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)

diff --git a/include/ddk/fltkernel.h b/include/ddk/fltkernel.h
index ce7f16e4ab8..426ce1309d2 100644
--- a/include/ddk/fltkernel.h
+++ b/include/ddk/fltkernel.h
@@ -48,6 +48,8 @@ typedef void*  PFLT_CONTEXT;
 #define FLT_STREAM_CONTEXT       0x0008
 #define FLT_STREAMHANDLE_CONTEXT 0x0010
 #define FLT_TRANSACTION_CONTEXT  0x0020
+#define FLT_SECTION_CONTEXT      0x0040
+#define FLT_CONTEXT_END          0xffff
 
 #define FLTFL_OPERATION_REGISTRATION_SKIP_PAGING_IO     0x00000001
 #define FLTFL_OPERATION_REGISTRATION_SKIP_CACHED_IO     0x00000002
@@ -59,6 +61,45 @@ typedef void*  PFLT_CONTEXT;
 #define FLTFL_INSTANCE_TEARDOWN_VOLUME_DISMOUNT         0x00000008
 #define FLTFL_INSTANCE_TEARDOWN_INTERNAL_ERROR          0x00000010
 
+#define IRP_MJ_ACQUIRE_FOR_SECTION_SYNCHRONIZATION     0xff
+#define IRP_MJ_RELEASE_FOR_SECTION_SYNCHRONIZATION     0xfe
+#define IRP_MJ_ACQUIRE_FOR_MOD_WRITE                   0xfd
+#define IRP_MJ_RELEASE_FOR_MOD_WRITE                   0xfc
+#define IRP_MJ_ACQUIRE_FOR_CC_FLUSH                    0xfb
+#define IRP_MJ_RELEASE_FOR_CC_FLUSH                    0xfa
+#define IRP_MJ_QUERY_OPEN                              0xf9
+#define IRP_MJ_FAST_IO_CHECK_IF_POSSIBLE               0xf3
+#define IRP_MJ_NETWORK_QUERY_OPEN                      0xf2
+#define IRP_MJ_MDL_READ                                0xf1
+#define IRP_MJ_MDL_READ_COMPLETE                       0xf0
+#define IRP_MJ_PREPARE_MDL_WRITE                       0xef
+#define IRP_MJ_MDL_WRITE_COMPLETE                      0xee
+#define IRP_MJ_VOLUME_MOUNT                            0xed
+#define IRP_MJ_VOLUME_DISMOUNT                         0xec
+
+#define IRP_MJ_OPERATION_END                           0x80
+
+#define FLT_PORT_CONNECT                               0x0001
+#define FLT_PORT_ALL_ACCESS                            (FLT_PORT_CONNECT | STANDARD_RIGHTS_ALL)
+
+#define FLT_REGISTRATION_VERSION                       0x0203
+
+#define FLT_VALID_FILE_NAME_FORMATS                    0x000000ff
+
+#define FLT_FILE_NAME_NORMALIZED                       0x01
+#define FLT_FILE_NAME_OPENED                           0x02
+#define FLT_FILE_NAME_SHORT                            0x03
+#define FltGetFileNameFormat( options )                ((options) & FLT_VALID_FILE_NAME_FORMATS)
+
+#define FLT_VALID_FILE_NAME_QUERY_METHODS               0x0000ff00
+
+#define FLT_FILE_NAME_QUERY_DEFAULT                     0x0100
+#define FLT_FILE_NAME_QUERY_CACHE_ONLY                  0x0200
+#define FLT_FILE_NAME_QUERY_FILESYSTEM_ONLY             0x0300
+#define FLT_FILE_NAME_QUERY_ALWAYS_ALLOW_CACHE_LOOKUP   0x0400
+
+#define FltGetFileNameQueryMethod( options )            ((options) & FLT_VALID_FILE_NAME_QUERY_METHODS)
+
 /* Belongs in fltuserstructures.h */
 typedef enum _FLT_FILESYSTEM_TYPE
 {
@@ -695,6 +736,21 @@ typedef struct _FLT_FILE_NAME_INFORMATION
     UNICODE_STRING ParentDir;
 } FLT_FILE_NAME_INFORMATION, *PFLT_FILE_NAME_INFORMATION;
 
+#define VOL_PROP_FL_DAX_VOLUME                      0x0001
+
+typedef struct _FLT_VOLUME_PROPERTIES
+{
+    DEVICE_TYPE    DeviceType;
+    ULONG          DeviceCharacteristics;
+    ULONG          DeviceObjectFlags;
+    ULONG          AlignmentRequirement;
+    USHORT         SectorSize;
+    USHORT         Flags;
+    UNICODE_STRING FileSystemDriverName;
+    UNICODE_STRING FileSystemDeviceName;
+    UNICODE_STRING RealDeviceName;
+} FLT_VOLUME_PROPERTIES, *PFLT_VOLUME_PROPERTIES;
+
 NTSTATUS WINAPI FltAllocateContext(PFLT_FILTER, FLT_CONTEXT_TYPE, SIZE_T, POOL_TYPE, PFLT_CONTEXT*);
 NTSTATUS WINAPI FltBuildDefaultSecurityDescriptor(PSECURITY_DESCRIPTOR *, ACCESS_MASK);
 void     WINAPI FltCloseCommunicationPort (PFLT_PORT);
-- 
2.53.0

