From b5aa76c89806311d5f394bad1fcf644bba0d60a3 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 9 Jul 2026 17:31:53 +1000
Subject: [PATCH] include: Add _FLT_RELATED_OBJECTS struct

---
 include/ddk/fltkernel.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/ddk/fltkernel.h b/include/ddk/fltkernel.h
index 681c6a277f4..57dd2af8ded 100644
--- a/include/ddk/fltkernel.h
+++ b/include/ddk/fltkernel.h
@@ -168,6 +168,20 @@ typedef struct _FLT_RELATED_CONTEXTS
     PFLT_CONTEXT TransactionContext;
 } FLT_RELATED_CONTEXTS, *PFLT_RELATED_CONTEXTS;
 
+typedef struct _KTRANSACTION *PKTRANSACTION;
+
+typedef struct _FLT_RELATED_OBJECTS
+{
+
+    USHORT const Size;
+    USHORT const TransactionContext;
+    PFLT_FILTER const Filter;
+    PFLT_VOLUME const Volume;
+    PFLT_INSTANCE const Instance;
+    PFILE_OBJECT const FileObject;
+    PKTRANSACTION const Transaction;
+} FLT_RELATED_OBJECTS, *PFLT_RELATED_OBJECTS;
+
 typedef const struct _FLT_RELATED_OBJECTS *PCFLT_RELATED_OBJECTS;
 
 typedef union _FLT_PARAMETERS
-- 
2.53.0

