From 56cd47b1d160c427772519183baaa8a6c6bd9dec Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 9 Jul 2026 22:59:01 +1000
Subject: [PATCH] fltlib: create import library

---
 dlls/fltlib/Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/fltlib/Makefile.in b/dlls/fltlib/Makefile.in
index 50760a220f4..37c490b320a 100644
--- a/dlls/fltlib/Makefile.in
+++ b/dlls/fltlib/Makefile.in
@@ -1,4 +1,5 @@
 MODULE    = fltlib.dll
+IMPORTLIB = fltlib
 
 VER_FILEDESCRIPTION_STR = "Wine Filter Library"
 VER_PRODUCTVERSION      = 5,1,2600,5512
-- 
2.53.0

