From 80431dfd39764b6aadbfa167b104caed25f95807 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Fri, 6 Sep 2019 16:22:32 +1000
Subject: [PATCH] dsound: Report when a DMO object has tailing data.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dsound/mixer.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/dsound/mixer.c b/dlls/dsound/mixer.c
index 05ca99b5df..53b5f9f576 100644
--- a/dlls/dsound/mixer.c
+++ b/dlls/dsound/mixer.c
@@ -443,8 +443,9 @@ static void DSOUND_MixToTemporary(IDirectSoundBufferImpl *dsb, DWORD frames)
 		for (i = 0; i < dsb->num_filters; i++) {
 			if (dsb->filters[i].inplace) {
 				hr = IMediaObjectInPlace_Process(dsb->filters[i].inplace, size_bytes, (BYTE*)dsb->device->tmp_buffer, 0, DMO_INPLACE_NORMAL);
-
-				if (FAILED(hr))
+				if(hr == S_FALSE)
+					FIXME("Processing of tailing data currently not implemented.\n");
+				else if (FAILED(hr))
 					WARN("IMediaObjectInPlace_Process failed for filter %u\n", i);
 			} else
 				WARN("filter %u has no inplace object - unsupported\n", i);
-- 
2.17.1

