From 43acfbf88a59015bf9cf5ad548cf2288ad183226 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 22 Oct 2025 19:06:33 +1100
Subject: [PATCH] msvcp90: Add time_get_wchar supprto for _Makeusloc

---
 dlls/msvcp90/locale.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/dlls/msvcp90/locale.c b/dlls/msvcp90/locale.c
index 1e1d6f40b0f..250995f45ef 100644
--- a/dlls/msvcp90/locale.c
+++ b/dlls/msvcp90/locale.c
@@ -12231,7 +12231,7 @@ locale__Locimp** __cdecl locale__Locimp__Clocptr_func(void)
 /* ?_Makeushloc@_Locimp@locale@std@@CAXABV_Locinfo@3@HPAV123@PBV23@@Z */
 /* ?_Makeushloc@_Locimp@locale@std@@CAXAEBV_Locinfo@3@HPEAV123@PEBV23@@Z */
 /* List of missing facets:
- * messages, money_get, money_put, moneypunct, moneypunct, time_get
+ * messages, money_get, money_put, moneypunct, moneypunct
  */
 void __cdecl locale__Locimp__Makeushloc(const _Locinfo *locinfo, category cat, locale__Locimp *locimp, const locale *loc)
 {
@@ -12320,6 +12320,18 @@ void __cdecl locale__Locimp__Makeushloc(const _Locinfo *locinfo, category cat, l
         }
         locale__Locimp__Addfac(locimp, &codecvt->base.facet, locale_id_operator_size_t(&codecvt_short_id));
     }
+
+    if(cat & (1<<(time_get_wchar__Getcat(NULL, NULL)-1))) {
+        time_get_wchar *t;
+
+        if(loc) {
+            t = time_get_wchar_use_facet(loc);
+        }else {
+            t = operator_new(sizeof(time_get_wchar));
+            time_get_wchar_ctor_locinfo(t, locinfo, 0);
+        }
+        locale__Locimp__Addfac(locimp, &t->facet, locale_id_operator_size_t(&time_get_wchar_id));
+    }
 }
 
 /* ?_Makewloc@_Locimp@locale@std@@CAXABV_Locinfo@3@HPAV123@PBV23@@Z */
-- 
2.51.0

