From 72a39a75ba85cfc889245eda93715e16ecbbd818 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 22 Dec 2015 10:07:30 +1100
Subject: [PATCH 3/4] odbccp32: Implement SQLInstallTranslator

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/odbccp32/odbccp32.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/dlls/odbccp32/odbccp32.c b/dlls/odbccp32/odbccp32.c
index 66754de..cb1058e 100644
--- a/dlls/odbccp32/odbccp32.c
+++ b/dlls/odbccp32/odbccp32.c
@@ -41,6 +41,7 @@ static const WCHAR drivers_key[] = {'S','o','f','t','w','a','r','e','\\','O','D'
 static const WCHAR odbcW[] = {'S','o','f','t','w','a','r','e','\\','O','D','B','C',0};
 static const WCHAR odbcini[] = {'S','o','f','t','w','a','r','e','\\','O','D','B','C','\\','O','D','B','C','I','N','S','T','.','I','N','I','\\',0};
 static const WCHAR odbcdrivers[] = {'O','D','B','C',' ','D','r','i','v','e','r','s',0};
+static const WCHAR odbctranslators[] = {'O','D','B','C',' ','T','r','a','n','s','l','a','t','o','r','s',0};
 
 /* This config mode is known to be process-wide.
  * MSDN documentation suggests that the value is hidden somewhere in the registry but I haven't found it yet.
@@ -635,6 +636,7 @@ static void write_registry_values(const WCHAR *regkey, const WCHAR *driver, cons
     static const WCHAR slash[] = {'\\', 0};
     static const WCHAR driverW[] = {'D','r','i','v','e','r',0};
     static const WCHAR setupW[] = {'S','e','t','u','p',0};
+    static const WCHAR translator[] = {'T','r','a','n','s','l','a','t','o','r',0};
     HKEY hkey, hkeydriver;
 
     if (RegCreateKeyW(HKEY_LOCAL_MACHINE, odbcini, &hkey) == ERROR_SUCCESS)
@@ -675,8 +677,9 @@ static void write_registry_values(const WCHAR *regkey, const WCHAR *driver, cons
                     divider++;
                     TRACE("Writing pair %s,%s\n", debugstr_w(entry), debugstr_w(divider));
 
-                    /* Driver and Setup entries use the system path unless a path is specified. */
-                    if(lstrcmpiW(driverW, entry) == 0 || lstrcmpiW(setupW, entry) == 0)
+                    /* Driver, Setup, Translator entries use the system path unless a path is specified. */
+                    if(lstrcmpiW(driverW, entry) == 0 || lstrcmpiW(setupW, entry) == 0 ||
+                       lstrcmpiW(translator, entry) == 0)
                     {
                         lstrcpyW(value, path);
                         lstrcatW(value, slash);
@@ -949,7 +952,6 @@ BOOL WINAPI SQLInstallTranslatorExW(LPCWSTR lpszTranslator, LPCWSTR lpszPathIn,
                WORD fRequest, LPDWORD lpdwUsageCount)
 {
     UINT len;
-    LPCWSTR p;
     WCHAR path[MAX_PATH];
 
     clear_errors();
@@ -957,14 +959,16 @@ BOOL WINAPI SQLInstallTranslatorExW(LPCWSTR lpszTranslator, LPCWSTR lpszPathIn,
           debugstr_w(lpszPathIn), lpszPathOut, cbPathOutMax, pcbPathOut,
           fRequest, lpdwUsageCount);
 
-    for (p = lpszTranslator; *p; p += lstrlenW(p) + 1)
-        TRACE("%s\n", debugstr_w(p));
+    write_registry_values(odbctranslators, lpszTranslator, lpszPathIn, path);
 
-    len = GetSystemDirectoryW(path, MAX_PATH);
+    len = lstrlenW(path);
 
     if (pcbPathOut)
         *pcbPathOut = len;
 
+    if(lpdwUsageCount)
+        *lpdwUsageCount = 1;
+
     if (lpszPathOut && cbPathOutMax > len)
     {
         lstrcpyW(lpszPathOut, path);
-- 
1.9.1

