From 1fdde8d91629063711e95b320bf717fba8b75e45 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 10 Jul 2017 14:27:58 +1000
Subject: [PATCH 2/2] include: Add Proxy stubs for IUnknown

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 include/unknwn.idl | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/unknwn.idl b/include/unknwn.idl
index 78a2d9d..7856a7f 100644
--- a/include/unknwn.idl
+++ b/include/unknwn.idl
@@ -46,6 +46,16 @@ interface IUnknown
   ULONG Release();
 }
 
+cpp_quote("HRESULT STDMETHODCALLTYPE IUnknown_QueryInterface_Proxy(IUnknown* This, REFIID riid, void **ppvObject);")
+cpp_quote("void __RPC_STUB IUnknown_QueryInterface_Stub(IRpcStubBuffer* This, IRpcChannelBuffer* pRpcChannelBuffer,")
+cpp_quote("     PRPC_MESSAGE pRpcMessage, DWORD* pdwStubPhase);")
+cpp_quote("ULONG STDMETHODCALLTYPE IUnknown_AddRef_Proxy(IUnknown* This);")
+cpp_quote("void __RPC_STUB IUnknown_AddRef_Stub(IRpcStubBuffer* This, IRpcChannelBuffer* pRpcChannelBuffer,")
+cpp_quote("     PRPC_MESSAGE pRpcMessage, DWORD* pdwStubPhase);")
+cpp_quote("ULONG STDMETHODCALLTYPE IUnknown_Release_Proxy(IUnknown* This);")
+cpp_quote("void __RPC_STUB IUnknown_Release_Stub(IRpcStubBuffer* This, IRpcChannelBuffer* pRpcChannelBuffer,")
+cpp_quote("     PRPC_MESSAGE pRpcMessage, DWORD* pdwStubPhase);")
+
 cpp_quote("#ifdef SAVE_WIDL_C_INLINE_WRAPPERS")
 cpp_quote("#define WIDL_C_INLINE_WRAPPERS")
 cpp_quote("#endif")
-- 
1.9.1

