From 782afbcbb5203fb80a6304c26a7c8dd2f3ccb09d Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 10 Apr 2017 10:45:29 +1000
Subject: [v2 PATCH 1/2] odbccp32: Look in LOCAL_MACHINE for the driver

v2 - No Change.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/odbccp32/odbccp32.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/odbccp32/odbccp32.c b/dlls/odbccp32/odbccp32.c
index 666d8c0..f28d464 100644
--- a/dlls/odbccp32/odbccp32.c
+++ b/dlls/odbccp32/odbccp32.c
@@ -260,7 +260,7 @@ static HMODULE load_config_driver(const WCHAR *driver)
     DWORD size = 0, type;
     HKEY hkey;
 
-    if ((ret = RegOpenKeyW(HKEY_CURRENT_USER, odbcini, &hkey)) == ERROR_SUCCESS)
+    if ((ret = RegOpenKeyW(HKEY_LOCAL_MACHINE, odbcini, &hkey)) == ERROR_SUCCESS)
     {
         HKEY hkeydriver;
 
-- 
1.9.1

