From 870358ebb1b44194b49c94f9ef76961dc9fb6a0b Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 28 Nov 2016 08:49:36 +1100
Subject: [PATCH 1/4] d3d11: Check parameter in ID3D11Device
 CreateShaderResourceView

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/d3d11/device.c      | 3 +++
 dlls/d3d11/tests/d3d11.c | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/dlls/d3d11/device.c b/dlls/d3d11/device.c
index 1cb033d..fd51aa2 100644
--- a/dlls/d3d11/device.c
+++ b/dlls/d3d11/device.c
@@ -2186,6 +2186,9 @@ static HRESULT STDMETHODCALLTYPE d3d11_device_CreateShaderResourceView(ID3D11Dev
 
     TRACE("iface %p, resource %p, desc %p, view %p.\n", iface, resource, desc, view);
 
+    if (!resource)
+        return E_INVALIDARG;
+
     if (FAILED(hr = d3d_shader_resource_view_create(device, resource, desc, &object)))
         return hr;
 
diff --git a/dlls/d3d11/tests/d3d11.c b/dlls/d3d11/tests/d3d11.c
index 496b715..515cd02 100644
--- a/dlls/d3d11/tests/d3d11.c
+++ b/dlls/d3d11/tests/d3d11.c
@@ -3283,6 +3283,9 @@ static void test_create_shader_resource_view(void)
     U(srv_desc).Buffer.ElementOffset = 0;
     U(srv_desc).Buffer.ElementWidth = 64;
 
+    hr = ID3D11Device_CreateShaderResourceView(device, NULL, &srv_desc, &srview);
+    ok(hr == E_INVALIDARG, "Got unexpected hr %#x.\n", hr);
+
     expected_refcount = get_refcount((IUnknown *)device) + 1;
     hr = ID3D11Device_CreateShaderResourceView(device, (ID3D11Resource *)buffer, &srv_desc, &srview);
     ok(SUCCEEDED(hr), "Failed to create a shader resource view, hr %#x.\n", hr);
-- 
1.9.1

