From 4c18e51a85afc29a9304b5fce9ed6b360f2a998e Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Thu, 15 Aug 2024 10:33:02 +1000
Subject: [PATCH] shell32: Allow FindExecutables to find unix files.

---
 dlls/shell32/shlexec.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/dlls/shell32/shlexec.c b/dlls/shell32/shlexec.c
index 828c7168a34..52e90b03b0d 100644
--- a/dlls/shell32/shlexec.c
+++ b/dlls/shell32/shlexec.c
@@ -677,7 +677,15 @@ static UINT SHELL_FindExecutable(LPCWSTR lpPath, LPCWSTR lpFile, LPCWSTR lpVerb,
             /* The file was found in lpPath or one of the directories in the system-wide search path */
         }
         else
-            xlpFile[0] = '\0';
+        {
+            /* Checking for an unix application */
+            if( (xlpFile[0] == '/') || PathFindOnPathW(xlpFile, search_paths))
+            {
+                lstrcpyW(lpResult, xlpFile);
+            }
+            else
+                xlpFile[0] = '\0';
+        }
     }
 
     attribs = GetFileAttributesW(lpFile);
-- 
2.34.1

