From 426302d02b2bc86f3d14654d80e3e413d6e72cf3 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Wed, 21 Aug 2024 13:59:06 +1000
Subject: [PATCH 2/3] odbccp32: SQLRemoveDSNFromIniW check for valid DSN before
 delete

---
 dlls/odbccp32/odbccp32.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/odbccp32/odbccp32.c b/dlls/odbccp32/odbccp32.c
index ab745ef1bb7..b77c5698c07 100644
--- a/dlls/odbccp32/odbccp32.c
+++ b/dlls/odbccp32/odbccp32.c
@@ -1586,6 +1586,12 @@ BOOL WINAPI SQLRemoveDSNFromIniW(LPCWSTR lpszDSN)
 
     TRACE("%s\n", debugstr_w(lpszDSN));
 
+    if (!SQLValidDSNW(lpszDSN))
+    {
+        push_error(ODBC_ERROR_INVALID_DSN, L"Invalid DSN");
+        return FALSE;
+    }
+
     clear_errors();
 
     if (RegOpenKeyW(HKEY_LOCAL_MACHINE, L"Software\\ODBC\\ODBC.INI\\ODBC Data Sources", &hkey) == ERROR_SUCCESS)
-- 
2.43.0

